// @hubspot/api-client/lib/codegen/cms/blogs/blog_posts/model/angle.js
"use strict";
/**
 * Blog Post endpoints
 * \"Use these endpoints for interacting with Blog Posts, Blog Authors, and Blog Tags\"
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Angle = void 0;
class Angle {
    static getAttributeTypeMap() {
        return Angle.attributeTypeMap;
    }
}
exports.Angle = Angle;
Angle.discriminator = undefined;
Angle.attributeTypeMap = [
    {
        "name": "value",
        "baseName": "value",
        "type": "number"
    },
    {
        "name": "units",
        "baseName": "units",
        "type": "Angle.UnitsEnum"
    }
];
(function (Angle) {
    let UnitsEnum;
    (function (UnitsEnum) {
        UnitsEnum[UnitsEnum["Deg"] = 'deg'] = "Deg";
        UnitsEnum[UnitsEnum["Grad"] = 'grad'] = "Grad";
        UnitsEnum[UnitsEnum["Rad"] = 'rad'] = "Rad";
        UnitsEnum[UnitsEnum["Turn"] = 'turn'] = "Turn";
    })(UnitsEnum = Angle.UnitsEnum || (Angle.UnitsEnum = {}));
})(Angle = exports.Angle || (exports.Angle = {}));
