// @hubspot/api-client/lib/codegen/cms/blogs/blog_posts/model/backgroundImage.js
"use strict";
/**
 * Blog Post endpoints
 * \"Use these endpoints for interacting with Blog Posts, Blog Authors, and Blog Tags\"
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackgroundImage = void 0;
class BackgroundImage {
    static getAttributeTypeMap() {
        return BackgroundImage.attributeTypeMap;
    }
}
exports.BackgroundImage = BackgroundImage;
BackgroundImage.discriminator = undefined;
BackgroundImage.attributeTypeMap = [
    {
        "name": "imageUrl",
        "baseName": "imageUrl",
        "type": "string"
    },
    {
        "name": "backgroundSize",
        "baseName": "backgroundSize",
        "type": "string"
    },
    {
        "name": "backgroundPosition",
        "baseName": "backgroundPosition",
        "type": "BackgroundImage.BackgroundPositionEnum"
    }
];
(function (BackgroundImage) {
    let BackgroundPositionEnum;
    (function (BackgroundPositionEnum) {
        BackgroundPositionEnum[BackgroundPositionEnum["TOPLEFT"] = 'TOP_LEFT'] = "TOPLEFT";
        BackgroundPositionEnum[BackgroundPositionEnum["TOPCENTER"] = 'TOP_CENTER'] = "TOPCENTER";
        BackgroundPositionEnum[BackgroundPositionEnum["TOPRIGHT"] = 'TOP_RIGHT'] = "TOPRIGHT";
        BackgroundPositionEnum[BackgroundPositionEnum["MIDDLELEFT"] = 'MIDDLE_LEFT'] = "MIDDLELEFT";
        BackgroundPositionEnum[BackgroundPositionEnum["MIDDLECENTER"] = 'MIDDLE_CENTER'] = "MIDDLECENTER";
        BackgroundPositionEnum[BackgroundPositionEnum["MIDDLERIGHT"] = 'MIDDLE_RIGHT'] = "MIDDLERIGHT";
        BackgroundPositionEnum[BackgroundPositionEnum["BOTTOMLEFT"] = 'BOTTOM_LEFT'] = "BOTTOMLEFT";
        BackgroundPositionEnum[BackgroundPositionEnum["BOTTOMCENTER"] = 'BOTTOM_CENTER'] = "BOTTOMCENTER";
        BackgroundPositionEnum[BackgroundPositionEnum["BOTTOMRIGHT"] = 'BOTTOM_RIGHT'] = "BOTTOMRIGHT";
    })(BackgroundPositionEnum = BackgroundImage.BackgroundPositionEnum || (BackgroundImage.BackgroundPositionEnum = {}));
})(BackgroundImage = exports.BackgroundImage || (exports.BackgroundImage = {}));
