// @hubspot/api-client/lib/codegen/cms/blogs/blog_posts/model/styles.js
"use strict";
/**
 * Blog Post endpoints
 * \"Use these endpoints for interacting with Blog Posts, Blog Authors, and Blog Tags\"
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Styles = void 0;
class Styles {
    static getAttributeTypeMap() {
        return Styles.attributeTypeMap;
    }
}
exports.Styles = Styles;
Styles.discriminator = undefined;
Styles.attributeTypeMap = [
    {
        "name": "verticalAlignment",
        "baseName": "verticalAlignment",
        "type": "Styles.VerticalAlignmentEnum"
    },
    {
        "name": "backgroundColor",
        "baseName": "backgroundColor",
        "type": "RGBAColor"
    },
    {
        "name": "backgroundImage",
        "baseName": "backgroundImage",
        "type": "BackgroundImage"
    },
    {
        "name": "backgroundGradient",
        "baseName": "backgroundGradient",
        "type": "Gradient"
    },
    {
        "name": "maxWidthSectionCentering",
        "baseName": "maxWidthSectionCentering",
        "type": "number"
    },
    {
        "name": "forceFullWidthSection",
        "baseName": "forceFullWidthSection",
        "type": "boolean"
    },
    {
        "name": "flexboxPositioning",
        "baseName": "flexboxPositioning",
        "type": "Styles.FlexboxPositioningEnum"
    }
];
(function (Styles) {
    let VerticalAlignmentEnum;
    (function (VerticalAlignmentEnum) {
        VerticalAlignmentEnum[VerticalAlignmentEnum["TOP"] = 'TOP'] = "TOP";
        VerticalAlignmentEnum[VerticalAlignmentEnum["MIDDLE"] = 'MIDDLE'] = "MIDDLE";
        VerticalAlignmentEnum[VerticalAlignmentEnum["BOTTOM"] = 'BOTTOM'] = "BOTTOM";
    })(VerticalAlignmentEnum = Styles.VerticalAlignmentEnum || (Styles.VerticalAlignmentEnum = {}));
    let FlexboxPositioningEnum;
    (function (FlexboxPositioningEnum) {
        FlexboxPositioningEnum[FlexboxPositioningEnum["TOPLEFT"] = 'TOP_LEFT'] = "TOPLEFT";
        FlexboxPositioningEnum[FlexboxPositioningEnum["TOPCENTER"] = 'TOP_CENTER'] = "TOPCENTER";
        FlexboxPositioningEnum[FlexboxPositioningEnum["TOPRIGHT"] = 'TOP_RIGHT'] = "TOPRIGHT";
        FlexboxPositioningEnum[FlexboxPositioningEnum["MIDDLELEFT"] = 'MIDDLE_LEFT'] = "MIDDLELEFT";
        FlexboxPositioningEnum[FlexboxPositioningEnum["MIDDLECENTER"] = 'MIDDLE_CENTER'] = "MIDDLECENTER";
        FlexboxPositioningEnum[FlexboxPositioningEnum["MIDDLERIGHT"] = 'MIDDLE_RIGHT'] = "MIDDLERIGHT";
        FlexboxPositioningEnum[FlexboxPositioningEnum["BOTTOMLEFT"] = 'BOTTOM_LEFT'] = "BOTTOMLEFT";
        FlexboxPositioningEnum[FlexboxPositioningEnum["BOTTOMCENTER"] = 'BOTTOM_CENTER'] = "BOTTOMCENTER";
        FlexboxPositioningEnum[FlexboxPositioningEnum["BOTTOMRIGHT"] = 'BOTTOM_RIGHT'] = "BOTTOMRIGHT";
    })(FlexboxPositioningEnum = Styles.FlexboxPositioningEnum || (Styles.FlexboxPositioningEnum = {}));
})(Styles = exports.Styles || (exports.Styles = {}));
