// @hubspot/api-client/lib/codegen/crm/companies/model/batchResponseSimplePublicObjectWithErrors.js
"use strict";
/**
 * Companies
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseSimplePublicObjectWithErrors = void 0;
class BatchResponseSimplePublicObjectWithErrors {
    static getAttributeTypeMap() {
        return BatchResponseSimplePublicObjectWithErrors.attributeTypeMap;
    }
}
exports.BatchResponseSimplePublicObjectWithErrors = BatchResponseSimplePublicObjectWithErrors;
BatchResponseSimplePublicObjectWithErrors.discriminator = undefined;
BatchResponseSimplePublicObjectWithErrors.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseSimplePublicObjectWithErrors.StatusEnum"
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<SimplePublicObject>"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number"
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date"
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }"
    }
];
(function (BatchResponseSimplePublicObjectWithErrors) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["PENDING"] = 'PENDING'] = "PENDING";
        StatusEnum[StatusEnum["PROCESSING"] = 'PROCESSING'] = "PROCESSING";
        StatusEnum[StatusEnum["CANCELED"] = 'CANCELED'] = "CANCELED";
        StatusEnum[StatusEnum["COMPLETE"] = 'COMPLETE'] = "COMPLETE";
    })(StatusEnum = BatchResponseSimplePublicObjectWithErrors.StatusEnum || (BatchResponseSimplePublicObjectWithErrors.StatusEnum = {}));
})(BatchResponseSimplePublicObjectWithErrors = exports.BatchResponseSimplePublicObjectWithErrors || (exports.BatchResponseSimplePublicObjectWithErrors = {}));
