// @hubspot/api-client/lib/codegen/crm/companies/model/models.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoidAuth = exports.OAuth = exports.ApiKeyAuth = exports.HttpBearerAuth = exports.HttpBasicAuth = exports.ObjectSerializer = void 0;
__exportStar(require("./associatedId"), exports);
__exportStar(require("./batchInputSimplePublicObjectBatchInput"), exports);
__exportStar(require("./batchInputSimplePublicObjectId"), exports);
__exportStar(require("./batchInputSimplePublicObjectInput"), exports);
__exportStar(require("./batchReadInputSimplePublicObjectId"), exports);
__exportStar(require("./batchResponseSimplePublicObject"), exports);
__exportStar(require("./batchResponseSimplePublicObjectWithErrors"), exports);
__exportStar(require("./collectionResponseAssociatedId"), exports);
__exportStar(require("./collectionResponseAssociatedIdForwardPaging"), exports);
__exportStar(require("./collectionResponseSimplePublicObjectWithAssociationsForwardPaging"), exports);
__exportStar(require("./collectionResponseWithTotalSimplePublicObjectForwardPaging"), exports);
__exportStar(require("./errorCategory"), exports);
__exportStar(require("./errorDetail"), exports);
__exportStar(require("./filter"), exports);
__exportStar(require("./filterGroup"), exports);
__exportStar(require("./forwardPaging"), exports);
__exportStar(require("./modelError"), exports);
__exportStar(require("./nextPage"), exports);
__exportStar(require("./paging"), exports);
__exportStar(require("./previousPage"), exports);
__exportStar(require("./publicObjectSearchRequest"), exports);
__exportStar(require("./simplePublicObject"), exports);
__exportStar(require("./simplePublicObjectBatchInput"), exports);
__exportStar(require("./simplePublicObjectId"), exports);
__exportStar(require("./simplePublicObjectInput"), exports);
__exportStar(require("./simplePublicObjectWithAssociations"), exports);
__exportStar(require("./standardError"), exports);
const associatedId_1 = require("./associatedId");
const batchInputSimplePublicObjectBatchInput_1 = require("./batchInputSimplePublicObjectBatchInput");
const batchInputSimplePublicObjectId_1 = require("./batchInputSimplePublicObjectId");
const batchInputSimplePublicObjectInput_1 = require("./batchInputSimplePublicObjectInput");
const batchReadInputSimplePublicObjectId_1 = require("./batchReadInputSimplePublicObjectId");
const batchResponseSimplePublicObject_1 = require("./batchResponseSimplePublicObject");
const batchResponseSimplePublicObjectWithErrors_1 = require("./batchResponseSimplePublicObjectWithErrors");
const collectionResponseAssociatedId_1 = require("./collectionResponseAssociatedId");
const collectionResponseAssociatedIdForwardPaging_1 = require("./collectionResponseAssociatedIdForwardPaging");
const collectionResponseSimplePublicObjectWithAssociationsForwardPaging_1 = require("./collectionResponseSimplePublicObjectWithAssociationsForwardPaging");
const collectionResponseWithTotalSimplePublicObjectForwardPaging_1 = require("./collectionResponseWithTotalSimplePublicObjectForwardPaging");
const errorCategory_1 = require("./errorCategory");
const errorDetail_1 = require("./errorDetail");
const filter_1 = require("./filter");
const filterGroup_1 = require("./filterGroup");
const forwardPaging_1 = require("./forwardPaging");
const modelError_1 = require("./modelError");
const nextPage_1 = require("./nextPage");
const paging_1 = require("./paging");
const previousPage_1 = require("./previousPage");
const publicObjectSearchRequest_1 = require("./publicObjectSearchRequest");
const simplePublicObject_1 = require("./simplePublicObject");
const simplePublicObjectBatchInput_1 = require("./simplePublicObjectBatchInput");
const simplePublicObjectId_1 = require("./simplePublicObjectId");
const simplePublicObjectInput_1 = require("./simplePublicObjectInput");
const simplePublicObjectWithAssociations_1 = require("./simplePublicObjectWithAssociations");
const standardError_1 = require("./standardError");
/* tslint:disable:no-unused-variable */
let primitives = [
    "string",
    "boolean",
    "double",
    "integer",
    "long",
    "float",
    "number",
    "any"
];
let enumsMap = {
    "BatchResponseSimplePublicObject.StatusEnum": batchResponseSimplePublicObject_1.BatchResponseSimplePublicObject.StatusEnum,
    "BatchResponseSimplePublicObjectWithErrors.StatusEnum": batchResponseSimplePublicObjectWithErrors_1.BatchResponseSimplePublicObjectWithErrors.StatusEnum,
    "ErrorCategory.HttpStatusEnum": errorCategory_1.ErrorCategory.HttpStatusEnum,
    "Filter.OperatorEnum": filter_1.Filter.OperatorEnum,
};
let typeMap = {
    "AssociatedId": associatedId_1.AssociatedId,
    "BatchInputSimplePublicObjectBatchInput": batchInputSimplePublicObjectBatchInput_1.BatchInputSimplePublicObjectBatchInput,
    "BatchInputSimplePublicObjectId": batchInputSimplePublicObjectId_1.BatchInputSimplePublicObjectId,
    "BatchInputSimplePublicObjectInput": batchInputSimplePublicObjectInput_1.BatchInputSimplePublicObjectInput,
    "BatchReadInputSimplePublicObjectId": batchReadInputSimplePublicObjectId_1.BatchReadInputSimplePublicObjectId,
    "BatchResponseSimplePublicObject": batchResponseSimplePublicObject_1.BatchResponseSimplePublicObject,
    "BatchResponseSimplePublicObjectWithErrors": batchResponseSimplePublicObjectWithErrors_1.BatchResponseSimplePublicObjectWithErrors,
    "CollectionResponseAssociatedId": collectionResponseAssociatedId_1.CollectionResponseAssociatedId,
    "CollectionResponseAssociatedIdForwardPaging": collectionResponseAssociatedIdForwardPaging_1.CollectionResponseAssociatedIdForwardPaging,
    "CollectionResponseSimplePublicObjectWithAssociationsForwardPaging": collectionResponseSimplePublicObjectWithAssociationsForwardPaging_1.CollectionResponseSimplePublicObjectWithAssociationsForwardPaging,
    "CollectionResponseWithTotalSimplePublicObjectForwardPaging": collectionResponseWithTotalSimplePublicObjectForwardPaging_1.CollectionResponseWithTotalSimplePublicObjectForwardPaging,
    "ErrorCategory": errorCategory_1.ErrorCategory,
    "ErrorDetail": errorDetail_1.ErrorDetail,
    "Filter": filter_1.Filter,
    "FilterGroup": filterGroup_1.FilterGroup,
    "ForwardPaging": forwardPaging_1.ForwardPaging,
    "ModelError": modelError_1.ModelError,
    "NextPage": nextPage_1.NextPage,
    "Paging": paging_1.Paging,
    "PreviousPage": previousPage_1.PreviousPage,
    "PublicObjectSearchRequest": publicObjectSearchRequest_1.PublicObjectSearchRequest,
    "SimplePublicObject": simplePublicObject_1.SimplePublicObject,
    "SimplePublicObjectBatchInput": simplePublicObjectBatchInput_1.SimplePublicObjectBatchInput,
    "SimplePublicObjectId": simplePublicObjectId_1.SimplePublicObjectId,
    "SimplePublicObjectInput": simplePublicObjectInput_1.SimplePublicObjectInput,
    "SimplePublicObjectWithAssociations": simplePublicObjectWithAssociations_1.SimplePublicObjectWithAssociations,
    "StandardError": standardError_1.StandardError,
};
class ObjectSerializer {
    static findCorrectType(data, expectedType) {
        if (data == undefined) {
            return expectedType;
        }
        else if (primitives.indexOf(expectedType.toLowerCase()) !== -1) {
            return expectedType;
        }
        else if (expectedType === "Date") {
            return expectedType;
        }
        else {
            if (enumsMap[expectedType]) {
                return expectedType;
            }
            if (!typeMap[expectedType]) {
                return expectedType; // w/e we don't know the type
            }
            // Check the discriminator
            let discriminatorProperty = typeMap[expectedType].discriminator;
            if (discriminatorProperty == null) {
                return expectedType; // the type does not have a discriminator. use it.
            }
            else {
                if (data[discriminatorProperty]) {
                    var discriminatorType = data[discriminatorProperty];
                    if (typeMap[discriminatorType]) {
                        return discriminatorType; // use the type given in the discriminator
                    }
                    else {
                        return expectedType; // discriminator did not map to a type
                    }
                }
                else {
                    return expectedType; // discriminator was not present (or an empty string)
                }
            }
        }
    }
    static serialize(data, type) {
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index in data) {
                let date = data[index];
                transformedData.push(ObjectSerializer.serialize(date, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return data.toISOString();
        }
        else {
            if (enumsMap[type]) {
                return data;
            }
            if (!typeMap[type]) { // in case we dont know the type
                return data;
            }
            // Get the actual type of this object
            type = this.findCorrectType(data, type);
            // get the map for the correct type.
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            let instance = {};
            for (let index in attributeTypes) {
                let attributeType = attributeTypes[index];
                instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type);
            }
            return instance;
        }
    }
    static deserialize(data, type) {
        // polymorphism may change the actual type.
        type = ObjectSerializer.findCorrectType(data, type);
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index in data) {
                let date = data[index];
                transformedData.push(ObjectSerializer.deserialize(date, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return new Date(data);
        }
        else {
            if (enumsMap[type]) { // is Enum
                return data;
            }
            if (!typeMap[type]) { // dont know the type
                return data;
            }
            let instance = new typeMap[type]();
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            for (let index in attributeTypes) {
                let attributeType = attributeTypes[index];
                instance[attributeType.name] = ObjectSerializer.deserialize(data[attributeType.baseName], attributeType.type);
            }
            return instance;
        }
    }
}
exports.ObjectSerializer = ObjectSerializer;
class HttpBasicAuth {
    constructor() {
        this.username = '';
        this.password = '';
    }
    applyToRequest(requestOptions) {
        requestOptions.auth = {
            username: this.username, password: this.password
        };
    }
}
exports.HttpBasicAuth = HttpBasicAuth;
class HttpBearerAuth {
    constructor() {
        this.accessToken = '';
    }
    applyToRequest(requestOptions) {
        if (requestOptions && requestOptions.headers) {
            const accessToken = typeof this.accessToken === 'function'
                ? this.accessToken()
                : this.accessToken;
            requestOptions.headers["Authorization"] = "Bearer " + accessToken;
        }
    }
}
exports.HttpBearerAuth = HttpBearerAuth;
class ApiKeyAuth {
    constructor(location, paramName) {
        this.location = location;
        this.paramName = paramName;
        this.apiKey = '';
    }
    applyToRequest(requestOptions) {
        if (this.location == "query") {
            requestOptions.qs[this.paramName] = this.apiKey;
        }
        else if (this.location == "header" && requestOptions && requestOptions.headers) {
            requestOptions.headers[this.paramName] = this.apiKey;
        }
        else if (this.location == 'cookie' && requestOptions && requestOptions.headers) {
            if (requestOptions.headers['Cookie']) {
                requestOptions.headers['Cookie'] += '; ' + this.paramName + '=' + encodeURIComponent(this.apiKey);
            }
            else {
                requestOptions.headers['Cookie'] = this.paramName + '=' + encodeURIComponent(this.apiKey);
            }
        }
    }
}
exports.ApiKeyAuth = ApiKeyAuth;
class OAuth {
    constructor() {
        this.accessToken = '';
    }
    applyToRequest(requestOptions) {
        if (requestOptions && requestOptions.headers) {
            requestOptions.headers["Authorization"] = "Bearer " + this.accessToken;
        }
    }
}
exports.OAuth = OAuth;
class VoidAuth {
    constructor() {
        this.username = '';
        this.password = '';
    }
    applyToRequest(_) {
        // Do nothing
    }
}
exports.VoidAuth = VoidAuth;
