// @hubspot/api-client/lib/codegen/cms/performance/model/publicPerformanceResponse.js
"use strict";
/**
 * CMS Performance API
 * Use these endpoints to get a time series view of your website\'s performance.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicPerformanceResponse = void 0;
class PublicPerformanceResponse {
    static getAttributeTypeMap() {
        return PublicPerformanceResponse.attributeTypeMap;
    }
}
exports.PublicPerformanceResponse = PublicPerformanceResponse;
PublicPerformanceResponse.discriminator = undefined;
PublicPerformanceResponse.attributeTypeMap = [
    {
        "name": "data",
        "baseName": "data",
        "type": "Array<PerformanceView>"
    },
    {
        "name": "domain",
        "baseName": "domain",
        "type": "string"
    },
    {
        "name": "path",
        "baseName": "path",
        "type": "string"
    },
    {
        "name": "startInterval",
        "baseName": "startInterval",
        "type": "number"
    },
    {
        "name": "endInterval",
        "baseName": "endInterval",
        "type": "number"
    },
    {
        "name": "interval",
        "baseName": "interval",
        "type": "PublicPerformanceResponse.IntervalEnum"
    },
    {
        "name": "period",
        "baseName": "period",
        "type": "PublicPerformanceResponse.PeriodEnum"
    }
];
(function (PublicPerformanceResponse) {
    let IntervalEnum;
    (function (IntervalEnum) {
        IntervalEnum[IntervalEnum["ONEMINUTE"] = 'ONE_MINUTE'] = "ONEMINUTE";
        IntervalEnum[IntervalEnum["FIVEMINUTES"] = 'FIVE_MINUTES'] = "FIVEMINUTES";
        IntervalEnum[IntervalEnum["TENMINUTES"] = 'TEN_MINUTES'] = "TENMINUTES";
        IntervalEnum[IntervalEnum["FIFTEENMINUTES"] = 'FIFTEEN_MINUTES'] = "FIFTEENMINUTES";
        IntervalEnum[IntervalEnum["THIRTYMINUTES"] = 'THIRTY_MINUTES'] = "THIRTYMINUTES";
        IntervalEnum[IntervalEnum["ONEHOUR"] = 'ONE_HOUR'] = "ONEHOUR";
        IntervalEnum[IntervalEnum["FOURHOURS"] = 'FOUR_HOURS'] = "FOURHOURS";
        IntervalEnum[IntervalEnum["TWELVEHOURS"] = 'TWELVE_HOURS'] = "TWELVEHOURS";
        IntervalEnum[IntervalEnum["ONEDAY"] = 'ONE_DAY'] = "ONEDAY";
        IntervalEnum[IntervalEnum["ONEWEEK"] = 'ONE_WEEK'] = "ONEWEEK";
    })(IntervalEnum = PublicPerformanceResponse.IntervalEnum || (PublicPerformanceResponse.IntervalEnum = {}));
    let PeriodEnum;
    (function (PeriodEnum) {
        PeriodEnum[PeriodEnum["ONEMINUTE"] = 'ONE_MINUTE'] = "ONEMINUTE";
        PeriodEnum[PeriodEnum["FIVEMINUTES"] = 'FIVE_MINUTES'] = "FIVEMINUTES";
        PeriodEnum[PeriodEnum["TENMINUTES"] = 'TEN_MINUTES'] = "TENMINUTES";
        PeriodEnum[PeriodEnum["FIFTEENMINUTES"] = 'FIFTEEN_MINUTES'] = "FIFTEENMINUTES";
        PeriodEnum[PeriodEnum["THIRTYMINUTES"] = 'THIRTY_MINUTES'] = "THIRTYMINUTES";
        PeriodEnum[PeriodEnum["ONEHOUR"] = 'ONE_HOUR'] = "ONEHOUR";
        PeriodEnum[PeriodEnum["FOURHOURS"] = 'FOUR_HOURS'] = "FOURHOURS";
        PeriodEnum[PeriodEnum["TWELVEHOURS"] = 'TWELVE_HOURS'] = "TWELVEHOURS";
        PeriodEnum[PeriodEnum["ONEDAY"] = 'ONE_DAY'] = "ONEDAY";
        PeriodEnum[PeriodEnum["ONEWEEK"] = 'ONE_WEEK'] = "ONEWEEK";
    })(PeriodEnum = PublicPerformanceResponse.PeriodEnum || (PublicPerformanceResponse.PeriodEnum = {}));
})(PublicPerformanceResponse = exports.PublicPerformanceResponse || (exports.PublicPerformanceResponse = {}));
