// @hubspot/api-client/lib/codegen/files/model/fileActionResponse.js
"use strict";
/**
 * Files
 * Upload and manage files.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileActionResponse = void 0;
class FileActionResponse {
    static getAttributeTypeMap() {
        return FileActionResponse.attributeTypeMap;
    }
}
exports.FileActionResponse = FileActionResponse;
FileActionResponse.discriminator = undefined;
FileActionResponse.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "FileActionResponse.StatusEnum"
    },
    {
        "name": "result",
        "baseName": "result",
        "type": "any"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number"
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date"
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }"
    },
    {
        "name": "taskId",
        "baseName": "taskId",
        "type": "string"
    }
];
(function (FileActionResponse) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["PENDING"] = 'PENDING'] = "PENDING";
        StatusEnum[StatusEnum["PROCESSING"] = 'PROCESSING'] = "PROCESSING";
        StatusEnum[StatusEnum["CANCELED"] = 'CANCELED'] = "CANCELED";
        StatusEnum[StatusEnum["COMPLETE"] = 'COMPLETE'] = "COMPLETE";
    })(StatusEnum = FileActionResponse.StatusEnum || (FileActionResponse.StatusEnum = {}));
})(FileActionResponse = exports.FileActionResponse || (exports.FileActionResponse = {}));
