// @hubspot/api-client/lib/codegen/files/model/importFromUrlInput.js
"use strict";
/**
 * Files
 * Upload and manage files.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImportFromUrlInput = void 0;
class ImportFromUrlInput {
    static getAttributeTypeMap() {
        return ImportFromUrlInput.attributeTypeMap;
    }
}
exports.ImportFromUrlInput = ImportFromUrlInput;
ImportFromUrlInput.discriminator = undefined;
ImportFromUrlInput.attributeTypeMap = [
    {
        "name": "access",
        "baseName": "access",
        "type": "ImportFromUrlInput.AccessEnum"
    },
    {
        "name": "ttl",
        "baseName": "ttl",
        "type": "string"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string"
    },
    {
        "name": "folderId",
        "baseName": "folderId",
        "type": "string"
    },
    {
        "name": "folderPath",
        "baseName": "folderPath",
        "type": "string"
    },
    {
        "name": "duplicateValidationStrategy",
        "baseName": "duplicateValidationStrategy",
        "type": "ImportFromUrlInput.DuplicateValidationStrategyEnum"
    },
    {
        "name": "duplicateValidationScope",
        "baseName": "duplicateValidationScope",
        "type": "ImportFromUrlInput.DuplicateValidationScopeEnum"
    },
    {
        "name": "overwrite",
        "baseName": "overwrite",
        "type": "boolean"
    }
];
(function (ImportFromUrlInput) {
    let AccessEnum;
    (function (AccessEnum) {
        AccessEnum[AccessEnum["PUBLICINDEXABLE"] = 'PUBLIC_INDEXABLE'] = "PUBLICINDEXABLE";
        AccessEnum[AccessEnum["PUBLICNOTINDEXABLE"] = 'PUBLIC_NOT_INDEXABLE'] = "PUBLICNOTINDEXABLE";
        AccessEnum[AccessEnum["HIDDENINDEXABLE"] = 'HIDDEN_INDEXABLE'] = "HIDDENINDEXABLE";
        AccessEnum[AccessEnum["HIDDENNOTINDEXABLE"] = 'HIDDEN_NOT_INDEXABLE'] = "HIDDENNOTINDEXABLE";
        AccessEnum[AccessEnum["HIDDENPRIVATE"] = 'HIDDEN_PRIVATE'] = "HIDDENPRIVATE";
        AccessEnum[AccessEnum["PRIVATE"] = 'PRIVATE'] = "PRIVATE";
    })(AccessEnum = ImportFromUrlInput.AccessEnum || (ImportFromUrlInput.AccessEnum = {}));
    let DuplicateValidationStrategyEnum;
    (function (DuplicateValidationStrategyEnum) {
        DuplicateValidationStrategyEnum[DuplicateValidationStrategyEnum["NONE"] = 'NONE'] = "NONE";
        DuplicateValidationStrategyEnum[DuplicateValidationStrategyEnum["REJECT"] = 'REJECT'] = "REJECT";
        DuplicateValidationStrategyEnum[DuplicateValidationStrategyEnum["RETURNEXISTING"] = 'RETURN_EXISTING'] = "RETURNEXISTING";
    })(DuplicateValidationStrategyEnum = ImportFromUrlInput.DuplicateValidationStrategyEnum || (ImportFromUrlInput.DuplicateValidationStrategyEnum = {}));
    let DuplicateValidationScopeEnum;
    (function (DuplicateValidationScopeEnum) {
        DuplicateValidationScopeEnum[DuplicateValidationScopeEnum["ENTIREPORTAL"] = 'ENTIRE_PORTAL'] = "ENTIREPORTAL";
        DuplicateValidationScopeEnum[DuplicateValidationScopeEnum["EXACTFOLDER"] = 'EXACT_FOLDER'] = "EXACTFOLDER";
    })(DuplicateValidationScopeEnum = ImportFromUrlInput.DuplicateValidationScopeEnum || (ImportFromUrlInput.DuplicateValidationScopeEnum = {}));
})(ImportFromUrlInput = exports.ImportFromUrlInput || (exports.ImportFromUrlInput = {}));
