// @hubspot/api-client/lib/codegen/files/model/modelFile.js
"use strict";
/**
 * Files
 * Upload and manage files.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelFile = void 0;
/**
* File
*/
class ModelFile {
    static getAttributeTypeMap() {
        return ModelFile.attributeTypeMap;
    }
}
exports.ModelFile = ModelFile;
ModelFile.discriminator = undefined;
ModelFile.attributeTypeMap = [
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date"
    },
    {
        "name": "archivedAt",
        "baseName": "archivedAt",
        "type": "Date"
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "Date"
    },
    {
        "name": "archived",
        "baseName": "archived",
        "type": "boolean"
    },
    {
        "name": "parentFolderId",
        "baseName": "parentFolderId",
        "type": "string"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "path",
        "baseName": "path",
        "type": "string"
    },
    {
        "name": "size",
        "baseName": "size",
        "type": "number"
    },
    {
        "name": "height",
        "baseName": "height",
        "type": "number"
    },
    {
        "name": "width",
        "baseName": "width",
        "type": "number"
    },
    {
        "name": "encoding",
        "baseName": "encoding",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "string"
    },
    {
        "name": "extension",
        "baseName": "extension",
        "type": "string"
    },
    {
        "name": "defaultHostingUrl",
        "baseName": "defaultHostingUrl",
        "type": "string"
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string"
    },
    {
        "name": "isUsableInContent",
        "baseName": "isUsableInContent",
        "type": "boolean"
    },
    {
        "name": "access",
        "baseName": "access",
        "type": "ModelFile.AccessEnum"
    }
];
(function (ModelFile) {
    let AccessEnum;
    (function (AccessEnum) {
        AccessEnum[AccessEnum["PUBLICINDEXABLE"] = 'PUBLIC_INDEXABLE'] = "PUBLICINDEXABLE";
        AccessEnum[AccessEnum["PUBLICNOTINDEXABLE"] = 'PUBLIC_NOT_INDEXABLE'] = "PUBLICNOTINDEXABLE";
        AccessEnum[AccessEnum["HIDDENINDEXABLE"] = 'HIDDEN_INDEXABLE'] = "HIDDENINDEXABLE";
        AccessEnum[AccessEnum["HIDDENNOTINDEXABLE"] = 'HIDDEN_NOT_INDEXABLE'] = "HIDDENNOTINDEXABLE";
        AccessEnum[AccessEnum["HIDDENPRIVATE"] = 'HIDDEN_PRIVATE'] = "HIDDENPRIVATE";
        AccessEnum[AccessEnum["PRIVATE"] = 'PRIVATE'] = "PRIVATE";
    })(AccessEnum = ModelFile.AccessEnum || (ModelFile.AccessEnum = {}));
})(ModelFile = exports.ModelFile || (exports.ModelFile = {}));
