// @hubspot/api-client/lib/codegen/files/model/models.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoidAuth = exports.OAuth = exports.ApiKeyAuth = exports.HttpBearerAuth = exports.HttpBasicAuth = exports.ObjectSerializer = void 0;
__exportStar(require("./collectionResponseFile"), exports);
__exportStar(require("./collectionResponseFolder"), exports);
__exportStar(require("./errorCategory"), exports);
__exportStar(require("./errorDetail"), exports);
__exportStar(require("./fileActionResponse"), exports);
__exportStar(require("./fileUpdateInput"), exports);
__exportStar(require("./folder"), exports);
__exportStar(require("./folderActionResponse"), exports);
__exportStar(require("./folderInput"), exports);
__exportStar(require("./folderUpdateInput"), exports);
__exportStar(require("./folderUpdateTaskLocator"), exports);
__exportStar(require("./importFromUrlInput"), exports);
__exportStar(require("./importFromUrlTaskLocator"), exports);
__exportStar(require("./modelError"), exports);
__exportStar(require("./modelFile"), exports);
__exportStar(require("./nextPage"), exports);
__exportStar(require("./paging"), exports);
__exportStar(require("./previousPage"), exports);
__exportStar(require("./signedUrl"), exports);
__exportStar(require("./standardError"), exports);
const collectionResponseFile_1 = require("./collectionResponseFile");
const collectionResponseFolder_1 = require("./collectionResponseFolder");
const errorCategory_1 = require("./errorCategory");
const errorDetail_1 = require("./errorDetail");
const fileActionResponse_1 = require("./fileActionResponse");
const fileUpdateInput_1 = require("./fileUpdateInput");
const folder_1 = require("./folder");
const folderActionResponse_1 = require("./folderActionResponse");
const folderInput_1 = require("./folderInput");
const folderUpdateInput_1 = require("./folderUpdateInput");
const folderUpdateTaskLocator_1 = require("./folderUpdateTaskLocator");
const importFromUrlInput_1 = require("./importFromUrlInput");
const importFromUrlTaskLocator_1 = require("./importFromUrlTaskLocator");
const modelError_1 = require("./modelError");
const modelFile_1 = require("./modelFile");
const nextPage_1 = require("./nextPage");
const paging_1 = require("./paging");
const previousPage_1 = require("./previousPage");
const signedUrl_1 = require("./signedUrl");
const standardError_1 = require("./standardError");
/* tslint:disable:no-unused-variable */
let primitives = [
    "string",
    "boolean",
    "double",
    "integer",
    "long",
    "float",
    "number",
    "any"
];
let enumsMap = {
    "ErrorCategory.HttpStatusEnum": errorCategory_1.ErrorCategory.HttpStatusEnum,
    "FileActionResponse.StatusEnum": fileActionResponse_1.FileActionResponse.StatusEnum,
    "FileUpdateInput.AccessEnum": fileUpdateInput_1.FileUpdateInput.AccessEnum,
    "FolderActionResponse.StatusEnum": folderActionResponse_1.FolderActionResponse.StatusEnum,
    "ImportFromUrlInput.AccessEnum": importFromUrlInput_1.ImportFromUrlInput.AccessEnum,
    "ImportFromUrlInput.DuplicateValidationStrategyEnum": importFromUrlInput_1.ImportFromUrlInput.DuplicateValidationStrategyEnum,
    "ImportFromUrlInput.DuplicateValidationScopeEnum": importFromUrlInput_1.ImportFromUrlInput.DuplicateValidationScopeEnum,
    "ModelFile.AccessEnum": modelFile_1.ModelFile.AccessEnum,
};
let typeMap = {
    "CollectionResponseFile": collectionResponseFile_1.CollectionResponseFile,
    "CollectionResponseFolder": collectionResponseFolder_1.CollectionResponseFolder,
    "ErrorCategory": errorCategory_1.ErrorCategory,
    "ErrorDetail": errorDetail_1.ErrorDetail,
    "FileActionResponse": fileActionResponse_1.FileActionResponse,
    "FileUpdateInput": fileUpdateInput_1.FileUpdateInput,
    "Folder": folder_1.Folder,
    "FolderActionResponse": folderActionResponse_1.FolderActionResponse,
    "FolderInput": folderInput_1.FolderInput,
    "FolderUpdateInput": folderUpdateInput_1.FolderUpdateInput,
    "FolderUpdateTaskLocator": folderUpdateTaskLocator_1.FolderUpdateTaskLocator,
    "ImportFromUrlInput": importFromUrlInput_1.ImportFromUrlInput,
    "ImportFromUrlTaskLocator": importFromUrlTaskLocator_1.ImportFromUrlTaskLocator,
    "ModelError": modelError_1.ModelError,
    "ModelFile": modelFile_1.ModelFile,
    "NextPage": nextPage_1.NextPage,
    "Paging": paging_1.Paging,
    "PreviousPage": previousPage_1.PreviousPage,
    "SignedUrl": signedUrl_1.SignedUrl,
    "StandardError": standardError_1.StandardError,
};
class ObjectSerializer {
    static findCorrectType(data, expectedType) {
        if (data == undefined) {
            return expectedType;
        }
        else if (primitives.indexOf(expectedType.toLowerCase()) !== -1) {
            return expectedType;
        }
        else if (expectedType === "Date") {
            return expectedType;
        }
        else {
            if (enumsMap[expectedType]) {
                return expectedType;
            }
            if (!typeMap[expectedType]) {
                return expectedType; // w/e we don't know the type
            }
            // Check the discriminator
            let discriminatorProperty = typeMap[expectedType].discriminator;
            if (discriminatorProperty == null) {
                return expectedType; // the type does not have a discriminator. use it.
            }
            else {
                if (data[discriminatorProperty]) {
                    var discriminatorType = data[discriminatorProperty];
                    if (typeMap[discriminatorType]) {
                        return discriminatorType; // use the type given in the discriminator
                    }
                    else {
                        return expectedType; // discriminator did not map to a type
                    }
                }
                else {
                    return expectedType; // discriminator was not present (or an empty string)
                }
            }
        }
    }
    static serialize(data, type) {
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index in data) {
                let date = data[index];
                transformedData.push(ObjectSerializer.serialize(date, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return data.toISOString();
        }
        else {
            if (enumsMap[type]) {
                return data;
            }
            if (!typeMap[type]) { // in case we dont know the type
                return data;
            }
            // Get the actual type of this object
            type = this.findCorrectType(data, type);
            // get the map for the correct type.
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            let instance = {};
            for (let index in attributeTypes) {
                let attributeType = attributeTypes[index];
                instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type);
            }
            return instance;
        }
    }
    static deserialize(data, type) {
        // polymorphism may change the actual type.
        type = ObjectSerializer.findCorrectType(data, type);
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index in data) {
                let date = data[index];
                transformedData.push(ObjectSerializer.deserialize(date, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return new Date(data);
        }
        else {
            if (enumsMap[type]) { // is Enum
                return data;
            }
            if (!typeMap[type]) { // dont know the type
                return data;
            }
            let instance = new typeMap[type]();
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            for (let index in attributeTypes) {
                let attributeType = attributeTypes[index];
                instance[attributeType.name] = ObjectSerializer.deserialize(data[attributeType.baseName], attributeType.type);
            }
            return instance;
        }
    }
}
exports.ObjectSerializer = ObjectSerializer;
class HttpBasicAuth {
    constructor() {
        this.username = '';
        this.password = '';
    }
    applyToRequest(requestOptions) {
        requestOptions.auth = {
            username: this.username, password: this.password
        };
    }
}
exports.HttpBasicAuth = HttpBasicAuth;
class HttpBearerAuth {
    constructor() {
        this.accessToken = '';
    }
    applyToRequest(requestOptions) {
        if (requestOptions && requestOptions.headers) {
            const accessToken = typeof this.accessToken === 'function'
                ? this.accessToken()
                : this.accessToken;
            requestOptions.headers["Authorization"] = "Bearer " + accessToken;
        }
    }
}
exports.HttpBearerAuth = HttpBearerAuth;
class ApiKeyAuth {
    constructor(location, paramName) {
        this.location = location;
        this.paramName = paramName;
        this.apiKey = '';
    }
    applyToRequest(requestOptions) {
        if (this.location == "query") {
            requestOptions.qs[this.paramName] = this.apiKey;
        }
        else if (this.location == "header" && requestOptions && requestOptions.headers) {
            requestOptions.headers[this.paramName] = this.apiKey;
        }
        else if (this.location == 'cookie' && requestOptions && requestOptions.headers) {
            if (requestOptions.headers['Cookie']) {
                requestOptions.headers['Cookie'] += '; ' + this.paramName + '=' + encodeURIComponent(this.apiKey);
            }
            else {
                requestOptions.headers['Cookie'] = this.paramName + '=' + encodeURIComponent(this.apiKey);
            }
        }
    }
}
exports.ApiKeyAuth = ApiKeyAuth;
class OAuth {
    constructor() {
        this.accessToken = '';
    }
    applyToRequest(requestOptions) {
        if (requestOptions && requestOptions.headers) {
            requestOptions.headers["Authorization"] = "Bearer " + this.accessToken;
        }
    }
}
exports.OAuth = OAuth;
class VoidAuth {
    constructor() {
        this.username = '';
        this.password = '';
    }
    applyToRequest(_) {
        // Do nothing
    }
}
exports.VoidAuth = VoidAuth;
