// @hubspot/api-client/lib/codegen/cms/hubdb/api/apis.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.APIS = exports.HttpError = void 0;
__exportStar(require("./rowsApi"), exports);
const rowsApi_1 = require("./rowsApi");
__exportStar(require("./rowsBatchApi"), exports);
const rowsBatchApi_1 = require("./rowsBatchApi");
__exportStar(require("./tablesApi"), exports);
const tablesApi_1 = require("./tablesApi");
class HttpError extends Error {
    constructor(response, body, statusCode) {
        super('HTTP request failed');
        this.response = response;
        this.body = body;
        this.statusCode = statusCode;
        this.name = 'HttpError';
    }
}
exports.HttpError = HttpError;
exports.APIS = [rowsApi_1.RowsApi, rowsBatchApi_1.RowsBatchApi, tablesApi_1.TablesApi];
