// @hubspot/api-client/lib/codegen/cms/audit_logs/model/publicAuditLog.js
"use strict";
/**
 * CMS Audit Logs
 * Use this endpoint to query audit logs of CMS changes that occurred on your HubSpot account.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicAuditLog = void 0;
class PublicAuditLog {
    static getAttributeTypeMap() {
        return PublicAuditLog.attributeTypeMap;
    }
}
exports.PublicAuditLog = PublicAuditLog;
PublicAuditLog.discriminator = undefined;
PublicAuditLog.attributeTypeMap = [
    {
        "name": "objectId",
        "baseName": "objectId",
        "type": "string"
    },
    {
        "name": "userId",
        "baseName": "userId",
        "type": "string"
    },
    {
        "name": "timestamp",
        "baseName": "timestamp",
        "type": "Date"
    },
    {
        "name": "objectName",
        "baseName": "objectName",
        "type": "string"
    },
    {
        "name": "fullName",
        "baseName": "fullName",
        "type": "string"
    },
    {
        "name": "event",
        "baseName": "event",
        "type": "PublicAuditLog.EventEnum"
    },
    {
        "name": "objectType",
        "baseName": "objectType",
        "type": "PublicAuditLog.ObjectTypeEnum"
    }
];
(function (PublicAuditLog) {
    let EventEnum;
    (function (EventEnum) {
        EventEnum[EventEnum["CREATED"] = 'CREATED'] = "CREATED";
        EventEnum[EventEnum["UPDATED"] = 'UPDATED'] = "UPDATED";
        EventEnum[EventEnum["PUBLISHED"] = 'PUBLISHED'] = "PUBLISHED";
        EventEnum[EventEnum["DELETED"] = 'DELETED'] = "DELETED";
        EventEnum[EventEnum["UNPUBLISHED"] = 'UNPUBLISHED'] = "UNPUBLISHED";
    })(EventEnum = PublicAuditLog.EventEnum || (PublicAuditLog.EventEnum = {}));
    let ObjectTypeEnum;
    (function (ObjectTypeEnum) {
        ObjectTypeEnum[ObjectTypeEnum["BLOG"] = 'BLOG'] = "BLOG";
        ObjectTypeEnum[ObjectTypeEnum["BLOGPOST"] = 'BLOG_POST'] = "BLOGPOST";
        ObjectTypeEnum[ObjectTypeEnum["LANDINGPAGE"] = 'LANDING_PAGE'] = "LANDINGPAGE";
        ObjectTypeEnum[ObjectTypeEnum["WEBSITEPAGE"] = 'WEBSITE_PAGE'] = "WEBSITEPAGE";
        ObjectTypeEnum[ObjectTypeEnum["TEMPLATE"] = 'TEMPLATE'] = "TEMPLATE";
        ObjectTypeEnum[ObjectTypeEnum["MODULE"] = 'MODULE'] = "MODULE";
        ObjectTypeEnum[ObjectTypeEnum["GLOBALMODULE"] = 'GLOBAL_MODULE'] = "GLOBALMODULE";
        ObjectTypeEnum[ObjectTypeEnum["SERVERLESSFUNCTION"] = 'SERVERLESS_FUNCTION'] = "SERVERLESSFUNCTION";
        ObjectTypeEnum[ObjectTypeEnum["DOMAIN"] = 'DOMAIN'] = "DOMAIN";
        ObjectTypeEnum[ObjectTypeEnum["URLMAPPING"] = 'URL_MAPPING'] = "URLMAPPING";
        ObjectTypeEnum[ObjectTypeEnum["EMAIL"] = 'EMAIL'] = "EMAIL";
        ObjectTypeEnum[ObjectTypeEnum["CONTENTSETTINGS"] = 'CONTENT_SETTINGS'] = "CONTENTSETTINGS";
        ObjectTypeEnum[ObjectTypeEnum["HUBDBTABLE"] = 'HUBDB_TABLE'] = "HUBDBTABLE";
        ObjectTypeEnum[ObjectTypeEnum["KNOWLEDGEBASEARTICLE"] = 'KNOWLEDGE_BASE_ARTICLE'] = "KNOWLEDGEBASEARTICLE";
        ObjectTypeEnum[ObjectTypeEnum["KNOWLEDGEBASE"] = 'KNOWLEDGE_BASE'] = "KNOWLEDGEBASE";
        ObjectTypeEnum[ObjectTypeEnum["THEME"] = 'THEME'] = "THEME";
        ObjectTypeEnum[ObjectTypeEnum["CSS"] = 'CSS'] = "CSS";
        ObjectTypeEnum[ObjectTypeEnum["JS"] = 'JS'] = "JS";
    })(ObjectTypeEnum = PublicAuditLog.ObjectTypeEnum || (PublicAuditLog.ObjectTypeEnum = {}));
})(PublicAuditLog = exports.PublicAuditLog || (exports.PublicAuditLog = {}));
