// @hubspot/api-client/lib/codegen/crm/associations/model/batchResponsePublicAssociationMulti.js
"use strict";
/**
 * Associations
 * Associations define the relationships between objects in HubSpot. These endpoints allow you to create, read, and remove associations.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponsePublicAssociationMulti = void 0;
class BatchResponsePublicAssociationMulti {
    static getAttributeTypeMap() {
        return BatchResponsePublicAssociationMulti.attributeTypeMap;
    }
}
exports.BatchResponsePublicAssociationMulti = BatchResponsePublicAssociationMulti;
BatchResponsePublicAssociationMulti.discriminator = undefined;
BatchResponsePublicAssociationMulti.attributeTypeMap = [
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<PublicAssociationMulti>"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number"
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponsePublicAssociationMulti.StatusEnum"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date"
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date"
    }
];
(function (BatchResponsePublicAssociationMulti) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["PENDING"] = 'PENDING'] = "PENDING";
        StatusEnum[StatusEnum["PROCESSING"] = 'PROCESSING'] = "PROCESSING";
        StatusEnum[StatusEnum["CANCELED"] = 'CANCELED'] = "CANCELED";
        StatusEnum[StatusEnum["COMPLETE"] = 'COMPLETE'] = "COMPLETE";
    })(StatusEnum = BatchResponsePublicAssociationMulti.StatusEnum || (BatchResponsePublicAssociationMulti.StatusEnum = {}));
})(BatchResponsePublicAssociationMulti = exports.BatchResponsePublicAssociationMulti || (exports.BatchResponsePublicAssociationMulti = {}));
