// @hubspot/api-client/lib/codegen/crm/timeline/model/models.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoidAuth = exports.OAuth = exports.ApiKeyAuth = exports.HttpBearerAuth = exports.HttpBasicAuth = exports.ObjectSerializer = void 0;
__exportStar(require("./batchInputTimelineEvent"), exports);
__exportStar(require("./batchResponseTimelineEventResponse"), exports);
__exportStar(require("./batchResponseTimelineEventResponseWithErrors"), exports);
__exportStar(require("./collectionResponseTimelineEventTemplate"), exports);
__exportStar(require("./errorCategory"), exports);
__exportStar(require("./errorDetail"), exports);
__exportStar(require("./eventDetail"), exports);
__exportStar(require("./modelError"), exports);
__exportStar(require("./nextPage"), exports);
__exportStar(require("./paging"), exports);
__exportStar(require("./previousPage"), exports);
__exportStar(require("./standardError"), exports);
__exportStar(require("./timelineEvent"), exports);
__exportStar(require("./timelineEventIFrame"), exports);
__exportStar(require("./timelineEventResponse"), exports);
__exportStar(require("./timelineEventTemplate"), exports);
__exportStar(require("./timelineEventTemplateCreateRequest"), exports);
__exportStar(require("./timelineEventTemplateToken"), exports);
__exportStar(require("./timelineEventTemplateTokenOption"), exports);
__exportStar(require("./timelineEventTemplateTokenUpdateRequest"), exports);
__exportStar(require("./timelineEventTemplateUpdateRequest"), exports);
const batchInputTimelineEvent_1 = require("./batchInputTimelineEvent");
const batchResponseTimelineEventResponse_1 = require("./batchResponseTimelineEventResponse");
const batchResponseTimelineEventResponseWithErrors_1 = require("./batchResponseTimelineEventResponseWithErrors");
const collectionResponseTimelineEventTemplate_1 = require("./collectionResponseTimelineEventTemplate");
const errorCategory_1 = require("./errorCategory");
const errorDetail_1 = require("./errorDetail");
const eventDetail_1 = require("./eventDetail");
const modelError_1 = require("./modelError");
const nextPage_1 = require("./nextPage");
const paging_1 = require("./paging");
const previousPage_1 = require("./previousPage");
const standardError_1 = require("./standardError");
const timelineEvent_1 = require("./timelineEvent");
const timelineEventIFrame_1 = require("./timelineEventIFrame");
const timelineEventResponse_1 = require("./timelineEventResponse");
const timelineEventTemplate_1 = require("./timelineEventTemplate");
const timelineEventTemplateCreateRequest_1 = require("./timelineEventTemplateCreateRequest");
const timelineEventTemplateToken_1 = require("./timelineEventTemplateToken");
const timelineEventTemplateTokenOption_1 = require("./timelineEventTemplateTokenOption");
const timelineEventTemplateTokenUpdateRequest_1 = require("./timelineEventTemplateTokenUpdateRequest");
const timelineEventTemplateUpdateRequest_1 = require("./timelineEventTemplateUpdateRequest");
/* tslint:disable:no-unused-variable */
let primitives = [
    "string",
    "boolean",
    "double",
    "integer",
    "long",
    "float",
    "number",
    "any"
];
let enumsMap = {
    "BatchResponseTimelineEventResponse.StatusEnum": batchResponseTimelineEventResponse_1.BatchResponseTimelineEventResponse.StatusEnum,
    "BatchResponseTimelineEventResponseWithErrors.StatusEnum": batchResponseTimelineEventResponseWithErrors_1.BatchResponseTimelineEventResponseWithErrors.StatusEnum,
    "ErrorCategory.HttpStatusEnum": errorCategory_1.ErrorCategory.HttpStatusEnum,
    "TimelineEventTemplateToken.TypeEnum": timelineEventTemplateToken_1.TimelineEventTemplateToken.TypeEnum,
};
let typeMap = {
    "BatchInputTimelineEvent": batchInputTimelineEvent_1.BatchInputTimelineEvent,
    "BatchResponseTimelineEventResponse": batchResponseTimelineEventResponse_1.BatchResponseTimelineEventResponse,
    "BatchResponseTimelineEventResponseWithErrors": batchResponseTimelineEventResponseWithErrors_1.BatchResponseTimelineEventResponseWithErrors,
    "CollectionResponseTimelineEventTemplate": collectionResponseTimelineEventTemplate_1.CollectionResponseTimelineEventTemplate,
    "ErrorCategory": errorCategory_1.ErrorCategory,
    "ErrorDetail": errorDetail_1.ErrorDetail,
    "EventDetail": eventDetail_1.EventDetail,
    "ModelError": modelError_1.ModelError,
    "NextPage": nextPage_1.NextPage,
    "Paging": paging_1.Paging,
    "PreviousPage": previousPage_1.PreviousPage,
    "StandardError": standardError_1.StandardError,
    "TimelineEvent": timelineEvent_1.TimelineEvent,
    "TimelineEventIFrame": timelineEventIFrame_1.TimelineEventIFrame,
    "TimelineEventResponse": timelineEventResponse_1.TimelineEventResponse,
    "TimelineEventTemplate": timelineEventTemplate_1.TimelineEventTemplate,
    "TimelineEventTemplateCreateRequest": timelineEventTemplateCreateRequest_1.TimelineEventTemplateCreateRequest,
    "TimelineEventTemplateToken": timelineEventTemplateToken_1.TimelineEventTemplateToken,
    "TimelineEventTemplateTokenOption": timelineEventTemplateTokenOption_1.TimelineEventTemplateTokenOption,
    "TimelineEventTemplateTokenUpdateRequest": timelineEventTemplateTokenUpdateRequest_1.TimelineEventTemplateTokenUpdateRequest,
    "TimelineEventTemplateUpdateRequest": timelineEventTemplateUpdateRequest_1.TimelineEventTemplateUpdateRequest,
};
class ObjectSerializer {
    static findCorrectType(data, expectedType) {
        if (data == undefined) {
            return expectedType;
        }
        else if (primitives.indexOf(expectedType.toLowerCase()) !== -1) {
            return expectedType;
        }
        else if (expectedType === "Date") {
            return expectedType;
        }
        else {
            if (enumsMap[expectedType]) {
                return expectedType;
            }
            if (!typeMap[expectedType]) {
                return expectedType; // w/e we don't know the type
            }
            // Check the discriminator
            let discriminatorProperty = typeMap[expectedType].discriminator;
            if (discriminatorProperty == null) {
                return expectedType; // the type does not have a discriminator. use it.
            }
            else {
                if (data[discriminatorProperty]) {
                    var discriminatorType = data[discriminatorProperty];
                    if (typeMap[discriminatorType]) {
                        return discriminatorType; // use the type given in the discriminator
                    }
                    else {
                        return expectedType; // discriminator did not map to a type
                    }
                }
                else {
                    return expectedType; // discriminator was not present (or an empty string)
                }
            }
        }
    }
    static serialize(data, type) {
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index in data) {
                let date = data[index];
                transformedData.push(ObjectSerializer.serialize(date, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return data.toISOString();
        }
        else {
            if (enumsMap[type]) {
                return data;
            }
            if (!typeMap[type]) { // in case we dont know the type
                return data;
            }
            // Get the actual type of this object
            type = this.findCorrectType(data, type);
            // get the map for the correct type.
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            let instance = {};
            for (let index in attributeTypes) {
                let attributeType = attributeTypes[index];
                instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type);
            }
            return instance;
        }
    }
    static deserialize(data, type) {
        // polymorphism may change the actual type.
        type = ObjectSerializer.findCorrectType(data, type);
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index in data) {
                let date = data[index];
                transformedData.push(ObjectSerializer.deserialize(date, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return new Date(data);
        }
        else {
            if (enumsMap[type]) { // is Enum
                return data;
            }
            if (!typeMap[type]) { // dont know the type
                return data;
            }
            let instance = new typeMap[type]();
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            for (let index in attributeTypes) {
                let attributeType = attributeTypes[index];
                instance[attributeType.name] = ObjectSerializer.deserialize(data[attributeType.baseName], attributeType.type);
            }
            return instance;
        }
    }
}
exports.ObjectSerializer = ObjectSerializer;
class HttpBasicAuth {
    constructor() {
        this.username = '';
        this.password = '';
    }
    applyToRequest(requestOptions) {
        requestOptions.auth = {
            username: this.username, password: this.password
        };
    }
}
exports.HttpBasicAuth = HttpBasicAuth;
class HttpBearerAuth {
    constructor() {
        this.accessToken = '';
    }
    applyToRequest(requestOptions) {
        if (requestOptions && requestOptions.headers) {
            const accessToken = typeof this.accessToken === 'function'
                ? this.accessToken()
                : this.accessToken;
            requestOptions.headers["Authorization"] = "Bearer " + accessToken;
        }
    }
}
exports.HttpBearerAuth = HttpBearerAuth;
class ApiKeyAuth {
    constructor(location, paramName) {
        this.location = location;
        this.paramName = paramName;
        this.apiKey = '';
    }
    applyToRequest(requestOptions) {
        if (this.location == "query") {
            requestOptions.qs[this.paramName] = this.apiKey;
        }
        else if (this.location == "header" && requestOptions && requestOptions.headers) {
            requestOptions.headers[this.paramName] = this.apiKey;
        }
        else if (this.location == 'cookie' && requestOptions && requestOptions.headers) {
            if (requestOptions.headers['Cookie']) {
                requestOptions.headers['Cookie'] += '; ' + this.paramName + '=' + encodeURIComponent(this.apiKey);
            }
            else {
                requestOptions.headers['Cookie'] = this.paramName + '=' + encodeURIComponent(this.apiKey);
            }
        }
    }
}
exports.ApiKeyAuth = ApiKeyAuth;
class OAuth {
    constructor() {
        this.accessToken = '';
    }
    applyToRequest(requestOptions) {
        if (requestOptions && requestOptions.headers) {
            requestOptions.headers["Authorization"] = "Bearer " + this.accessToken;
        }
    }
}
exports.OAuth = OAuth;
class VoidAuth {
    constructor() {
        this.username = '';
        this.password = '';
    }
    applyToRequest(_) {
        // Do nothing
    }
}
exports.VoidAuth = VoidAuth;
