// @hubspot/api-client/lib/codegen/crm/timeline/model/timelineEventTemplateToken.js
"use strict";
/**
 * Timeline events
 * This feature allows an app to create and configure custom events that can show up in the timelines of certain CRM objects like contacts, companies, tickets, or deals. You\'ll find multiple use cases for this API in the sections below.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimelineEventTemplateToken = void 0;
/**
* State of the token definition.
*/
class TimelineEventTemplateToken {
    static getAttributeTypeMap() {
        return TimelineEventTemplateToken.attributeTypeMap;
    }
}
exports.TimelineEventTemplateToken = TimelineEventTemplateToken;
TimelineEventTemplateToken.discriminator = undefined;
TimelineEventTemplateToken.attributeTypeMap = [
    {
        "name": "label",
        "baseName": "label",
        "type": "string"
    },
    {
        "name": "objectPropertyName",
        "baseName": "objectPropertyName",
        "type": "string"
    },
    {
        "name": "options",
        "baseName": "options",
        "type": "Array<TimelineEventTemplateTokenOption>"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TimelineEventTemplateToken.TypeEnum"
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date"
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "Date"
    }
];
(function (TimelineEventTemplateToken) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Date"] = 'date'] = "Date";
        TypeEnum[TypeEnum["Enumeration"] = 'enumeration'] = "Enumeration";
        TypeEnum[TypeEnum["Number"] = 'number'] = "Number";
        TypeEnum[TypeEnum["String"] = 'string'] = "String";
    })(TypeEnum = TimelineEventTemplateToken.TypeEnum || (TimelineEventTemplateToken.TypeEnum = {}));
})(TimelineEventTemplateToken = exports.TimelineEventTemplateToken || (exports.TimelineEventTemplateToken = {}));
