// @hubspot/api-client/lib/codegen/crm/quotes/model/filter.js
"use strict";
/**
 * Quotes
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Filter = void 0;
class Filter {
    static getAttributeTypeMap() {
        return Filter.attributeTypeMap;
    }
}
exports.Filter = Filter;
Filter.discriminator = undefined;
Filter.attributeTypeMap = [
    {
        "name": "value",
        "baseName": "value",
        "type": "string"
    },
    {
        "name": "propertyName",
        "baseName": "propertyName",
        "type": "string"
    },
    {
        "name": "operator",
        "baseName": "operator",
        "type": "Filter.OperatorEnum"
    }
];
(function (Filter) {
    let OperatorEnum;
    (function (OperatorEnum) {
        OperatorEnum[OperatorEnum["EQ"] = 'EQ'] = "EQ";
        OperatorEnum[OperatorEnum["NEQ"] = 'NEQ'] = "NEQ";
        OperatorEnum[OperatorEnum["LT"] = 'LT'] = "LT";
        OperatorEnum[OperatorEnum["LTE"] = 'LTE'] = "LTE";
        OperatorEnum[OperatorEnum["GT"] = 'GT'] = "GT";
        OperatorEnum[OperatorEnum["GTE"] = 'GTE'] = "GTE";
        OperatorEnum[OperatorEnum["HASPROPERTY"] = 'HAS_PROPERTY'] = "HASPROPERTY";
        OperatorEnum[OperatorEnum["NOTHASPROPERTY"] = 'NOT_HAS_PROPERTY'] = "NOTHASPROPERTY";
        OperatorEnum[OperatorEnum["CONTAINSTOKEN"] = 'CONTAINS_TOKEN'] = "CONTAINSTOKEN";
        OperatorEnum[OperatorEnum["NOTCONTAINSTOKEN"] = 'NOT_CONTAINS_TOKEN'] = "NOTCONTAINSTOKEN";
    })(OperatorEnum = Filter.OperatorEnum || (Filter.OperatorEnum = {}));
})(Filter = exports.Filter || (exports.Filter = {}));
