// @hubspot/api-client/lib/codegen/communication_preferences/model/publicSubscriptionStatus.js
"use strict";
/**
 * Subscriptions
 * Subscriptions allow contacts to control what forms of communications they receive. Contacts can decide whether they want to receive communication pertaining to a specific topic, brand, or an entire HubSpot account.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicSubscriptionStatus = void 0;
/**
* The status of a subscription for a contact.
*/
class PublicSubscriptionStatus {
    static getAttributeTypeMap() {
        return PublicSubscriptionStatus.attributeTypeMap;
    }
}
exports.PublicSubscriptionStatus = PublicSubscriptionStatus;
PublicSubscriptionStatus.discriminator = undefined;
PublicSubscriptionStatus.attributeTypeMap = [
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "PublicSubscriptionStatus.StatusEnum"
    },
    {
        "name": "sourceOfStatus",
        "baseName": "sourceOfStatus",
        "type": "PublicSubscriptionStatus.SourceOfStatusEnum"
    },
    {
        "name": "brandId",
        "baseName": "brandId",
        "type": "number"
    },
    {
        "name": "preferenceGroupName",
        "baseName": "preferenceGroupName",
        "type": "string"
    },
    {
        "name": "legalBasis",
        "baseName": "legalBasis",
        "type": "PublicSubscriptionStatus.LegalBasisEnum"
    },
    {
        "name": "legalBasisExplanation",
        "baseName": "legalBasisExplanation",
        "type": "string"
    }
];
(function (PublicSubscriptionStatus) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["SUBSCRIBED"] = 'SUBSCRIBED'] = "SUBSCRIBED";
        StatusEnum[StatusEnum["NOTSUBSCRIBED"] = 'NOT_SUBSCRIBED'] = "NOTSUBSCRIBED";
    })(StatusEnum = PublicSubscriptionStatus.StatusEnum || (PublicSubscriptionStatus.StatusEnum = {}));
    let SourceOfStatusEnum;
    (function (SourceOfStatusEnum) {
        SourceOfStatusEnum[SourceOfStatusEnum["PORTALWIDESTATUS"] = 'PORTAL_WIDE_STATUS'] = "PORTALWIDESTATUS";
        SourceOfStatusEnum[SourceOfStatusEnum["BRANDWIDESTATUS"] = 'BRAND_WIDE_STATUS'] = "BRANDWIDESTATUS";
        SourceOfStatusEnum[SourceOfStatusEnum["SUBSCRIPTIONSTATUS"] = 'SUBSCRIPTION_STATUS'] = "SUBSCRIPTIONSTATUS";
    })(SourceOfStatusEnum = PublicSubscriptionStatus.SourceOfStatusEnum || (PublicSubscriptionStatus.SourceOfStatusEnum = {}));
    let LegalBasisEnum;
    (function (LegalBasisEnum) {
        LegalBasisEnum[LegalBasisEnum["LEGITIMATEINTERESTPQL"] = 'LEGITIMATE_INTEREST_PQL'] = "LEGITIMATEINTERESTPQL";
        LegalBasisEnum[LegalBasisEnum["LEGITIMATEINTERESTCLIENT"] = 'LEGITIMATE_INTEREST_CLIENT'] = "LEGITIMATEINTERESTCLIENT";
        LegalBasisEnum[LegalBasisEnum["PERFORMANCEOFCONTRACT"] = 'PERFORMANCE_OF_CONTRACT'] = "PERFORMANCEOFCONTRACT";
        LegalBasisEnum[LegalBasisEnum["CONSENTWITHNOTICE"] = 'CONSENT_WITH_NOTICE'] = "CONSENTWITHNOTICE";
        LegalBasisEnum[LegalBasisEnum["NONGDPR"] = 'NON_GDPR'] = "NONGDPR";
        LegalBasisEnum[LegalBasisEnum["PROCESSANDSTORE"] = 'PROCESS_AND_STORE'] = "PROCESSANDSTORE";
        LegalBasisEnum[LegalBasisEnum["LEGITIMATEINTERESTOTHER"] = 'LEGITIMATE_INTEREST_OTHER'] = "LEGITIMATEINTERESTOTHER";
    })(LegalBasisEnum = PublicSubscriptionStatus.LegalBasisEnum || (PublicSubscriptionStatus.LegalBasisEnum = {}));
})(PublicSubscriptionStatus = exports.PublicSubscriptionStatus || (exports.PublicSubscriptionStatus = {}));
