// @hubspot/api-client/lib/codegen/communication_preferences/model/publicUpdateSubscriptionStatusRequest.js
"use strict";
/**
 * Subscriptions
 * Subscriptions allow contacts to control what forms of communications they receive. Contacts can decide whether they want to receive communication pertaining to a specific topic, brand, or an entire HubSpot account.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicUpdateSubscriptionStatusRequest = void 0;
/**
* A request to change the status of a contact\'s subscription.
*/
class PublicUpdateSubscriptionStatusRequest {
    static getAttributeTypeMap() {
        return PublicUpdateSubscriptionStatusRequest.attributeTypeMap;
    }
}
exports.PublicUpdateSubscriptionStatusRequest = PublicUpdateSubscriptionStatusRequest;
PublicUpdateSubscriptionStatusRequest.discriminator = undefined;
PublicUpdateSubscriptionStatusRequest.attributeTypeMap = [
    {
        "name": "emailAddress",
        "baseName": "emailAddress",
        "type": "string"
    },
    {
        "name": "subscriptionId",
        "baseName": "subscriptionId",
        "type": "string"
    },
    {
        "name": "legalBasis",
        "baseName": "legalBasis",
        "type": "PublicUpdateSubscriptionStatusRequest.LegalBasisEnum"
    },
    {
        "name": "legalBasisExplanation",
        "baseName": "legalBasisExplanation",
        "type": "string"
    }
];
(function (PublicUpdateSubscriptionStatusRequest) {
    let LegalBasisEnum;
    (function (LegalBasisEnum) {
        LegalBasisEnum[LegalBasisEnum["LEGITIMATEINTERESTPQL"] = 'LEGITIMATE_INTEREST_PQL'] = "LEGITIMATEINTERESTPQL";
        LegalBasisEnum[LegalBasisEnum["LEGITIMATEINTERESTCLIENT"] = 'LEGITIMATE_INTEREST_CLIENT'] = "LEGITIMATEINTERESTCLIENT";
        LegalBasisEnum[LegalBasisEnum["PERFORMANCEOFCONTRACT"] = 'PERFORMANCE_OF_CONTRACT'] = "PERFORMANCEOFCONTRACT";
        LegalBasisEnum[LegalBasisEnum["CONSENTWITHNOTICE"] = 'CONSENT_WITH_NOTICE'] = "CONSENTWITHNOTICE";
        LegalBasisEnum[LegalBasisEnum["NONGDPR"] = 'NON_GDPR'] = "NONGDPR";
        LegalBasisEnum[LegalBasisEnum["PROCESSANDSTORE"] = 'PROCESS_AND_STORE'] = "PROCESSANDSTORE";
        LegalBasisEnum[LegalBasisEnum["LEGITIMATEINTERESTOTHER"] = 'LEGITIMATE_INTEREST_OTHER'] = "LEGITIMATEINTERESTOTHER";
    })(LegalBasisEnum = PublicUpdateSubscriptionStatusRequest.LegalBasisEnum || (PublicUpdateSubscriptionStatusRequest.LegalBasisEnum = {}));
})(PublicUpdateSubscriptionStatusRequest = exports.PublicUpdateSubscriptionStatusRequest || (exports.PublicUpdateSubscriptionStatusRequest = {}));
