// @hubspot/api-client/lib/codegen/marketing/transactional/model/emailSendStatusView.js
"use strict";
/**
 * Transactional Email
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailSendStatusView = void 0;
/**
* Describes the status of an email send request.
*/
class EmailSendStatusView {
    static getAttributeTypeMap() {
        return EmailSendStatusView.attributeTypeMap;
    }
}
exports.EmailSendStatusView = EmailSendStatusView;
EmailSendStatusView.discriminator = undefined;
EmailSendStatusView.attributeTypeMap = [
    {
        "name": "eventId",
        "baseName": "eventId",
        "type": "EventIdView"
    },
    {
        "name": "statusId",
        "baseName": "statusId",
        "type": "string"
    },
    {
        "name": "sendResult",
        "baseName": "sendResult",
        "type": "EmailSendStatusView.SendResultEnum"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date"
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "EmailSendStatusView.StatusEnum"
    }
];
(function (EmailSendStatusView) {
    let SendResultEnum;
    (function (SendResultEnum) {
        SendResultEnum[SendResultEnum["SENT"] = 'SENT'] = "SENT";
        SendResultEnum[SendResultEnum["IDEMPOTENTIGNORE"] = 'IDEMPOTENT_IGNORE'] = "IDEMPOTENTIGNORE";
        SendResultEnum[SendResultEnum["QUEUED"] = 'QUEUED'] = "QUEUED";
        SendResultEnum[SendResultEnum["IDEMPOTENTFAIL"] = 'IDEMPOTENT_FAIL'] = "IDEMPOTENTFAIL";
        SendResultEnum[SendResultEnum["THROTTLED"] = 'THROTTLED'] = "THROTTLED";
        SendResultEnum[SendResultEnum["EMAILDISABLED"] = 'EMAIL_DISABLED'] = "EMAILDISABLED";
        SendResultEnum[SendResultEnum["PORTALSUSPENDED"] = 'PORTAL_SUSPENDED'] = "PORTALSUSPENDED";
        SendResultEnum[SendResultEnum["INVALIDTOADDRESS"] = 'INVALID_TO_ADDRESS'] = "INVALIDTOADDRESS";
        SendResultEnum[SendResultEnum["BLOCKEDDOMAIN"] = 'BLOCKED_DOMAIN'] = "BLOCKEDDOMAIN";
        SendResultEnum[SendResultEnum["PREVIOUSLYBOUNCED"] = 'PREVIOUSLY_BOUNCED'] = "PREVIOUSLYBOUNCED";
        SendResultEnum[SendResultEnum["EMAILUNCONFIRMED"] = 'EMAIL_UNCONFIRMED'] = "EMAILUNCONFIRMED";
        SendResultEnum[SendResultEnum["PREVIOUSSPAM"] = 'PREVIOUS_SPAM'] = "PREVIOUSSPAM";
        SendResultEnum[SendResultEnum["PREVIOUSLYUNSUBSCRIBEDMESSAGE"] = 'PREVIOUSLY_UNSUBSCRIBED_MESSAGE'] = "PREVIOUSLYUNSUBSCRIBEDMESSAGE";
        SendResultEnum[SendResultEnum["PREVIOUSLYUNSUBSCRIBEDPORTAL"] = 'PREVIOUSLY_UNSUBSCRIBED_PORTAL'] = "PREVIOUSLYUNSUBSCRIBEDPORTAL";
        SendResultEnum[SendResultEnum["INVALIDFROMADDRESS"] = 'INVALID_FROM_ADDRESS'] = "INVALIDFROMADDRESS";
        SendResultEnum[SendResultEnum["CAMPAIGNCANCELLED"] = 'CAMPAIGN_CANCELLED'] = "CAMPAIGNCANCELLED";
        SendResultEnum[SendResultEnum["VALIDATIONFAILED"] = 'VALIDATION_FAILED'] = "VALIDATIONFAILED";
        SendResultEnum[SendResultEnum["MTAIGNORE"] = 'MTA_IGNORE'] = "MTAIGNORE";
        SendResultEnum[SendResultEnum["BLOCKEDADDRESS"] = 'BLOCKED_ADDRESS'] = "BLOCKEDADDRESS";
        SendResultEnum[SendResultEnum["PORTALOVERLIMIT"] = 'PORTAL_OVER_LIMIT'] = "PORTALOVERLIMIT";
        SendResultEnum[SendResultEnum["PORTALEXPIRED"] = 'PORTAL_EXPIRED'] = "PORTALEXPIRED";
        SendResultEnum[SendResultEnum["PORTALMISSINGMARKETINGSCOPE"] = 'PORTAL_MISSING_MARKETING_SCOPE'] = "PORTALMISSINGMARKETINGSCOPE";
        SendResultEnum[SendResultEnum["MISSINGTEMPLATEPROPERTIES"] = 'MISSING_TEMPLATE_PROPERTIES'] = "MISSINGTEMPLATEPROPERTIES";
        SendResultEnum[SendResultEnum["MISSINGREQUIREDPARAMETER"] = 'MISSING_REQUIRED_PARAMETER'] = "MISSINGREQUIREDPARAMETER";
        SendResultEnum[SendResultEnum["PORTALAUTHENTICATIONFAILURE"] = 'PORTAL_AUTHENTICATION_FAILURE'] = "PORTALAUTHENTICATIONFAILURE";
        SendResultEnum[SendResultEnum["MISSINGCONTENT"] = 'MISSING_CONTENT'] = "MISSINGCONTENT";
        SendResultEnum[SendResultEnum["CORRUPTINPUT"] = 'CORRUPT_INPUT'] = "CORRUPTINPUT";
        SendResultEnum[SendResultEnum["TEMPLATERENDEREXCEPTION"] = 'TEMPLATE_RENDER_EXCEPTION'] = "TEMPLATERENDEREXCEPTION";
        SendResultEnum[SendResultEnum["GRAYMAILSUPPRESSED"] = 'GRAYMAIL_SUPPRESSED'] = "GRAYMAILSUPPRESSED";
        SendResultEnum[SendResultEnum["UNCONFIGUREDSENDINGDOMAIN"] = 'UNCONFIGURED_SENDING_DOMAIN'] = "UNCONFIGUREDSENDINGDOMAIN";
        SendResultEnum[SendResultEnum["UNDELIVERABLE"] = 'UNDELIVERABLE'] = "UNDELIVERABLE";
        SendResultEnum[SendResultEnum["CANCELLEDABUSE"] = 'CANCELLED_ABUSE'] = "CANCELLEDABUSE";
        SendResultEnum[SendResultEnum["QUARANTINEDADDRESS"] = 'QUARANTINED_ADDRESS'] = "QUARANTINEDADDRESS";
        SendResultEnum[SendResultEnum["ADDRESSONLYACCEPTEDONPROD"] = 'ADDRESS_ONLY_ACCEPTED_ON_PROD'] = "ADDRESSONLYACCEPTEDONPROD";
        SendResultEnum[SendResultEnum["PORTALNOTAUTHORIZEDFORAPPLICATION"] = 'PORTAL_NOT_AUTHORIZED_FOR_APPLICATION'] = "PORTALNOTAUTHORIZEDFORAPPLICATION";
        SendResultEnum[SendResultEnum["ADDRESSLISTBOMBED"] = 'ADDRESS_LIST_BOMBED'] = "ADDRESSLISTBOMBED";
        SendResultEnum[SendResultEnum["ADDRESSOPTEDOUT"] = 'ADDRESS_OPTED_OUT'] = "ADDRESSOPTEDOUT";
        SendResultEnum[SendResultEnum["RECIPIENTFATIGUESUPPRESSED"] = 'RECIPIENT_FATIGUE_SUPPRESSED'] = "RECIPIENTFATIGUESUPPRESSED";
        SendResultEnum[SendResultEnum["TOOMANYRECIPIENTS"] = 'TOO_MANY_RECIPIENTS'] = "TOOMANYRECIPIENTS";
        SendResultEnum[SendResultEnum["PREVIOUSLYUNSUBSCRIBEDBRAND"] = 'PREVIOUSLY_UNSUBSCRIBED_BRAND'] = "PREVIOUSLYUNSUBSCRIBEDBRAND";
        SendResultEnum[SendResultEnum["NONMARKETABLECONTACT"] = 'NON_MARKETABLE_CONTACT'] = "NONMARKETABLECONTACT";
    })(SendResultEnum = EmailSendStatusView.SendResultEnum || (EmailSendStatusView.SendResultEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["PENDING"] = 'PENDING'] = "PENDING";
        StatusEnum[StatusEnum["PROCESSING"] = 'PROCESSING'] = "PROCESSING";
        StatusEnum[StatusEnum["CANCELED"] = 'CANCELED'] = "CANCELED";
        StatusEnum[StatusEnum["COMPLETE"] = 'COMPLETE'] = "COMPLETE";
    })(StatusEnum = EmailSendStatusView.StatusEnum || (EmailSendStatusView.StatusEnum = {}));
})(EmailSendStatusView = exports.EmailSendStatusView || (exports.EmailSendStatusView = {}));
