// @hubspot/api-client/lib/codegen/crm/tickets/model/batchResponseSimplePublicObject.js
"use strict";
/**
 * Tickets
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseSimplePublicObject = void 0;
class BatchResponseSimplePublicObject {
    static getAttributeTypeMap() {
        return BatchResponseSimplePublicObject.attributeTypeMap;
    }
}
exports.BatchResponseSimplePublicObject = BatchResponseSimplePublicObject;
BatchResponseSimplePublicObject.discriminator = undefined;
BatchResponseSimplePublicObject.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseSimplePublicObject.StatusEnum"
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<SimplePublicObject>"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date"
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }"
    }
];
(function (BatchResponseSimplePublicObject) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["PENDING"] = 'PENDING'] = "PENDING";
        StatusEnum[StatusEnum["PROCESSING"] = 'PROCESSING'] = "PROCESSING";
        StatusEnum[StatusEnum["CANCELED"] = 'CANCELED'] = "CANCELED";
        StatusEnum[StatusEnum["COMPLETE"] = 'COMPLETE'] = "COMPLETE";
    })(StatusEnum = BatchResponseSimplePublicObject.StatusEnum || (BatchResponseSimplePublicObject.StatusEnum = {}));
})(BatchResponseSimplePublicObject = exports.BatchResponseSimplePublicObject || (exports.BatchResponseSimplePublicObject = {}));
