// @hubspot/api-client/lib/codegen/crm/tickets/model/errorCategory.js
"use strict";
/**
 * Tickets
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ErrorCategory = void 0;
class ErrorCategory {
    static getAttributeTypeMap() {
        return ErrorCategory.attributeTypeMap;
    }
}
exports.ErrorCategory = ErrorCategory;
ErrorCategory.discriminator = undefined;
ErrorCategory.attributeTypeMap = [
    {
        "name": "httpStatus",
        "baseName": "httpStatus",
        "type": "ErrorCategory.HttpStatusEnum"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    }
];
(function (ErrorCategory) {
    let HttpStatusEnum;
    (function (HttpStatusEnum) {
        HttpStatusEnum[HttpStatusEnum["CONTINUE"] = 'CONTINUE'] = "CONTINUE";
        HttpStatusEnum[HttpStatusEnum["SWITCHINGPROTOCOLS"] = 'SWITCHING_PROTOCOLS'] = "SWITCHINGPROTOCOLS";
        HttpStatusEnum[HttpStatusEnum["PROCESSING"] = 'PROCESSING'] = "PROCESSING";
        HttpStatusEnum[HttpStatusEnum["OK"] = 'OK'] = "OK";
        HttpStatusEnum[HttpStatusEnum["CREATED"] = 'CREATED'] = "CREATED";
        HttpStatusEnum[HttpStatusEnum["ACCEPTED"] = 'ACCEPTED'] = "ACCEPTED";
        HttpStatusEnum[HttpStatusEnum["NONAUTHORITATIVEINFORMATION"] = 'NON_AUTHORITATIVE_INFORMATION'] = "NONAUTHORITATIVEINFORMATION";
        HttpStatusEnum[HttpStatusEnum["NOCONTENT"] = 'NO_CONTENT'] = "NOCONTENT";
        HttpStatusEnum[HttpStatusEnum["RESETCONTENT"] = 'RESET_CONTENT'] = "RESETCONTENT";
        HttpStatusEnum[HttpStatusEnum["PARTIALCONTENT"] = 'PARTIAL_CONTENT'] = "PARTIALCONTENT";
        HttpStatusEnum[HttpStatusEnum["MULTISTATUS"] = 'MULTI_STATUS'] = "MULTISTATUS";
        HttpStatusEnum[HttpStatusEnum["ALREADYREPORTED"] = 'ALREADY_REPORTED'] = "ALREADYREPORTED";
        HttpStatusEnum[HttpStatusEnum["IMUSED"] = 'IM_USED'] = "IMUSED";
        HttpStatusEnum[HttpStatusEnum["MULTIPLECHOICES"] = 'MULTIPLE_CHOICES'] = "MULTIPLECHOICES";
        HttpStatusEnum[HttpStatusEnum["MOVEDPERMANENTLY"] = 'MOVED_PERMANENTLY'] = "MOVEDPERMANENTLY";
        HttpStatusEnum[HttpStatusEnum["FOUND"] = 'FOUND'] = "FOUND";
        HttpStatusEnum[HttpStatusEnum["SEEOTHER"] = 'SEE_OTHER'] = "SEEOTHER";
        HttpStatusEnum[HttpStatusEnum["NOTMODIFIED"] = 'NOT_MODIFIED'] = "NOTMODIFIED";
        HttpStatusEnum[HttpStatusEnum["USEPROXY"] = 'USE_PROXY'] = "USEPROXY";
        HttpStatusEnum[HttpStatusEnum["TEMPORARYREDIRECT"] = 'TEMPORARY_REDIRECT'] = "TEMPORARYREDIRECT";
        HttpStatusEnum[HttpStatusEnum["PERMANENTREDIRECT"] = 'PERMANENT_REDIRECT'] = "PERMANENTREDIRECT";
        HttpStatusEnum[HttpStatusEnum["BADREQUEST"] = 'BAD_REQUEST'] = "BADREQUEST";
        HttpStatusEnum[HttpStatusEnum["UNAUTHORIZED"] = 'UNAUTHORIZED'] = "UNAUTHORIZED";
        HttpStatusEnum[HttpStatusEnum["PAYMENTREQUIRED"] = 'PAYMENT_REQUIRED'] = "PAYMENTREQUIRED";
        HttpStatusEnum[HttpStatusEnum["FORBIDDEN"] = 'FORBIDDEN'] = "FORBIDDEN";
        HttpStatusEnum[HttpStatusEnum["NOTFOUND"] = 'NOT_FOUND'] = "NOTFOUND";
        HttpStatusEnum[HttpStatusEnum["METHODNOTALLOWED"] = 'METHOD_NOT_ALLOWED'] = "METHODNOTALLOWED";
        HttpStatusEnum[HttpStatusEnum["NOTACCEPTABLE"] = 'NOT_ACCEPTABLE'] = "NOTACCEPTABLE";
        HttpStatusEnum[HttpStatusEnum["PROXYAUTHENTICATIONREQUIRED"] = 'PROXY_AUTHENTICATION_REQUIRED'] = "PROXYAUTHENTICATIONREQUIRED";
        HttpStatusEnum[HttpStatusEnum["REQUESTTIMEOUT"] = 'REQUEST_TIMEOUT'] = "REQUESTTIMEOUT";
        HttpStatusEnum[HttpStatusEnum["CONFLICT"] = 'CONFLICT'] = "CONFLICT";
        HttpStatusEnum[HttpStatusEnum["GONE"] = 'GONE'] = "GONE";
        HttpStatusEnum[HttpStatusEnum["LENGTHREQUIRED"] = 'LENGTH_REQUIRED'] = "LENGTHREQUIRED";
        HttpStatusEnum[HttpStatusEnum["PRECONDITIONFAILED"] = 'PRECONDITION_FAILED'] = "PRECONDITIONFAILED";
        HttpStatusEnum[HttpStatusEnum["REQUESTENTITYTOOLARGE"] = 'REQUEST_ENTITY_TOO_LARGE'] = "REQUESTENTITYTOOLARGE";
        HttpStatusEnum[HttpStatusEnum["REQUESTURITOOLONG"] = 'REQUEST_URI_TOO_LONG'] = "REQUESTURITOOLONG";
        HttpStatusEnum[HttpStatusEnum["UNSUPPORTEDMEDIATYPE"] = 'UNSUPPORTED_MEDIA_TYPE'] = "UNSUPPORTEDMEDIATYPE";
        HttpStatusEnum[HttpStatusEnum["REQUESTEDRANGENOTSATISFIABLE"] = 'REQUESTED_RANGE_NOT_SATISFIABLE'] = "REQUESTEDRANGENOTSATISFIABLE";
        HttpStatusEnum[HttpStatusEnum["EXPECTATIONFAILED"] = 'EXPECTATION_FAILED'] = "EXPECTATIONFAILED";
        HttpStatusEnum[HttpStatusEnum["IMATEAPOT"] = 'IM_A_TEAPOT'] = "IMATEAPOT";
        HttpStatusEnum[HttpStatusEnum["MISDIRECTEDREQUEST"] = 'MISDIRECTED_REQUEST'] = "MISDIRECTEDREQUEST";
        HttpStatusEnum[HttpStatusEnum["UNPROCESSABLEENTITY"] = 'UNPROCESSABLE_ENTITY'] = "UNPROCESSABLEENTITY";
        HttpStatusEnum[HttpStatusEnum["LOCKED"] = 'LOCKED'] = "LOCKED";
        HttpStatusEnum[HttpStatusEnum["FAILEDDEPENDENCY"] = 'FAILED_DEPENDENCY'] = "FAILEDDEPENDENCY";
        HttpStatusEnum[HttpStatusEnum["UPGRADEREQUIRED"] = 'UPGRADE_REQUIRED'] = "UPGRADEREQUIRED";
        HttpStatusEnum[HttpStatusEnum["PRECONDITIONREQUIRED"] = 'PRECONDITION_REQUIRED'] = "PRECONDITIONREQUIRED";
        HttpStatusEnum[HttpStatusEnum["TOOMANYREQUESTS"] = 'TOO_MANY_REQUESTS'] = "TOOMANYREQUESTS";
        HttpStatusEnum[HttpStatusEnum["REQUESTHEADERSFIELDSTOOLARGE"] = 'REQUEST_HEADERS_FIELDS_TOO_LARGE'] = "REQUESTHEADERSFIELDSTOOLARGE";
        HttpStatusEnum[HttpStatusEnum["INTERNALSTALESERVICEDISCOVERY"] = 'INTERNAL_STALE_SERVICE_DISCOVERY'] = "INTERNALSTALESERVICEDISCOVERY";
        HttpStatusEnum[HttpStatusEnum["UNAVAILABLEFORLEGALREASONS"] = 'UNAVAILABLE_FOR_LEGAL_REASONS'] = "UNAVAILABLEFORLEGALREASONS";
        HttpStatusEnum[HttpStatusEnum["INTERNALSERVERERROR"] = 'INTERNAL_SERVER_ERROR'] = "INTERNALSERVERERROR";
        HttpStatusEnum[HttpStatusEnum["NOTIMPLEMENTED"] = 'NOT_IMPLEMENTED'] = "NOTIMPLEMENTED";
        HttpStatusEnum[HttpStatusEnum["BADGATEWAY"] = 'BAD_GATEWAY'] = "BADGATEWAY";
        HttpStatusEnum[HttpStatusEnum["SERVICEUNAVAILABLE"] = 'SERVICE_UNAVAILABLE'] = "SERVICEUNAVAILABLE";
        HttpStatusEnum[HttpStatusEnum["GATEWAYTIMEOUT"] = 'GATEWAY_TIMEOUT'] = "GATEWAYTIMEOUT";
        HttpStatusEnum[HttpStatusEnum["HTTPVERSIONNOTSUPPORTED"] = 'HTTP_VERSION_NOT_SUPPORTED'] = "HTTPVERSIONNOTSUPPORTED";
        HttpStatusEnum[HttpStatusEnum["VARIANTALSONEGOTIATES"] = 'VARIANT_ALSO_NEGOTIATES'] = "VARIANTALSONEGOTIATES";
        HttpStatusEnum[HttpStatusEnum["INSUFFICIENTSTORAGE"] = 'INSUFFICIENT_STORAGE'] = "INSUFFICIENTSTORAGE";
        HttpStatusEnum[HttpStatusEnum["LOOPDETECTED"] = 'LOOP_DETECTED'] = "LOOPDETECTED";
        HttpStatusEnum[HttpStatusEnum["NOTEXTENDED"] = 'NOT_EXTENDED'] = "NOTEXTENDED";
        HttpStatusEnum[HttpStatusEnum["NETWORKAUTHENTICATIONREQUIRED"] = 'NETWORK_AUTHENTICATION_REQUIRED'] = "NETWORKAUTHENTICATIONREQUIRED";
    })(HttpStatusEnum = ErrorCategory.HttpStatusEnum || (ErrorCategory.HttpStatusEnum = {}));
})(ErrorCategory = exports.ErrorCategory || (exports.ErrorCategory = {}));
