// @hubspot/api-client/lib/codegen/automation/actions/api/apis.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.APIS = exports.HttpError = void 0;
__exportStar(require("./callbacksApi"), exports);
const callbacksApi_1 = require("./callbacksApi");
__exportStar(require("./definitionsApi"), exports);
const definitionsApi_1 = require("./definitionsApi");
__exportStar(require("./functionsApi"), exports);
const functionsApi_1 = require("./functionsApi");
__exportStar(require("./revisionsApi"), exports);
const revisionsApi_1 = require("./revisionsApi");
class HttpError extends Error {
    constructor(response, body, statusCode) {
        super('HTTP request failed');
        this.response = response;
        this.body = body;
        this.statusCode = statusCode;
        this.name = 'HttpError';
    }
}
exports.HttpError = HttpError;
exports.APIS = [callbacksApi_1.CallbacksApi, definitionsApi_1.DefinitionsApi, functionsApi_1.FunctionsApi, revisionsApi_1.RevisionsApi];
