// @hubspot/api-client/lib/codegen/cms/site_search/model/contentSearchResult.js
"use strict";
/**
 * CMS Site Search
 * Use these endpoints for searching content on your HubSpot hosted CMS website(s).
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentSearchResult = void 0;
/**
* An individual search result.
*/
class ContentSearchResult {
    static getAttributeTypeMap() {
        return ContentSearchResult.attributeTypeMap;
    }
}
exports.ContentSearchResult = ContentSearchResult;
ContentSearchResult.discriminator = undefined;
ContentSearchResult.attributeTypeMap = [
    {
        "name": "id",
        "baseName": "id",
        "type": "number"
    },
    {
        "name": "score",
        "baseName": "score",
        "type": "number"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "ContentSearchResult.TypeEnum"
    },
    {
        "name": "domain",
        "baseName": "domain",
        "type": "string"
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string"
    },
    {
        "name": "featuredImageUrl",
        "baseName": "featuredImageUrl",
        "type": "string"
    },
    {
        "name": "language",
        "baseName": "language",
        "type": "ContentSearchResult.LanguageEnum"
    },
    {
        "name": "title",
        "baseName": "title",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "category",
        "baseName": "category",
        "type": "string"
    },
    {
        "name": "subcategory",
        "baseName": "subcategory",
        "type": "string"
    },
    {
        "name": "authorFullName",
        "baseName": "authorFullName",
        "type": "string"
    },
    {
        "name": "tags",
        "baseName": "tags",
        "type": "Array<string>"
    },
    {
        "name": "tableId",
        "baseName": "tableId",
        "type": "number"
    },
    {
        "name": "rowId",
        "baseName": "rowId",
        "type": "number"
    },
    {
        "name": "publishedDate",
        "baseName": "publishedDate",
        "type": "number"
    },
    {
        "name": "combinedId",
        "baseName": "combinedId",
        "type": "string"
    }
];
(function (ContentSearchResult) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["LANDINGPAGE"] = 'LANDING_PAGE'] = "LANDINGPAGE";
        TypeEnum[TypeEnum["BLOGPOST"] = 'BLOG_POST'] = "BLOGPOST";
        TypeEnum[TypeEnum["SITEPAGE"] = 'SITE_PAGE'] = "SITEPAGE";
        TypeEnum[TypeEnum["KNOWLEDGEARTICLE"] = 'KNOWLEDGE_ARTICLE'] = "KNOWLEDGEARTICLE";
        TypeEnum[TypeEnum["LISTINGPAGE"] = 'LISTING_PAGE'] = "LISTINGPAGE";
    })(TypeEnum = ContentSearchResult.TypeEnum || (ContentSearchResult.TypeEnum = {}));
    let LanguageEnum;
    (function (LanguageEnum) {
        LanguageEnum[LanguageEnum["Af"] = 'af'] = "Af";
        LanguageEnum[LanguageEnum["AfNa"] = 'af-na'] = "AfNa";
        LanguageEnum[LanguageEnum["AfZa"] = 'af-za'] = "AfZa";
        LanguageEnum[LanguageEnum["Agq"] = 'agq'] = "Agq";
        LanguageEnum[LanguageEnum["AgqCm"] = 'agq-cm'] = "AgqCm";
        LanguageEnum[LanguageEnum["Ak"] = 'ak'] = "Ak";
        LanguageEnum[LanguageEnum["AkGh"] = 'ak-gh'] = "AkGh";
        LanguageEnum[LanguageEnum["Am"] = 'am'] = "Am";
        LanguageEnum[LanguageEnum["AmEt"] = 'am-et'] = "AmEt";
        LanguageEnum[LanguageEnum["Ar"] = 'ar'] = "Ar";
        LanguageEnum[LanguageEnum["Ar001"] = 'ar-001'] = "Ar001";
        LanguageEnum[LanguageEnum["ArAe"] = 'ar-ae'] = "ArAe";
        LanguageEnum[LanguageEnum["ArBh"] = 'ar-bh'] = "ArBh";
        LanguageEnum[LanguageEnum["ArDj"] = 'ar-dj'] = "ArDj";
        LanguageEnum[LanguageEnum["ArDz"] = 'ar-dz'] = "ArDz";
        LanguageEnum[LanguageEnum["ArEg"] = 'ar-eg'] = "ArEg";
        LanguageEnum[LanguageEnum["ArEh"] = 'ar-eh'] = "ArEh";
        LanguageEnum[LanguageEnum["ArEr"] = 'ar-er'] = "ArEr";
        LanguageEnum[LanguageEnum["ArIl"] = 'ar-il'] = "ArIl";
        LanguageEnum[LanguageEnum["ArIq"] = 'ar-iq'] = "ArIq";
        LanguageEnum[LanguageEnum["ArJo"] = 'ar-jo'] = "ArJo";
        LanguageEnum[LanguageEnum["ArKm"] = 'ar-km'] = "ArKm";
        LanguageEnum[LanguageEnum["ArKw"] = 'ar-kw'] = "ArKw";
        LanguageEnum[LanguageEnum["ArLb"] = 'ar-lb'] = "ArLb";
        LanguageEnum[LanguageEnum["ArLy"] = 'ar-ly'] = "ArLy";
        LanguageEnum[LanguageEnum["ArMa"] = 'ar-ma'] = "ArMa";
        LanguageEnum[LanguageEnum["ArMr"] = 'ar-mr'] = "ArMr";
        LanguageEnum[LanguageEnum["ArOm"] = 'ar-om'] = "ArOm";
        LanguageEnum[LanguageEnum["ArPs"] = 'ar-ps'] = "ArPs";
        LanguageEnum[LanguageEnum["ArQa"] = 'ar-qa'] = "ArQa";
        LanguageEnum[LanguageEnum["ArSa"] = 'ar-sa'] = "ArSa";
        LanguageEnum[LanguageEnum["ArSd"] = 'ar-sd'] = "ArSd";
        LanguageEnum[LanguageEnum["ArSo"] = 'ar-so'] = "ArSo";
        LanguageEnum[LanguageEnum["ArSs"] = 'ar-ss'] = "ArSs";
        LanguageEnum[LanguageEnum["ArSy"] = 'ar-sy'] = "ArSy";
        LanguageEnum[LanguageEnum["ArTd"] = 'ar-td'] = "ArTd";
        LanguageEnum[LanguageEnum["ArTn"] = 'ar-tn'] = "ArTn";
        LanguageEnum[LanguageEnum["ArYe"] = 'ar-ye'] = "ArYe";
        LanguageEnum[LanguageEnum["As"] = 'as'] = "As";
        LanguageEnum[LanguageEnum["AsIn"] = 'as-in'] = "AsIn";
        LanguageEnum[LanguageEnum["Asa"] = 'asa'] = "Asa";
        LanguageEnum[LanguageEnum["AsaTz"] = 'asa-tz'] = "AsaTz";
        LanguageEnum[LanguageEnum["Ast"] = 'ast'] = "Ast";
        LanguageEnum[LanguageEnum["AstEs"] = 'ast-es'] = "AstEs";
        LanguageEnum[LanguageEnum["Az"] = 'az'] = "Az";
        LanguageEnum[LanguageEnum["AzAz"] = 'az-az'] = "AzAz";
        LanguageEnum[LanguageEnum["Bas"] = 'bas'] = "Bas";
        LanguageEnum[LanguageEnum["BasCm"] = 'bas-cm'] = "BasCm";
        LanguageEnum[LanguageEnum["Be"] = 'be'] = "Be";
        LanguageEnum[LanguageEnum["BeBy"] = 'be-by'] = "BeBy";
        LanguageEnum[LanguageEnum["Bem"] = 'bem'] = "Bem";
        LanguageEnum[LanguageEnum["BemZm"] = 'bem-zm'] = "BemZm";
        LanguageEnum[LanguageEnum["Bez"] = 'bez'] = "Bez";
        LanguageEnum[LanguageEnum["BezTz"] = 'bez-tz'] = "BezTz";
        LanguageEnum[LanguageEnum["Bg"] = 'bg'] = "Bg";
        LanguageEnum[LanguageEnum["BgBg"] = 'bg-bg'] = "BgBg";
        LanguageEnum[LanguageEnum["Bm"] = 'bm'] = "Bm";
        LanguageEnum[LanguageEnum["BmMl"] = 'bm-ml'] = "BmMl";
        LanguageEnum[LanguageEnum["Bn"] = 'bn'] = "Bn";
        LanguageEnum[LanguageEnum["BnBd"] = 'bn-bd'] = "BnBd";
        LanguageEnum[LanguageEnum["BnIn"] = 'bn-in'] = "BnIn";
        LanguageEnum[LanguageEnum["Bo"] = 'bo'] = "Bo";
        LanguageEnum[LanguageEnum["BoCn"] = 'bo-cn'] = "BoCn";
        LanguageEnum[LanguageEnum["BoIn"] = 'bo-in'] = "BoIn";
        LanguageEnum[LanguageEnum["Br"] = 'br'] = "Br";
        LanguageEnum[LanguageEnum["BrFr"] = 'br-fr'] = "BrFr";
        LanguageEnum[LanguageEnum["Brx"] = 'brx'] = "Brx";
        LanguageEnum[LanguageEnum["BrxIn"] = 'brx-in'] = "BrxIn";
        LanguageEnum[LanguageEnum["Bs"] = 'bs'] = "Bs";
        LanguageEnum[LanguageEnum["BsBa"] = 'bs-ba'] = "BsBa";
        LanguageEnum[LanguageEnum["Ca"] = 'ca'] = "Ca";
        LanguageEnum[LanguageEnum["CaAd"] = 'ca-ad'] = "CaAd";
        LanguageEnum[LanguageEnum["CaEs"] = 'ca-es'] = "CaEs";
        LanguageEnum[LanguageEnum["CaFr"] = 'ca-fr'] = "CaFr";
        LanguageEnum[LanguageEnum["CaIt"] = 'ca-it'] = "CaIt";
        LanguageEnum[LanguageEnum["Ccp"] = 'ccp'] = "Ccp";
        LanguageEnum[LanguageEnum["CcpBd"] = 'ccp-bd'] = "CcpBd";
        LanguageEnum[LanguageEnum["CcpIn"] = 'ccp-in'] = "CcpIn";
        LanguageEnum[LanguageEnum["Ce"] = 'ce'] = "Ce";
        LanguageEnum[LanguageEnum["CeRu"] = 'ce-ru'] = "CeRu";
        LanguageEnum[LanguageEnum["Cgg"] = 'cgg'] = "Cgg";
        LanguageEnum[LanguageEnum["CggUg"] = 'cgg-ug'] = "CggUg";
        LanguageEnum[LanguageEnum["Chr"] = 'chr'] = "Chr";
        LanguageEnum[LanguageEnum["ChrUs"] = 'chr-us'] = "ChrUs";
        LanguageEnum[LanguageEnum["Ckb"] = 'ckb'] = "Ckb";
        LanguageEnum[LanguageEnum["CkbIq"] = 'ckb-iq'] = "CkbIq";
        LanguageEnum[LanguageEnum["CkbIr"] = 'ckb-ir'] = "CkbIr";
        LanguageEnum[LanguageEnum["Cs"] = 'cs'] = "Cs";
        LanguageEnum[LanguageEnum["CsCz"] = 'cs-cz'] = "CsCz";
        LanguageEnum[LanguageEnum["Cu"] = 'cu'] = "Cu";
        LanguageEnum[LanguageEnum["CuRu"] = 'cu-ru'] = "CuRu";
        LanguageEnum[LanguageEnum["Cy"] = 'cy'] = "Cy";
        LanguageEnum[LanguageEnum["CyGb"] = 'cy-gb'] = "CyGb";
        LanguageEnum[LanguageEnum["Da"] = 'da'] = "Da";
        LanguageEnum[LanguageEnum["DaDk"] = 'da-dk'] = "DaDk";
        LanguageEnum[LanguageEnum["DaGl"] = 'da-gl'] = "DaGl";
        LanguageEnum[LanguageEnum["Dav"] = 'dav'] = "Dav";
        LanguageEnum[LanguageEnum["DavKe"] = 'dav-ke'] = "DavKe";
        LanguageEnum[LanguageEnum["De"] = 'de'] = "De";
        LanguageEnum[LanguageEnum["DeAt"] = 'de-at'] = "DeAt";
        LanguageEnum[LanguageEnum["DeBe"] = 'de-be'] = "DeBe";
        LanguageEnum[LanguageEnum["DeCh"] = 'de-ch'] = "DeCh";
        LanguageEnum[LanguageEnum["DeDe"] = 'de-de'] = "DeDe";
        LanguageEnum[LanguageEnum["DeGr"] = 'de-gr'] = "DeGr";
        LanguageEnum[LanguageEnum["DeIt"] = 'de-it'] = "DeIt";
        LanguageEnum[LanguageEnum["DeLi"] = 'de-li'] = "DeLi";
        LanguageEnum[LanguageEnum["DeLu"] = 'de-lu'] = "DeLu";
        LanguageEnum[LanguageEnum["Dje"] = 'dje'] = "Dje";
        LanguageEnum[LanguageEnum["DjeNe"] = 'dje-ne'] = "DjeNe";
        LanguageEnum[LanguageEnum["Dsb"] = 'dsb'] = "Dsb";
        LanguageEnum[LanguageEnum["DsbDe"] = 'dsb-de'] = "DsbDe";
        LanguageEnum[LanguageEnum["Dua"] = 'dua'] = "Dua";
        LanguageEnum[LanguageEnum["DuaCm"] = 'dua-cm'] = "DuaCm";
        LanguageEnum[LanguageEnum["Dyo"] = 'dyo'] = "Dyo";
        LanguageEnum[LanguageEnum["DyoSn"] = 'dyo-sn'] = "DyoSn";
        LanguageEnum[LanguageEnum["Dz"] = 'dz'] = "Dz";
        LanguageEnum[LanguageEnum["DzBt"] = 'dz-bt'] = "DzBt";
        LanguageEnum[LanguageEnum["Ebu"] = 'ebu'] = "Ebu";
        LanguageEnum[LanguageEnum["EbuKe"] = 'ebu-ke'] = "EbuKe";
        LanguageEnum[LanguageEnum["Ee"] = 'ee'] = "Ee";
        LanguageEnum[LanguageEnum["EeGh"] = 'ee-gh'] = "EeGh";
        LanguageEnum[LanguageEnum["EeTg"] = 'ee-tg'] = "EeTg";
        LanguageEnum[LanguageEnum["El"] = 'el'] = "El";
        LanguageEnum[LanguageEnum["ElCy"] = 'el-cy'] = "ElCy";
        LanguageEnum[LanguageEnum["ElGr"] = 'el-gr'] = "ElGr";
        LanguageEnum[LanguageEnum["En"] = 'en'] = "En";
        LanguageEnum[LanguageEnum["En001"] = 'en-001'] = "En001";
        LanguageEnum[LanguageEnum["En150"] = 'en-150'] = "En150";
        LanguageEnum[LanguageEnum["EnAg"] = 'en-ag'] = "EnAg";
        LanguageEnum[LanguageEnum["EnAi"] = 'en-ai'] = "EnAi";
        LanguageEnum[LanguageEnum["EnAs"] = 'en-as'] = "EnAs";
        LanguageEnum[LanguageEnum["EnAt"] = 'en-at'] = "EnAt";
        LanguageEnum[LanguageEnum["EnAu"] = 'en-au'] = "EnAu";
        LanguageEnum[LanguageEnum["EnBb"] = 'en-bb'] = "EnBb";
        LanguageEnum[LanguageEnum["EnBe"] = 'en-be'] = "EnBe";
        LanguageEnum[LanguageEnum["EnBi"] = 'en-bi'] = "EnBi";
        LanguageEnum[LanguageEnum["EnBm"] = 'en-bm'] = "EnBm";
        LanguageEnum[LanguageEnum["EnBs"] = 'en-bs'] = "EnBs";
        LanguageEnum[LanguageEnum["EnBw"] = 'en-bw'] = "EnBw";
        LanguageEnum[LanguageEnum["EnBz"] = 'en-bz'] = "EnBz";
        LanguageEnum[LanguageEnum["EnCa"] = 'en-ca'] = "EnCa";
        LanguageEnum[LanguageEnum["EnCc"] = 'en-cc'] = "EnCc";
        LanguageEnum[LanguageEnum["EnCh"] = 'en-ch'] = "EnCh";
        LanguageEnum[LanguageEnum["EnCk"] = 'en-ck'] = "EnCk";
        LanguageEnum[LanguageEnum["EnCm"] = 'en-cm'] = "EnCm";
        LanguageEnum[LanguageEnum["EnCx"] = 'en-cx'] = "EnCx";
        LanguageEnum[LanguageEnum["EnCy"] = 'en-cy'] = "EnCy";
        LanguageEnum[LanguageEnum["EnDe"] = 'en-de'] = "EnDe";
        LanguageEnum[LanguageEnum["EnDg"] = 'en-dg'] = "EnDg";
        LanguageEnum[LanguageEnum["EnDk"] = 'en-dk'] = "EnDk";
        LanguageEnum[LanguageEnum["EnDm"] = 'en-dm'] = "EnDm";
        LanguageEnum[LanguageEnum["EnEr"] = 'en-er'] = "EnEr";
        LanguageEnum[LanguageEnum["EnFi"] = 'en-fi'] = "EnFi";
        LanguageEnum[LanguageEnum["EnFj"] = 'en-fj'] = "EnFj";
        LanguageEnum[LanguageEnum["EnFk"] = 'en-fk'] = "EnFk";
        LanguageEnum[LanguageEnum["EnFm"] = 'en-fm'] = "EnFm";
        LanguageEnum[LanguageEnum["EnGb"] = 'en-gb'] = "EnGb";
        LanguageEnum[LanguageEnum["EnGd"] = 'en-gd'] = "EnGd";
        LanguageEnum[LanguageEnum["EnGg"] = 'en-gg'] = "EnGg";
        LanguageEnum[LanguageEnum["EnGh"] = 'en-gh'] = "EnGh";
        LanguageEnum[LanguageEnum["EnGi"] = 'en-gi'] = "EnGi";
        LanguageEnum[LanguageEnum["EnGm"] = 'en-gm'] = "EnGm";
        LanguageEnum[LanguageEnum["EnGu"] = 'en-gu'] = "EnGu";
        LanguageEnum[LanguageEnum["EnGy"] = 'en-gy'] = "EnGy";
        LanguageEnum[LanguageEnum["EnHk"] = 'en-hk'] = "EnHk";
        LanguageEnum[LanguageEnum["EnIe"] = 'en-ie'] = "EnIe";
        LanguageEnum[LanguageEnum["EnIl"] = 'en-il'] = "EnIl";
        LanguageEnum[LanguageEnum["EnIm"] = 'en-im'] = "EnIm";
        LanguageEnum[LanguageEnum["EnIn"] = 'en-in'] = "EnIn";
        LanguageEnum[LanguageEnum["EnIo"] = 'en-io'] = "EnIo";
        LanguageEnum[LanguageEnum["EnJe"] = 'en-je'] = "EnJe";
        LanguageEnum[LanguageEnum["EnJm"] = 'en-jm'] = "EnJm";
        LanguageEnum[LanguageEnum["EnKe"] = 'en-ke'] = "EnKe";
        LanguageEnum[LanguageEnum["EnKi"] = 'en-ki'] = "EnKi";
        LanguageEnum[LanguageEnum["EnKn"] = 'en-kn'] = "EnKn";
        LanguageEnum[LanguageEnum["EnKy"] = 'en-ky'] = "EnKy";
        LanguageEnum[LanguageEnum["EnLc"] = 'en-lc'] = "EnLc";
        LanguageEnum[LanguageEnum["EnLr"] = 'en-lr'] = "EnLr";
        LanguageEnum[LanguageEnum["EnLs"] = 'en-ls'] = "EnLs";
        LanguageEnum[LanguageEnum["EnMg"] = 'en-mg'] = "EnMg";
        LanguageEnum[LanguageEnum["EnMh"] = 'en-mh'] = "EnMh";
        LanguageEnum[LanguageEnum["EnMo"] = 'en-mo'] = "EnMo";
        LanguageEnum[LanguageEnum["EnMp"] = 'en-mp'] = "EnMp";
        LanguageEnum[LanguageEnum["EnMs"] = 'en-ms'] = "EnMs";
        LanguageEnum[LanguageEnum["EnMt"] = 'en-mt'] = "EnMt";
        LanguageEnum[LanguageEnum["EnMu"] = 'en-mu'] = "EnMu";
        LanguageEnum[LanguageEnum["EnMw"] = 'en-mw'] = "EnMw";
        LanguageEnum[LanguageEnum["EnMy"] = 'en-my'] = "EnMy";
        LanguageEnum[LanguageEnum["EnNa"] = 'en-na'] = "EnNa";
        LanguageEnum[LanguageEnum["EnNf"] = 'en-nf'] = "EnNf";
        LanguageEnum[LanguageEnum["EnNg"] = 'en-ng'] = "EnNg";
        LanguageEnum[LanguageEnum["EnNl"] = 'en-nl'] = "EnNl";
        LanguageEnum[LanguageEnum["EnNr"] = 'en-nr'] = "EnNr";
        LanguageEnum[LanguageEnum["EnNu"] = 'en-nu'] = "EnNu";
        LanguageEnum[LanguageEnum["EnNz"] = 'en-nz'] = "EnNz";
        LanguageEnum[LanguageEnum["EnPg"] = 'en-pg'] = "EnPg";
        LanguageEnum[LanguageEnum["EnPh"] = 'en-ph'] = "EnPh";
        LanguageEnum[LanguageEnum["EnPk"] = 'en-pk'] = "EnPk";
        LanguageEnum[LanguageEnum["EnPn"] = 'en-pn'] = "EnPn";
        LanguageEnum[LanguageEnum["EnPr"] = 'en-pr'] = "EnPr";
        LanguageEnum[LanguageEnum["EnPw"] = 'en-pw'] = "EnPw";
        LanguageEnum[LanguageEnum["EnRw"] = 'en-rw'] = "EnRw";
        LanguageEnum[LanguageEnum["EnSb"] = 'en-sb'] = "EnSb";
        LanguageEnum[LanguageEnum["EnSc"] = 'en-sc'] = "EnSc";
        LanguageEnum[LanguageEnum["EnSd"] = 'en-sd'] = "EnSd";
        LanguageEnum[LanguageEnum["EnSe"] = 'en-se'] = "EnSe";
        LanguageEnum[LanguageEnum["EnSg"] = 'en-sg'] = "EnSg";
        LanguageEnum[LanguageEnum["EnSh"] = 'en-sh'] = "EnSh";
        LanguageEnum[LanguageEnum["EnSi"] = 'en-si'] = "EnSi";
        LanguageEnum[LanguageEnum["EnSl"] = 'en-sl'] = "EnSl";
        LanguageEnum[LanguageEnum["EnSs"] = 'en-ss'] = "EnSs";
        LanguageEnum[LanguageEnum["EnSx"] = 'en-sx'] = "EnSx";
        LanguageEnum[LanguageEnum["EnSz"] = 'en-sz'] = "EnSz";
        LanguageEnum[LanguageEnum["EnTc"] = 'en-tc'] = "EnTc";
        LanguageEnum[LanguageEnum["EnTk"] = 'en-tk'] = "EnTk";
        LanguageEnum[LanguageEnum["EnTo"] = 'en-to'] = "EnTo";
        LanguageEnum[LanguageEnum["EnTt"] = 'en-tt'] = "EnTt";
        LanguageEnum[LanguageEnum["EnTv"] = 'en-tv'] = "EnTv";
        LanguageEnum[LanguageEnum["EnTz"] = 'en-tz'] = "EnTz";
        LanguageEnum[LanguageEnum["EnUg"] = 'en-ug'] = "EnUg";
        LanguageEnum[LanguageEnum["EnUm"] = 'en-um'] = "EnUm";
        LanguageEnum[LanguageEnum["EnUs"] = 'en-us'] = "EnUs";
        LanguageEnum[LanguageEnum["EnVc"] = 'en-vc'] = "EnVc";
        LanguageEnum[LanguageEnum["EnVg"] = 'en-vg'] = "EnVg";
        LanguageEnum[LanguageEnum["EnVi"] = 'en-vi'] = "EnVi";
        LanguageEnum[LanguageEnum["EnVu"] = 'en-vu'] = "EnVu";
        LanguageEnum[LanguageEnum["EnWs"] = 'en-ws'] = "EnWs";
        LanguageEnum[LanguageEnum["EnZa"] = 'en-za'] = "EnZa";
        LanguageEnum[LanguageEnum["EnZm"] = 'en-zm'] = "EnZm";
        LanguageEnum[LanguageEnum["EnZw"] = 'en-zw'] = "EnZw";
        LanguageEnum[LanguageEnum["Eo"] = 'eo'] = "Eo";
        LanguageEnum[LanguageEnum["Eo001"] = 'eo-001'] = "Eo001";
        LanguageEnum[LanguageEnum["Es"] = 'es'] = "Es";
        LanguageEnum[LanguageEnum["Es419"] = 'es-419'] = "Es419";
        LanguageEnum[LanguageEnum["EsAr"] = 'es-ar'] = "EsAr";
        LanguageEnum[LanguageEnum["EsBo"] = 'es-bo'] = "EsBo";
        LanguageEnum[LanguageEnum["EsBr"] = 'es-br'] = "EsBr";
        LanguageEnum[LanguageEnum["EsBz"] = 'es-bz'] = "EsBz";
        LanguageEnum[LanguageEnum["EsCl"] = 'es-cl'] = "EsCl";
        LanguageEnum[LanguageEnum["EsCo"] = 'es-co'] = "EsCo";
        LanguageEnum[LanguageEnum["EsCr"] = 'es-cr'] = "EsCr";
        LanguageEnum[LanguageEnum["EsCu"] = 'es-cu'] = "EsCu";
        LanguageEnum[LanguageEnum["EsDo"] = 'es-do'] = "EsDo";
        LanguageEnum[LanguageEnum["EsEa"] = 'es-ea'] = "EsEa";
        LanguageEnum[LanguageEnum["EsEc"] = 'es-ec'] = "EsEc";
        LanguageEnum[LanguageEnum["EsEs"] = 'es-es'] = "EsEs";
        LanguageEnum[LanguageEnum["EsGq"] = 'es-gq'] = "EsGq";
        LanguageEnum[LanguageEnum["EsGt"] = 'es-gt'] = "EsGt";
        LanguageEnum[LanguageEnum["EsHn"] = 'es-hn'] = "EsHn";
        LanguageEnum[LanguageEnum["EsIc"] = 'es-ic'] = "EsIc";
        LanguageEnum[LanguageEnum["EsMx"] = 'es-mx'] = "EsMx";
        LanguageEnum[LanguageEnum["EsNi"] = 'es-ni'] = "EsNi";
        LanguageEnum[LanguageEnum["EsPa"] = 'es-pa'] = "EsPa";
        LanguageEnum[LanguageEnum["EsPe"] = 'es-pe'] = "EsPe";
        LanguageEnum[LanguageEnum["EsPh"] = 'es-ph'] = "EsPh";
        LanguageEnum[LanguageEnum["EsPr"] = 'es-pr'] = "EsPr";
        LanguageEnum[LanguageEnum["EsPy"] = 'es-py'] = "EsPy";
        LanguageEnum[LanguageEnum["EsSv"] = 'es-sv'] = "EsSv";
        LanguageEnum[LanguageEnum["EsUs"] = 'es-us'] = "EsUs";
        LanguageEnum[LanguageEnum["EsUy"] = 'es-uy'] = "EsUy";
        LanguageEnum[LanguageEnum["EsVe"] = 'es-ve'] = "EsVe";
        LanguageEnum[LanguageEnum["Et"] = 'et'] = "Et";
        LanguageEnum[LanguageEnum["EtEe"] = 'et-ee'] = "EtEe";
        LanguageEnum[LanguageEnum["Eu"] = 'eu'] = "Eu";
        LanguageEnum[LanguageEnum["EuEs"] = 'eu-es'] = "EuEs";
        LanguageEnum[LanguageEnum["Ewo"] = 'ewo'] = "Ewo";
        LanguageEnum[LanguageEnum["EwoCm"] = 'ewo-cm'] = "EwoCm";
        LanguageEnum[LanguageEnum["Fa"] = 'fa'] = "Fa";
        LanguageEnum[LanguageEnum["FaAf"] = 'fa-af'] = "FaAf";
        LanguageEnum[LanguageEnum["FaIr"] = 'fa-ir'] = "FaIr";
        LanguageEnum[LanguageEnum["Ff"] = 'ff'] = "Ff";
        LanguageEnum[LanguageEnum["FfCm"] = 'ff-cm'] = "FfCm";
        LanguageEnum[LanguageEnum["FfGn"] = 'ff-gn'] = "FfGn";
        LanguageEnum[LanguageEnum["FfMr"] = 'ff-mr'] = "FfMr";
        LanguageEnum[LanguageEnum["FfSn"] = 'ff-sn'] = "FfSn";
        LanguageEnum[LanguageEnum["Fi"] = 'fi'] = "Fi";
        LanguageEnum[LanguageEnum["FiFi"] = 'fi-fi'] = "FiFi";
        LanguageEnum[LanguageEnum["Fil"] = 'fil'] = "Fil";
        LanguageEnum[LanguageEnum["FilPh"] = 'fil-ph'] = "FilPh";
        LanguageEnum[LanguageEnum["Fo"] = 'fo'] = "Fo";
        LanguageEnum[LanguageEnum["FoDk"] = 'fo-dk'] = "FoDk";
        LanguageEnum[LanguageEnum["FoFo"] = 'fo-fo'] = "FoFo";
        LanguageEnum[LanguageEnum["Fr"] = 'fr'] = "Fr";
        LanguageEnum[LanguageEnum["FrBe"] = 'fr-be'] = "FrBe";
        LanguageEnum[LanguageEnum["FrBf"] = 'fr-bf'] = "FrBf";
        LanguageEnum[LanguageEnum["FrBi"] = 'fr-bi'] = "FrBi";
        LanguageEnum[LanguageEnum["FrBj"] = 'fr-bj'] = "FrBj";
        LanguageEnum[LanguageEnum["FrBl"] = 'fr-bl'] = "FrBl";
        LanguageEnum[LanguageEnum["FrCa"] = 'fr-ca'] = "FrCa";
        LanguageEnum[LanguageEnum["FrCd"] = 'fr-cd'] = "FrCd";
        LanguageEnum[LanguageEnum["FrCf"] = 'fr-cf'] = "FrCf";
        LanguageEnum[LanguageEnum["FrCg"] = 'fr-cg'] = "FrCg";
        LanguageEnum[LanguageEnum["FrCh"] = 'fr-ch'] = "FrCh";
        LanguageEnum[LanguageEnum["FrCi"] = 'fr-ci'] = "FrCi";
        LanguageEnum[LanguageEnum["FrCm"] = 'fr-cm'] = "FrCm";
        LanguageEnum[LanguageEnum["FrDj"] = 'fr-dj'] = "FrDj";
        LanguageEnum[LanguageEnum["FrDz"] = 'fr-dz'] = "FrDz";
        LanguageEnum[LanguageEnum["FrFr"] = 'fr-fr'] = "FrFr";
        LanguageEnum[LanguageEnum["FrGa"] = 'fr-ga'] = "FrGa";
        LanguageEnum[LanguageEnum["FrGf"] = 'fr-gf'] = "FrGf";
        LanguageEnum[LanguageEnum["FrGn"] = 'fr-gn'] = "FrGn";
        LanguageEnum[LanguageEnum["FrGp"] = 'fr-gp'] = "FrGp";
        LanguageEnum[LanguageEnum["FrGq"] = 'fr-gq'] = "FrGq";
        LanguageEnum[LanguageEnum["FrHt"] = 'fr-ht'] = "FrHt";
        LanguageEnum[LanguageEnum["FrKm"] = 'fr-km'] = "FrKm";
        LanguageEnum[LanguageEnum["FrLu"] = 'fr-lu'] = "FrLu";
        LanguageEnum[LanguageEnum["FrMa"] = 'fr-ma'] = "FrMa";
        LanguageEnum[LanguageEnum["FrMc"] = 'fr-mc'] = "FrMc";
        LanguageEnum[LanguageEnum["FrMf"] = 'fr-mf'] = "FrMf";
        LanguageEnum[LanguageEnum["FrMg"] = 'fr-mg'] = "FrMg";
        LanguageEnum[LanguageEnum["FrMl"] = 'fr-ml'] = "FrMl";
        LanguageEnum[LanguageEnum["FrMq"] = 'fr-mq'] = "FrMq";
        LanguageEnum[LanguageEnum["FrMr"] = 'fr-mr'] = "FrMr";
        LanguageEnum[LanguageEnum["FrMu"] = 'fr-mu'] = "FrMu";
        LanguageEnum[LanguageEnum["FrNc"] = 'fr-nc'] = "FrNc";
        LanguageEnum[LanguageEnum["FrNe"] = 'fr-ne'] = "FrNe";
        LanguageEnum[LanguageEnum["FrPf"] = 'fr-pf'] = "FrPf";
        LanguageEnum[LanguageEnum["FrPm"] = 'fr-pm'] = "FrPm";
        LanguageEnum[LanguageEnum["FrRe"] = 'fr-re'] = "FrRe";
        LanguageEnum[LanguageEnum["FrRw"] = 'fr-rw'] = "FrRw";
        LanguageEnum[LanguageEnum["FrSc"] = 'fr-sc'] = "FrSc";
        LanguageEnum[LanguageEnum["FrSn"] = 'fr-sn'] = "FrSn";
        LanguageEnum[LanguageEnum["FrSy"] = 'fr-sy'] = "FrSy";
        LanguageEnum[LanguageEnum["FrTd"] = 'fr-td'] = "FrTd";
        LanguageEnum[LanguageEnum["FrTg"] = 'fr-tg'] = "FrTg";
        LanguageEnum[LanguageEnum["FrTn"] = 'fr-tn'] = "FrTn";
        LanguageEnum[LanguageEnum["FrVu"] = 'fr-vu'] = "FrVu";
        LanguageEnum[LanguageEnum["FrWf"] = 'fr-wf'] = "FrWf";
        LanguageEnum[LanguageEnum["FrYt"] = 'fr-yt'] = "FrYt";
        LanguageEnum[LanguageEnum["Fur"] = 'fur'] = "Fur";
        LanguageEnum[LanguageEnum["FurIt"] = 'fur-it'] = "FurIt";
        LanguageEnum[LanguageEnum["Fy"] = 'fy'] = "Fy";
        LanguageEnum[LanguageEnum["FyNl"] = 'fy-nl'] = "FyNl";
        LanguageEnum[LanguageEnum["Ga"] = 'ga'] = "Ga";
        LanguageEnum[LanguageEnum["GaIe"] = 'ga-ie'] = "GaIe";
        LanguageEnum[LanguageEnum["Gd"] = 'gd'] = "Gd";
        LanguageEnum[LanguageEnum["GdGb"] = 'gd-gb'] = "GdGb";
        LanguageEnum[LanguageEnum["Gl"] = 'gl'] = "Gl";
        LanguageEnum[LanguageEnum["GlEs"] = 'gl-es'] = "GlEs";
        LanguageEnum[LanguageEnum["Gsw"] = 'gsw'] = "Gsw";
        LanguageEnum[LanguageEnum["GswCh"] = 'gsw-ch'] = "GswCh";
        LanguageEnum[LanguageEnum["GswFr"] = 'gsw-fr'] = "GswFr";
        LanguageEnum[LanguageEnum["GswLi"] = 'gsw-li'] = "GswLi";
        LanguageEnum[LanguageEnum["Gu"] = 'gu'] = "Gu";
        LanguageEnum[LanguageEnum["GuIn"] = 'gu-in'] = "GuIn";
        LanguageEnum[LanguageEnum["Guz"] = 'guz'] = "Guz";
        LanguageEnum[LanguageEnum["GuzKe"] = 'guz-ke'] = "GuzKe";
        LanguageEnum[LanguageEnum["Gv"] = 'gv'] = "Gv";
        LanguageEnum[LanguageEnum["GvIm"] = 'gv-im'] = "GvIm";
        LanguageEnum[LanguageEnum["Ha"] = 'ha'] = "Ha";
        LanguageEnum[LanguageEnum["HaGh"] = 'ha-gh'] = "HaGh";
        LanguageEnum[LanguageEnum["HaNe"] = 'ha-ne'] = "HaNe";
        LanguageEnum[LanguageEnum["HaNg"] = 'ha-ng'] = "HaNg";
        LanguageEnum[LanguageEnum["Haw"] = 'haw'] = "Haw";
        LanguageEnum[LanguageEnum["HawUs"] = 'haw-us'] = "HawUs";
        LanguageEnum[LanguageEnum["He"] = 'he'] = "He";
        LanguageEnum[LanguageEnum["Hi"] = 'hi'] = "Hi";
        LanguageEnum[LanguageEnum["HiIn"] = 'hi-in'] = "HiIn";
        LanguageEnum[LanguageEnum["Hr"] = 'hr'] = "Hr";
        LanguageEnum[LanguageEnum["HrBa"] = 'hr-ba'] = "HrBa";
        LanguageEnum[LanguageEnum["HrHr"] = 'hr-hr'] = "HrHr";
        LanguageEnum[LanguageEnum["Hsb"] = 'hsb'] = "Hsb";
        LanguageEnum[LanguageEnum["HsbDe"] = 'hsb-de'] = "HsbDe";
        LanguageEnum[LanguageEnum["Hu"] = 'hu'] = "Hu";
        LanguageEnum[LanguageEnum["HuHu"] = 'hu-hu'] = "HuHu";
        LanguageEnum[LanguageEnum["Hy"] = 'hy'] = "Hy";
        LanguageEnum[LanguageEnum["HyAm"] = 'hy-am'] = "HyAm";
        LanguageEnum[LanguageEnum["Id"] = 'id'] = "Id";
        LanguageEnum[LanguageEnum["Ig"] = 'ig'] = "Ig";
        LanguageEnum[LanguageEnum["IgNg"] = 'ig-ng'] = "IgNg";
        LanguageEnum[LanguageEnum["Ii"] = 'ii'] = "Ii";
        LanguageEnum[LanguageEnum["IiCn"] = 'ii-cn'] = "IiCn";
        LanguageEnum[LanguageEnum["IdId"] = 'id-id'] = "IdId";
        LanguageEnum[LanguageEnum["Is"] = 'is'] = "Is";
        LanguageEnum[LanguageEnum["IsIs"] = 'is-is'] = "IsIs";
        LanguageEnum[LanguageEnum["It"] = 'it'] = "It";
        LanguageEnum[LanguageEnum["ItCh"] = 'it-ch'] = "ItCh";
        LanguageEnum[LanguageEnum["ItIt"] = 'it-it'] = "ItIt";
        LanguageEnum[LanguageEnum["ItSm"] = 'it-sm'] = "ItSm";
        LanguageEnum[LanguageEnum["ItVa"] = 'it-va'] = "ItVa";
        LanguageEnum[LanguageEnum["HeIl"] = 'he-il'] = "HeIl";
        LanguageEnum[LanguageEnum["Ja"] = 'ja'] = "Ja";
        LanguageEnum[LanguageEnum["JaJp"] = 'ja-jp'] = "JaJp";
        LanguageEnum[LanguageEnum["Jgo"] = 'jgo'] = "Jgo";
        LanguageEnum[LanguageEnum["JgoCm"] = 'jgo-cm'] = "JgoCm";
        LanguageEnum[LanguageEnum["Yi"] = 'yi'] = "Yi";
        LanguageEnum[LanguageEnum["Yi001"] = 'yi-001'] = "Yi001";
        LanguageEnum[LanguageEnum["Jmc"] = 'jmc'] = "Jmc";
        LanguageEnum[LanguageEnum["JmcTz"] = 'jmc-tz'] = "JmcTz";
        LanguageEnum[LanguageEnum["Ka"] = 'ka'] = "Ka";
        LanguageEnum[LanguageEnum["KaGe"] = 'ka-ge'] = "KaGe";
        LanguageEnum[LanguageEnum["Kab"] = 'kab'] = "Kab";
        LanguageEnum[LanguageEnum["KabDz"] = 'kab-dz'] = "KabDz";
        LanguageEnum[LanguageEnum["Kam"] = 'kam'] = "Kam";
        LanguageEnum[LanguageEnum["KamKe"] = 'kam-ke'] = "KamKe";
        LanguageEnum[LanguageEnum["Kde"] = 'kde'] = "Kde";
        LanguageEnum[LanguageEnum["KdeTz"] = 'kde-tz'] = "KdeTz";
        LanguageEnum[LanguageEnum["Kea"] = 'kea'] = "Kea";
        LanguageEnum[LanguageEnum["KeaCv"] = 'kea-cv'] = "KeaCv";
        LanguageEnum[LanguageEnum["Khq"] = 'khq'] = "Khq";
        LanguageEnum[LanguageEnum["KhqMl"] = 'khq-ml'] = "KhqMl";
        LanguageEnum[LanguageEnum["Ki"] = 'ki'] = "Ki";
        LanguageEnum[LanguageEnum["KiKe"] = 'ki-ke'] = "KiKe";
        LanguageEnum[LanguageEnum["Kk"] = 'kk'] = "Kk";
        LanguageEnum[LanguageEnum["KkKz"] = 'kk-kz'] = "KkKz";
        LanguageEnum[LanguageEnum["Kkj"] = 'kkj'] = "Kkj";
        LanguageEnum[LanguageEnum["KkjCm"] = 'kkj-cm'] = "KkjCm";
        LanguageEnum[LanguageEnum["Kl"] = 'kl'] = "Kl";
        LanguageEnum[LanguageEnum["KlGl"] = 'kl-gl'] = "KlGl";
        LanguageEnum[LanguageEnum["Kln"] = 'kln'] = "Kln";
        LanguageEnum[LanguageEnum["KlnKe"] = 'kln-ke'] = "KlnKe";
        LanguageEnum[LanguageEnum["Km"] = 'km'] = "Km";
        LanguageEnum[LanguageEnum["KmKh"] = 'km-kh'] = "KmKh";
        LanguageEnum[LanguageEnum["Kn"] = 'kn'] = "Kn";
        LanguageEnum[LanguageEnum["KnIn"] = 'kn-in'] = "KnIn";
        LanguageEnum[LanguageEnum["Ko"] = 'ko'] = "Ko";
        LanguageEnum[LanguageEnum["KoKp"] = 'ko-kp'] = "KoKp";
        LanguageEnum[LanguageEnum["KoKr"] = 'ko-kr'] = "KoKr";
        LanguageEnum[LanguageEnum["Kok"] = 'kok'] = "Kok";
        LanguageEnum[LanguageEnum["KokIn"] = 'kok-in'] = "KokIn";
        LanguageEnum[LanguageEnum["Ks"] = 'ks'] = "Ks";
        LanguageEnum[LanguageEnum["KsIn"] = 'ks-in'] = "KsIn";
        LanguageEnum[LanguageEnum["Ksb"] = 'ksb'] = "Ksb";
        LanguageEnum[LanguageEnum["KsbTz"] = 'ksb-tz'] = "KsbTz";
        LanguageEnum[LanguageEnum["Ksf"] = 'ksf'] = "Ksf";
        LanguageEnum[LanguageEnum["KsfCm"] = 'ksf-cm'] = "KsfCm";
        LanguageEnum[LanguageEnum["Ksh"] = 'ksh'] = "Ksh";
        LanguageEnum[LanguageEnum["KshDe"] = 'ksh-de'] = "KshDe";
        LanguageEnum[LanguageEnum["Kw"] = 'kw'] = "Kw";
        LanguageEnum[LanguageEnum["KwGb"] = 'kw-gb'] = "KwGb";
        LanguageEnum[LanguageEnum["Ky"] = 'ky'] = "Ky";
        LanguageEnum[LanguageEnum["KyKg"] = 'ky-kg'] = "KyKg";
        LanguageEnum[LanguageEnum["Lag"] = 'lag'] = "Lag";
        LanguageEnum[LanguageEnum["LagTz"] = 'lag-tz'] = "LagTz";
        LanguageEnum[LanguageEnum["Lb"] = 'lb'] = "Lb";
        LanguageEnum[LanguageEnum["LbLu"] = 'lb-lu'] = "LbLu";
        LanguageEnum[LanguageEnum["Lg"] = 'lg'] = "Lg";
        LanguageEnum[LanguageEnum["LgUg"] = 'lg-ug'] = "LgUg";
        LanguageEnum[LanguageEnum["Lkt"] = 'lkt'] = "Lkt";
        LanguageEnum[LanguageEnum["LktUs"] = 'lkt-us'] = "LktUs";
        LanguageEnum[LanguageEnum["Ln"] = 'ln'] = "Ln";
        LanguageEnum[LanguageEnum["LnAo"] = 'ln-ao'] = "LnAo";
        LanguageEnum[LanguageEnum["LnCd"] = 'ln-cd'] = "LnCd";
        LanguageEnum[LanguageEnum["LnCf"] = 'ln-cf'] = "LnCf";
        LanguageEnum[LanguageEnum["LnCg"] = 'ln-cg'] = "LnCg";
        LanguageEnum[LanguageEnum["Lo"] = 'lo'] = "Lo";
        LanguageEnum[LanguageEnum["LoLa"] = 'lo-la'] = "LoLa";
        LanguageEnum[LanguageEnum["Lrc"] = 'lrc'] = "Lrc";
        LanguageEnum[LanguageEnum["LrcIq"] = 'lrc-iq'] = "LrcIq";
        LanguageEnum[LanguageEnum["LrcIr"] = 'lrc-ir'] = "LrcIr";
        LanguageEnum[LanguageEnum["Lt"] = 'lt'] = "Lt";
        LanguageEnum[LanguageEnum["LtLt"] = 'lt-lt'] = "LtLt";
        LanguageEnum[LanguageEnum["Lu"] = 'lu'] = "Lu";
        LanguageEnum[LanguageEnum["LuCd"] = 'lu-cd'] = "LuCd";
        LanguageEnum[LanguageEnum["Luo"] = 'luo'] = "Luo";
        LanguageEnum[LanguageEnum["LuoKe"] = 'luo-ke'] = "LuoKe";
        LanguageEnum[LanguageEnum["Luy"] = 'luy'] = "Luy";
        LanguageEnum[LanguageEnum["LuyKe"] = 'luy-ke'] = "LuyKe";
        LanguageEnum[LanguageEnum["Lv"] = 'lv'] = "Lv";
        LanguageEnum[LanguageEnum["LvLv"] = 'lv-lv'] = "LvLv";
        LanguageEnum[LanguageEnum["Mas"] = 'mas'] = "Mas";
        LanguageEnum[LanguageEnum["MasKe"] = 'mas-ke'] = "MasKe";
        LanguageEnum[LanguageEnum["MasTz"] = 'mas-tz'] = "MasTz";
        LanguageEnum[LanguageEnum["Mer"] = 'mer'] = "Mer";
        LanguageEnum[LanguageEnum["MerKe"] = 'mer-ke'] = "MerKe";
        LanguageEnum[LanguageEnum["Mfe"] = 'mfe'] = "Mfe";
        LanguageEnum[LanguageEnum["MfeMu"] = 'mfe-mu'] = "MfeMu";
        LanguageEnum[LanguageEnum["Mg"] = 'mg'] = "Mg";
        LanguageEnum[LanguageEnum["MgMg"] = 'mg-mg'] = "MgMg";
        LanguageEnum[LanguageEnum["Mgh"] = 'mgh'] = "Mgh";
        LanguageEnum[LanguageEnum["MghMz"] = 'mgh-mz'] = "MghMz";
        LanguageEnum[LanguageEnum["Mgo"] = 'mgo'] = "Mgo";
        LanguageEnum[LanguageEnum["MgoCm"] = 'mgo-cm'] = "MgoCm";
        LanguageEnum[LanguageEnum["Mk"] = 'mk'] = "Mk";
        LanguageEnum[LanguageEnum["MkMk"] = 'mk-mk'] = "MkMk";
        LanguageEnum[LanguageEnum["Ml"] = 'ml'] = "Ml";
        LanguageEnum[LanguageEnum["MlIn"] = 'ml-in'] = "MlIn";
        LanguageEnum[LanguageEnum["Mn"] = 'mn'] = "Mn";
        LanguageEnum[LanguageEnum["MnMn"] = 'mn-mn'] = "MnMn";
        LanguageEnum[LanguageEnum["Mr"] = 'mr'] = "Mr";
        LanguageEnum[LanguageEnum["MrIn"] = 'mr-in'] = "MrIn";
        LanguageEnum[LanguageEnum["Ms"] = 'ms'] = "Ms";
        LanguageEnum[LanguageEnum["MsBn"] = 'ms-bn'] = "MsBn";
        LanguageEnum[LanguageEnum["MsMy"] = 'ms-my'] = "MsMy";
        LanguageEnum[LanguageEnum["MsSg"] = 'ms-sg'] = "MsSg";
        LanguageEnum[LanguageEnum["Mt"] = 'mt'] = "Mt";
        LanguageEnum[LanguageEnum["MtMt"] = 'mt-mt'] = "MtMt";
        LanguageEnum[LanguageEnum["Mua"] = 'mua'] = "Mua";
        LanguageEnum[LanguageEnum["MuaCm"] = 'mua-cm'] = "MuaCm";
        LanguageEnum[LanguageEnum["My"] = 'my'] = "My";
        LanguageEnum[LanguageEnum["MyMm"] = 'my-mm'] = "MyMm";
        LanguageEnum[LanguageEnum["Mzn"] = 'mzn'] = "Mzn";
        LanguageEnum[LanguageEnum["MznIr"] = 'mzn-ir'] = "MznIr";
        LanguageEnum[LanguageEnum["Naq"] = 'naq'] = "Naq";
        LanguageEnum[LanguageEnum["NaqNa"] = 'naq-na'] = "NaqNa";
        LanguageEnum[LanguageEnum["Nb"] = 'nb'] = "Nb";
        LanguageEnum[LanguageEnum["NbNo"] = 'nb-no'] = "NbNo";
        LanguageEnum[LanguageEnum["NbSj"] = 'nb-sj'] = "NbSj";
        LanguageEnum[LanguageEnum["Nd"] = 'nd'] = "Nd";
        LanguageEnum[LanguageEnum["NdZw"] = 'nd-zw'] = "NdZw";
        LanguageEnum[LanguageEnum["Nds"] = 'nds'] = "Nds";
        LanguageEnum[LanguageEnum["NdsDe"] = 'nds-de'] = "NdsDe";
        LanguageEnum[LanguageEnum["NdsNl"] = 'nds-nl'] = "NdsNl";
        LanguageEnum[LanguageEnum["Ne"] = 'ne'] = "Ne";
        LanguageEnum[LanguageEnum["NeIn"] = 'ne-in'] = "NeIn";
        LanguageEnum[LanguageEnum["NeNp"] = 'ne-np'] = "NeNp";
        LanguageEnum[LanguageEnum["Nl"] = 'nl'] = "Nl";
        LanguageEnum[LanguageEnum["NlAw"] = 'nl-aw'] = "NlAw";
        LanguageEnum[LanguageEnum["NlBe"] = 'nl-be'] = "NlBe";
        LanguageEnum[LanguageEnum["NlBq"] = 'nl-bq'] = "NlBq";
        LanguageEnum[LanguageEnum["NlCw"] = 'nl-cw'] = "NlCw";
        LanguageEnum[LanguageEnum["NlNl"] = 'nl-nl'] = "NlNl";
        LanguageEnum[LanguageEnum["NlSr"] = 'nl-sr'] = "NlSr";
        LanguageEnum[LanguageEnum["NlSx"] = 'nl-sx'] = "NlSx";
        LanguageEnum[LanguageEnum["Nmg"] = 'nmg'] = "Nmg";
        LanguageEnum[LanguageEnum["NmgCm"] = 'nmg-cm'] = "NmgCm";
        LanguageEnum[LanguageEnum["Nn"] = 'nn'] = "Nn";
        LanguageEnum[LanguageEnum["NnNo"] = 'nn-no'] = "NnNo";
        LanguageEnum[LanguageEnum["Nnh"] = 'nnh'] = "Nnh";
        LanguageEnum[LanguageEnum["NnhCm"] = 'nnh-cm'] = "NnhCm";
        LanguageEnum[LanguageEnum["No"] = 'no'] = "No";
        LanguageEnum[LanguageEnum["NoNo"] = 'no-no'] = "NoNo";
        LanguageEnum[LanguageEnum["Nus"] = 'nus'] = "Nus";
        LanguageEnum[LanguageEnum["NusSs"] = 'nus-ss'] = "NusSs";
        LanguageEnum[LanguageEnum["Nyn"] = 'nyn'] = "Nyn";
        LanguageEnum[LanguageEnum["NynUg"] = 'nyn-ug'] = "NynUg";
        LanguageEnum[LanguageEnum["Om"] = 'om'] = "Om";
        LanguageEnum[LanguageEnum["OmEt"] = 'om-et'] = "OmEt";
        LanguageEnum[LanguageEnum["OmKe"] = 'om-ke'] = "OmKe";
        LanguageEnum[LanguageEnum["Or"] = 'or'] = "Or";
        LanguageEnum[LanguageEnum["OrIn"] = 'or-in'] = "OrIn";
        LanguageEnum[LanguageEnum["Os"] = 'os'] = "Os";
        LanguageEnum[LanguageEnum["OsGe"] = 'os-ge'] = "OsGe";
        LanguageEnum[LanguageEnum["OsRu"] = 'os-ru'] = "OsRu";
        LanguageEnum[LanguageEnum["Pa"] = 'pa'] = "Pa";
        LanguageEnum[LanguageEnum["PaIn"] = 'pa-in'] = "PaIn";
        LanguageEnum[LanguageEnum["PaPk"] = 'pa-pk'] = "PaPk";
        LanguageEnum[LanguageEnum["Pl"] = 'pl'] = "Pl";
        LanguageEnum[LanguageEnum["PlPl"] = 'pl-pl'] = "PlPl";
        LanguageEnum[LanguageEnum["Prg"] = 'prg'] = "Prg";
        LanguageEnum[LanguageEnum["Prg001"] = 'prg-001'] = "Prg001";
        LanguageEnum[LanguageEnum["Ps"] = 'ps'] = "Ps";
        LanguageEnum[LanguageEnum["PsAf"] = 'ps-af'] = "PsAf";
        LanguageEnum[LanguageEnum["Pt"] = 'pt'] = "Pt";
        LanguageEnum[LanguageEnum["PtAo"] = 'pt-ao'] = "PtAo";
        LanguageEnum[LanguageEnum["PtBr"] = 'pt-br'] = "PtBr";
        LanguageEnum[LanguageEnum["PtCh"] = 'pt-ch'] = "PtCh";
        LanguageEnum[LanguageEnum["PtCv"] = 'pt-cv'] = "PtCv";
        LanguageEnum[LanguageEnum["PtGq"] = 'pt-gq'] = "PtGq";
        LanguageEnum[LanguageEnum["PtGw"] = 'pt-gw'] = "PtGw";
        LanguageEnum[LanguageEnum["PtLu"] = 'pt-lu'] = "PtLu";
        LanguageEnum[LanguageEnum["PtMo"] = 'pt-mo'] = "PtMo";
        LanguageEnum[LanguageEnum["PtMz"] = 'pt-mz'] = "PtMz";
        LanguageEnum[LanguageEnum["PtPt"] = 'pt-pt'] = "PtPt";
        LanguageEnum[LanguageEnum["PtSt"] = 'pt-st'] = "PtSt";
        LanguageEnum[LanguageEnum["PtTl"] = 'pt-tl'] = "PtTl";
        LanguageEnum[LanguageEnum["Qu"] = 'qu'] = "Qu";
        LanguageEnum[LanguageEnum["QuBo"] = 'qu-bo'] = "QuBo";
        LanguageEnum[LanguageEnum["QuEc"] = 'qu-ec'] = "QuEc";
        LanguageEnum[LanguageEnum["QuPe"] = 'qu-pe'] = "QuPe";
        LanguageEnum[LanguageEnum["Rm"] = 'rm'] = "Rm";
        LanguageEnum[LanguageEnum["RmCh"] = 'rm-ch'] = "RmCh";
        LanguageEnum[LanguageEnum["Rn"] = 'rn'] = "Rn";
        LanguageEnum[LanguageEnum["RnBi"] = 'rn-bi'] = "RnBi";
        LanguageEnum[LanguageEnum["Ro"] = 'ro'] = "Ro";
        LanguageEnum[LanguageEnum["RoMd"] = 'ro-md'] = "RoMd";
        LanguageEnum[LanguageEnum["RoRo"] = 'ro-ro'] = "RoRo";
        LanguageEnum[LanguageEnum["Rof"] = 'rof'] = "Rof";
        LanguageEnum[LanguageEnum["RofTz"] = 'rof-tz'] = "RofTz";
        LanguageEnum[LanguageEnum["Ru"] = 'ru'] = "Ru";
        LanguageEnum[LanguageEnum["RuBy"] = 'ru-by'] = "RuBy";
        LanguageEnum[LanguageEnum["RuKg"] = 'ru-kg'] = "RuKg";
        LanguageEnum[LanguageEnum["RuKz"] = 'ru-kz'] = "RuKz";
        LanguageEnum[LanguageEnum["RuMd"] = 'ru-md'] = "RuMd";
        LanguageEnum[LanguageEnum["RuRu"] = 'ru-ru'] = "RuRu";
        LanguageEnum[LanguageEnum["RuUa"] = 'ru-ua'] = "RuUa";
        LanguageEnum[LanguageEnum["Rw"] = 'rw'] = "Rw";
        LanguageEnum[LanguageEnum["RwRw"] = 'rw-rw'] = "RwRw";
        LanguageEnum[LanguageEnum["Rwk"] = 'rwk'] = "Rwk";
        LanguageEnum[LanguageEnum["RwkTz"] = 'rwk-tz'] = "RwkTz";
        LanguageEnum[LanguageEnum["Sa"] = 'sa'] = "Sa";
        LanguageEnum[LanguageEnum["Sah"] = 'sah'] = "Sah";
        LanguageEnum[LanguageEnum["SahRu"] = 'sah-ru'] = "SahRu";
        LanguageEnum[LanguageEnum["Saq"] = 'saq'] = "Saq";
        LanguageEnum[LanguageEnum["SaqKe"] = 'saq-ke'] = "SaqKe";
        LanguageEnum[LanguageEnum["Sbp"] = 'sbp'] = "Sbp";
        LanguageEnum[LanguageEnum["SbpTz"] = 'sbp-tz'] = "SbpTz";
        LanguageEnum[LanguageEnum["Sd"] = 'sd'] = "Sd";
        LanguageEnum[LanguageEnum["SdPk"] = 'sd-pk'] = "SdPk";
        LanguageEnum[LanguageEnum["Se"] = 'se'] = "Se";
        LanguageEnum[LanguageEnum["SeFi"] = 'se-fi'] = "SeFi";
        LanguageEnum[LanguageEnum["SeNo"] = 'se-no'] = "SeNo";
        LanguageEnum[LanguageEnum["SeSe"] = 'se-se'] = "SeSe";
        LanguageEnum[LanguageEnum["Seh"] = 'seh'] = "Seh";
        LanguageEnum[LanguageEnum["SehMz"] = 'seh-mz'] = "SehMz";
        LanguageEnum[LanguageEnum["Ses"] = 'ses'] = "Ses";
        LanguageEnum[LanguageEnum["SesMl"] = 'ses-ml'] = "SesMl";
        LanguageEnum[LanguageEnum["Sg"] = 'sg'] = "Sg";
        LanguageEnum[LanguageEnum["SgCf"] = 'sg-cf'] = "SgCf";
        LanguageEnum[LanguageEnum["Shi"] = 'shi'] = "Shi";
        LanguageEnum[LanguageEnum["ShiMa"] = 'shi-ma'] = "ShiMa";
        LanguageEnum[LanguageEnum["Si"] = 'si'] = "Si";
        LanguageEnum[LanguageEnum["SiLk"] = 'si-lk'] = "SiLk";
        LanguageEnum[LanguageEnum["Sk"] = 'sk'] = "Sk";
        LanguageEnum[LanguageEnum["SkSk"] = 'sk-sk'] = "SkSk";
        LanguageEnum[LanguageEnum["Sl"] = 'sl'] = "Sl";
        LanguageEnum[LanguageEnum["SlSi"] = 'sl-si'] = "SlSi";
        LanguageEnum[LanguageEnum["Smn"] = 'smn'] = "Smn";
        LanguageEnum[LanguageEnum["SmnFi"] = 'smn-fi'] = "SmnFi";
        LanguageEnum[LanguageEnum["Sn"] = 'sn'] = "Sn";
        LanguageEnum[LanguageEnum["SnZw"] = 'sn-zw'] = "SnZw";
        LanguageEnum[LanguageEnum["So"] = 'so'] = "So";
        LanguageEnum[LanguageEnum["SoDj"] = 'so-dj'] = "SoDj";
        LanguageEnum[LanguageEnum["SoEt"] = 'so-et'] = "SoEt";
        LanguageEnum[LanguageEnum["SoKe"] = 'so-ke'] = "SoKe";
        LanguageEnum[LanguageEnum["SoSo"] = 'so-so'] = "SoSo";
        LanguageEnum[LanguageEnum["Sq"] = 'sq'] = "Sq";
        LanguageEnum[LanguageEnum["SqAl"] = 'sq-al'] = "SqAl";
        LanguageEnum[LanguageEnum["SqMk"] = 'sq-mk'] = "SqMk";
        LanguageEnum[LanguageEnum["SqXk"] = 'sq-xk'] = "SqXk";
        LanguageEnum[LanguageEnum["Sr"] = 'sr'] = "Sr";
        LanguageEnum[LanguageEnum["SrBa"] = 'sr-ba'] = "SrBa";
        LanguageEnum[LanguageEnum["SrCs"] = 'sr-cs'] = "SrCs";
        LanguageEnum[LanguageEnum["SrMe"] = 'sr-me'] = "SrMe";
        LanguageEnum[LanguageEnum["SrRs"] = 'sr-rs'] = "SrRs";
        LanguageEnum[LanguageEnum["SrXk"] = 'sr-xk'] = "SrXk";
        LanguageEnum[LanguageEnum["Sv"] = 'sv'] = "Sv";
        LanguageEnum[LanguageEnum["SvAx"] = 'sv-ax'] = "SvAx";
        LanguageEnum[LanguageEnum["SvFi"] = 'sv-fi'] = "SvFi";
        LanguageEnum[LanguageEnum["SvSe"] = 'sv-se'] = "SvSe";
        LanguageEnum[LanguageEnum["Sw"] = 'sw'] = "Sw";
        LanguageEnum[LanguageEnum["SwCd"] = 'sw-cd'] = "SwCd";
        LanguageEnum[LanguageEnum["SwKe"] = 'sw-ke'] = "SwKe";
        LanguageEnum[LanguageEnum["SwTz"] = 'sw-tz'] = "SwTz";
        LanguageEnum[LanguageEnum["SwUg"] = 'sw-ug'] = "SwUg";
        LanguageEnum[LanguageEnum["Sy"] = 'sy'] = "Sy";
        LanguageEnum[LanguageEnum["Ta"] = 'ta'] = "Ta";
        LanguageEnum[LanguageEnum["TaIn"] = 'ta-in'] = "TaIn";
        LanguageEnum[LanguageEnum["TaLk"] = 'ta-lk'] = "TaLk";
        LanguageEnum[LanguageEnum["TaMy"] = 'ta-my'] = "TaMy";
        LanguageEnum[LanguageEnum["TaSg"] = 'ta-sg'] = "TaSg";
        LanguageEnum[LanguageEnum["Te"] = 'te'] = "Te";
        LanguageEnum[LanguageEnum["TeIn"] = 'te-in'] = "TeIn";
        LanguageEnum[LanguageEnum["Teo"] = 'teo'] = "Teo";
        LanguageEnum[LanguageEnum["TeoKe"] = 'teo-ke'] = "TeoKe";
        LanguageEnum[LanguageEnum["TeoUg"] = 'teo-ug'] = "TeoUg";
        LanguageEnum[LanguageEnum["Tg"] = 'tg'] = "Tg";
        LanguageEnum[LanguageEnum["TgTj"] = 'tg-tj'] = "TgTj";
        LanguageEnum[LanguageEnum["Th"] = 'th'] = "Th";
        LanguageEnum[LanguageEnum["ThTh"] = 'th-th'] = "ThTh";
        LanguageEnum[LanguageEnum["Ti"] = 'ti'] = "Ti";
        LanguageEnum[LanguageEnum["TiEr"] = 'ti-er'] = "TiEr";
        LanguageEnum[LanguageEnum["TiEt"] = 'ti-et'] = "TiEt";
        LanguageEnum[LanguageEnum["Tk"] = 'tk'] = "Tk";
        LanguageEnum[LanguageEnum["TkTm"] = 'tk-tm'] = "TkTm";
        LanguageEnum[LanguageEnum["To"] = 'to'] = "To";
        LanguageEnum[LanguageEnum["ToTo"] = 'to-to'] = "ToTo";
        LanguageEnum[LanguageEnum["Tr"] = 'tr'] = "Tr";
        LanguageEnum[LanguageEnum["TrCy"] = 'tr-cy'] = "TrCy";
        LanguageEnum[LanguageEnum["TrTr"] = 'tr-tr'] = "TrTr";
        LanguageEnum[LanguageEnum["Tt"] = 'tt'] = "Tt";
        LanguageEnum[LanguageEnum["TtRu"] = 'tt-ru'] = "TtRu";
        LanguageEnum[LanguageEnum["Twq"] = 'twq'] = "Twq";
        LanguageEnum[LanguageEnum["TwqNe"] = 'twq-ne'] = "TwqNe";
        LanguageEnum[LanguageEnum["Tzm"] = 'tzm'] = "Tzm";
        LanguageEnum[LanguageEnum["TzmMa"] = 'tzm-ma'] = "TzmMa";
        LanguageEnum[LanguageEnum["Ug"] = 'ug'] = "Ug";
        LanguageEnum[LanguageEnum["UgCn"] = 'ug-cn'] = "UgCn";
        LanguageEnum[LanguageEnum["Uk"] = 'uk'] = "Uk";
        LanguageEnum[LanguageEnum["UkUa"] = 'uk-ua'] = "UkUa";
        LanguageEnum[LanguageEnum["Ur"] = 'ur'] = "Ur";
        LanguageEnum[LanguageEnum["UrIn"] = 'ur-in'] = "UrIn";
        LanguageEnum[LanguageEnum["UrPk"] = 'ur-pk'] = "UrPk";
        LanguageEnum[LanguageEnum["Uz"] = 'uz'] = "Uz";
        LanguageEnum[LanguageEnum["UzAf"] = 'uz-af'] = "UzAf";
        LanguageEnum[LanguageEnum["UzUz"] = 'uz-uz'] = "UzUz";
        LanguageEnum[LanguageEnum["Vai"] = 'vai'] = "Vai";
        LanguageEnum[LanguageEnum["VaiLr"] = 'vai-lr'] = "VaiLr";
        LanguageEnum[LanguageEnum["Vi"] = 'vi'] = "Vi";
        LanguageEnum[LanguageEnum["ViVn"] = 'vi-vn'] = "ViVn";
        LanguageEnum[LanguageEnum["Vo"] = 'vo'] = "Vo";
        LanguageEnum[LanguageEnum["Vo001"] = 'vo-001'] = "Vo001";
        LanguageEnum[LanguageEnum["Vun"] = 'vun'] = "Vun";
        LanguageEnum[LanguageEnum["VunTz"] = 'vun-tz'] = "VunTz";
        LanguageEnum[LanguageEnum["Wae"] = 'wae'] = "Wae";
        LanguageEnum[LanguageEnum["WaeCh"] = 'wae-ch'] = "WaeCh";
        LanguageEnum[LanguageEnum["Wo"] = 'wo'] = "Wo";
        LanguageEnum[LanguageEnum["WoSn"] = 'wo-sn'] = "WoSn";
        LanguageEnum[LanguageEnum["Xog"] = 'xog'] = "Xog";
        LanguageEnum[LanguageEnum["XogUg"] = 'xog-ug'] = "XogUg";
        LanguageEnum[LanguageEnum["Yav"] = 'yav'] = "Yav";
        LanguageEnum[LanguageEnum["YavCm"] = 'yav-cm'] = "YavCm";
        LanguageEnum[LanguageEnum["Yo"] = 'yo'] = "Yo";
        LanguageEnum[LanguageEnum["YoBj"] = 'yo-bj'] = "YoBj";
        LanguageEnum[LanguageEnum["YoNg"] = 'yo-ng'] = "YoNg";
        LanguageEnum[LanguageEnum["Yue"] = 'yue'] = "Yue";
        LanguageEnum[LanguageEnum["YueCn"] = 'yue-cn'] = "YueCn";
        LanguageEnum[LanguageEnum["YueHk"] = 'yue-hk'] = "YueHk";
        LanguageEnum[LanguageEnum["Zgh"] = 'zgh'] = "Zgh";
        LanguageEnum[LanguageEnum["ZghMa"] = 'zgh-ma'] = "ZghMa";
        LanguageEnum[LanguageEnum["Zh"] = 'zh'] = "Zh";
        LanguageEnum[LanguageEnum["ZhCn"] = 'zh-cn'] = "ZhCn";
        LanguageEnum[LanguageEnum["ZhHk"] = 'zh-hk'] = "ZhHk";
        LanguageEnum[LanguageEnum["ZhMo"] = 'zh-mo'] = "ZhMo";
        LanguageEnum[LanguageEnum["ZhSg"] = 'zh-sg'] = "ZhSg";
        LanguageEnum[LanguageEnum["ZhTw"] = 'zh-tw'] = "ZhTw";
        LanguageEnum[LanguageEnum["ZhHans"] = 'zh-hans'] = "ZhHans";
        LanguageEnum[LanguageEnum["ZhHant"] = 'zh-hant'] = "ZhHant";
        LanguageEnum[LanguageEnum["Zu"] = 'zu'] = "Zu";
        LanguageEnum[LanguageEnum["ZuZa"] = 'zu-za'] = "ZuZa";
    })(LanguageEnum = ContentSearchResult.LanguageEnum || (ContentSearchResult.LanguageEnum = {}));
})(ContentSearchResult = exports.ContentSearchResult || (exports.ContentSearchResult = {}));
