// @hubspot/api-client/lib/codegen/cms/site_search/model/indexedData.js
"use strict";
/**
 * CMS Site Search
 * Use these endpoints for searching content on your HubSpot hosted CMS website(s).
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IndexedData = void 0;
/**
* The indexed data in HubSpot
*/
class IndexedData {
    static getAttributeTypeMap() {
        return IndexedData.attributeTypeMap;
    }
}
exports.IndexedData = IndexedData;
IndexedData.discriminator = undefined;
IndexedData.attributeTypeMap = [
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "IndexedData.TypeEnum"
    },
    {
        "name": "fields",
        "baseName": "fields",
        "type": "{ [key: string]: SearchHitField; }"
    }
];
(function (IndexedData) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["LANDINGPAGE"] = 'LANDING_PAGE'] = "LANDINGPAGE";
        TypeEnum[TypeEnum["BLOGPOST"] = 'BLOG_POST'] = "BLOGPOST";
        TypeEnum[TypeEnum["SITEPAGE"] = 'SITE_PAGE'] = "SITEPAGE";
        TypeEnum[TypeEnum["KNOWLEDGEARTICLE"] = 'KNOWLEDGE_ARTICLE'] = "KNOWLEDGEARTICLE";
        TypeEnum[TypeEnum["LISTINGPAGE"] = 'LISTING_PAGE'] = "LISTINGPAGE";
    })(TypeEnum = IndexedData.TypeEnum || (IndexedData.TypeEnum = {}));
})(IndexedData = exports.IndexedData || (exports.IndexedData = {}));
