// @hubspot/api-client/lib/codegen/crm/owners/model/publicTeam.js
"use strict";
/**
 * CRM Owners
 * HubSpot uses **owners** to assign CRM objects to specific people in your organization. The endpoints described here are used to get a list of the owners that are available for an account. To assign an owner to an object, set the hubspot_owner_id property using the appropriate CRM object update or create a request.  If teams are available for your HubSpot tier, these endpoints will also indicate which team an owner belongs to. Team membership can be one of PRIMARY (default), SECONDARY, or CHILD.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicTeam = void 0;
class PublicTeam {
    static getAttributeTypeMap() {
        return PublicTeam.attributeTypeMap;
    }
}
exports.PublicTeam = PublicTeam;
PublicTeam.discriminator = undefined;
PublicTeam.attributeTypeMap = [
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "membership",
        "baseName": "membership",
        "type": "PublicTeam.MembershipEnum"
    }
];
(function (PublicTeam) {
    let MembershipEnum;
    (function (MembershipEnum) {
        MembershipEnum[MembershipEnum["PRIMARY"] = 'PRIMARY'] = "PRIMARY";
        MembershipEnum[MembershipEnum["SECONDARY"] = 'SECONDARY'] = "SECONDARY";
        MembershipEnum[MembershipEnum["CHILD"] = 'CHILD'] = "CHILD";
    })(MembershipEnum = PublicTeam.MembershipEnum || (PublicTeam.MembershipEnum = {}));
})(PublicTeam = exports.PublicTeam || (exports.PublicTeam = {}));
