// @hubspot/api-client/lib/codegen/cms/url_redirects/model/urlMapping.js
"use strict";
/**
 * URL redirects
 * URL redirect operations
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UrlMapping = void 0;
class UrlMapping {
    static getAttributeTypeMap() {
        return UrlMapping.attributeTypeMap;
    }
}
exports.UrlMapping = UrlMapping;
UrlMapping.discriminator = undefined;
UrlMapping.attributeTypeMap = [
    {
        "name": "id",
        "baseName": "id",
        "type": "number"
    },
    {
        "name": "portalId",
        "baseName": "portalId",
        "type": "number"
    },
    {
        "name": "created",
        "baseName": "created",
        "type": "number"
    },
    {
        "name": "updated",
        "baseName": "updated",
        "type": "number"
    },
    {
        "name": "createdById",
        "baseName": "createdById",
        "type": "number"
    },
    {
        "name": "updatedById",
        "baseName": "updatedById",
        "type": "number"
    },
    {
        "name": "routePrefix",
        "baseName": "routePrefix",
        "type": "string"
    },
    {
        "name": "destination",
        "baseName": "destination",
        "type": "string"
    },
    {
        "name": "redirectStyle",
        "baseName": "redirectStyle",
        "type": "number"
    },
    {
        "name": "contentGroupId",
        "baseName": "contentGroupId",
        "type": "number"
    },
    {
        "name": "isOnlyAfterNotFound",
        "baseName": "isOnlyAfterNotFound",
        "type": "boolean"
    },
    {
        "name": "isRegex",
        "baseName": "isRegex",
        "type": "boolean"
    },
    {
        "name": "isMatchFullUrl",
        "baseName": "isMatchFullUrl",
        "type": "boolean"
    },
    {
        "name": "isMatchQueryString",
        "baseName": "isMatchQueryString",
        "type": "boolean"
    },
    {
        "name": "isPattern",
        "baseName": "isPattern",
        "type": "boolean"
    },
    {
        "name": "isTrailingSlashOptional",
        "baseName": "isTrailingSlashOptional",
        "type": "boolean"
    },
    {
        "name": "isProtocolAgnostic",
        "baseName": "isProtocolAgnostic",
        "type": "boolean"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "precedence",
        "baseName": "precedence",
        "type": "number"
    },
    {
        "name": "deletedAt",
        "baseName": "deletedAt",
        "type": "number"
    },
    {
        "name": "note",
        "baseName": "note",
        "type": "string"
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string"
    },
    {
        "name": "internallyCreated",
        "baseName": "internallyCreated",
        "type": "boolean"
    },
    {
        "name": "cosObjectType",
        "baseName": "cosObjectType",
        "type": "UrlMapping.CosObjectTypeEnum"
    },
    {
        "name": "cdnPurgeEmbargoTime",
        "baseName": "cdnPurgeEmbargoTime",
        "type": "number"
    }
];
(function (UrlMapping) {
    let CosObjectTypeEnum;
    (function (CosObjectTypeEnum) {
        CosObjectTypeEnum[CosObjectTypeEnum["CONTENT"] = 'CONTENT'] = "CONTENT";
        CosObjectTypeEnum[CosObjectTypeEnum["EXTENSIONRESOURCE"] = 'EXTENSION_RESOURCE'] = "EXTENSIONRESOURCE";
        CosObjectTypeEnum[CosObjectTypeEnum["LAYOUT"] = 'LAYOUT'] = "LAYOUT";
        CosObjectTypeEnum[CosObjectTypeEnum["CUSTOMWIDGET"] = 'CUSTOM_WIDGET'] = "CUSTOMWIDGET";
        CosObjectTypeEnum[CosObjectTypeEnum["WIDGET"] = 'WIDGET'] = "WIDGET";
        CosObjectTypeEnum[CosObjectTypeEnum["FORM"] = 'FORM'] = "FORM";
        CosObjectTypeEnum[CosObjectTypeEnum["PLACEMENT"] = 'PLACEMENT'] = "PLACEMENT";
        CosObjectTypeEnum[CosObjectTypeEnum["IMAGE"] = 'IMAGE'] = "IMAGE";
        CosObjectTypeEnum[CosObjectTypeEnum["DOMAINSETTINGS"] = 'DOMAIN_SETTINGS'] = "DOMAINSETTINGS";
        CosObjectTypeEnum[CosObjectTypeEnum["SITESETTINGS"] = 'SITE_SETTINGS'] = "SITESETTINGS";
        CosObjectTypeEnum[CosObjectTypeEnum["EMAILADDRESS"] = 'EMAIL_ADDRESS'] = "EMAILADDRESS";
        CosObjectTypeEnum[CosObjectTypeEnum["WORKFLOW"] = 'WORKFLOW'] = "WORKFLOW";
        CosObjectTypeEnum[CosObjectTypeEnum["HUBDBTABLE"] = 'HUBDB_TABLE'] = "HUBDBTABLE";
        CosObjectTypeEnum[CosObjectTypeEnum["REDIRECTURL"] = 'REDIRECT_URL'] = "REDIRECTURL";
        CosObjectTypeEnum[CosObjectTypeEnum["DESIGNFOLDER"] = 'DESIGN_FOLDER'] = "DESIGNFOLDER";
        CosObjectTypeEnum[CosObjectTypeEnum["SITEMAP"] = 'SITE_MAP'] = "SITEMAP";
        CosObjectTypeEnum[CosObjectTypeEnum["DOMAIN"] = 'DOMAIN'] = "DOMAIN";
        CosObjectTypeEnum[CosObjectTypeEnum["BLOG"] = 'BLOG'] = "BLOG";
        CosObjectTypeEnum[CosObjectTypeEnum["FILE"] = 'FILE'] = "FILE";
        CosObjectTypeEnum[CosObjectTypeEnum["FOLDER"] = 'FOLDER'] = "FOLDER";
        CosObjectTypeEnum[CosObjectTypeEnum["SITEMENU"] = 'SITE_MENU'] = "SITEMENU";
        CosObjectTypeEnum[CosObjectTypeEnum["THEME"] = 'THEME'] = "THEME";
        CosObjectTypeEnum[CosObjectTypeEnum["CONTENTGROUP"] = 'CONTENT_GROUP'] = "CONTENTGROUP";
        CosObjectTypeEnum[CosObjectTypeEnum["FOLLOWME"] = 'FOLLOW_ME'] = "FOLLOWME";
        CosObjectTypeEnum[CosObjectTypeEnum["KNOWLEDGEBASE"] = 'KNOWLEDGE_BASE'] = "KNOWLEDGEBASE";
        CosObjectTypeEnum[CosObjectTypeEnum["LISTMEMBERSHIP"] = 'LIST_MEMBERSHIP'] = "LISTMEMBERSHIP";
        CosObjectTypeEnum[CosObjectTypeEnum["CONTACTMEMBERSHIP"] = 'CONTACT_MEMBERSHIP'] = "CONTACTMEMBERSHIP";
        CosObjectTypeEnum[CosObjectTypeEnum["PASSWORDPROTECTED"] = 'PASSWORD_PROTECTED'] = "PASSWORDPROTECTED";
        CosObjectTypeEnum[CosObjectTypeEnum["UNRESTRICTEDACCESS"] = 'UNRESTRICTED_ACCESS'] = "UNRESTRICTEDACCESS";
        CosObjectTypeEnum[CosObjectTypeEnum["MARKETPLACELISTING"] = 'MARKETPLACE_LISTING'] = "MARKETPLACELISTING";
        CosObjectTypeEnum[CosObjectTypeEnum["LAYOUTSECTION"] = 'LAYOUT_SECTION'] = "LAYOUTSECTION";
        CosObjectTypeEnum[CosObjectTypeEnum["THEMESETTINGS"] = 'THEME_SETTINGS'] = "THEMESETTINGS";
        CosObjectTypeEnum[CosObjectTypeEnum["VIDEOPLAYER"] = 'VIDEO_PLAYER'] = "VIDEOPLAYER";
        CosObjectTypeEnum[CosObjectTypeEnum["URLMAPPING"] = 'URL_MAPPING'] = "URLMAPPING";
        CosObjectTypeEnum[CosObjectTypeEnum["KNOWLEDGECATEGORY"] = 'KNOWLEDGE_CATEGORY'] = "KNOWLEDGECATEGORY";
        CosObjectTypeEnum[CosObjectTypeEnum["KNOWLEDGEHOMEPAGECATEGORY"] = 'KNOWLEDGE_HOMEPAGE_CATEGORY'] = "KNOWLEDGEHOMEPAGECATEGORY";
        CosObjectTypeEnum[CosObjectTypeEnum["RAWASSET"] = 'RAW_ASSET'] = "RAWASSET";
        CosObjectTypeEnum[CosObjectTypeEnum["GLOBALCONTENT"] = 'GLOBAL_CONTENT'] = "GLOBALCONTENT";
        CosObjectTypeEnum[CosObjectTypeEnum["HUBDBTABLEROW"] = 'HUBDB_TABLE_ROW'] = "HUBDBTABLEROW";
        CosObjectTypeEnum[CosObjectTypeEnum["BLOGAUTHOR"] = 'BLOG_AUTHOR'] = "BLOGAUTHOR";
        CosObjectTypeEnum[CosObjectTypeEnum["SERVERLESSFUNCTION"] = 'SERVERLESS_FUNCTION'] = "SERVERLESSFUNCTION";
        CosObjectTypeEnum[CosObjectTypeEnum["KNOWLEDGECATEGORYTRANSLATION"] = 'KNOWLEDGE_CATEGORY_TRANSLATION'] = "KNOWLEDGECATEGORYTRANSLATION";
    })(CosObjectTypeEnum = UrlMapping.CosObjectTypeEnum || (UrlMapping.CosObjectTypeEnum = {}));
})(UrlMapping = exports.UrlMapping || (exports.UrlMapping = {}));
