// @hubspot/api-client/lib/codegen/webhooks/model/batchResponseSubscriptionResponse.js
"use strict";
/**
 * Webhooks API
 * Provides a way for apps to subscribe to certain change events in HubSpot. Once configured, apps will receive event payloads containing details about the changes at a specified target URL. There can only be one target URL for receiving event notifications per app.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseSubscriptionResponse = void 0;
class BatchResponseSubscriptionResponse {
    static getAttributeTypeMap() {
        return BatchResponseSubscriptionResponse.attributeTypeMap;
    }
}
exports.BatchResponseSubscriptionResponse = BatchResponseSubscriptionResponse;
BatchResponseSubscriptionResponse.discriminator = undefined;
BatchResponseSubscriptionResponse.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseSubscriptionResponse.StatusEnum"
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<SubscriptionResponse>"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date"
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }"
    }
];
(function (BatchResponseSubscriptionResponse) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["PENDING"] = 'PENDING'] = "PENDING";
        StatusEnum[StatusEnum["PROCESSING"] = 'PROCESSING'] = "PROCESSING";
        StatusEnum[StatusEnum["CANCELED"] = 'CANCELED'] = "CANCELED";
        StatusEnum[StatusEnum["COMPLETE"] = 'COMPLETE'] = "COMPLETE";
    })(StatusEnum = BatchResponseSubscriptionResponse.StatusEnum || (BatchResponseSubscriptionResponse.StatusEnum = {}));
})(BatchResponseSubscriptionResponse = exports.BatchResponseSubscriptionResponse || (exports.BatchResponseSubscriptionResponse = {}));
