// @hubspot/api-client/lib/codegen/webhooks/model/batchResponseSubscriptionResponseWithErrors.js
"use strict";
/**
 * Webhooks API
 * Provides a way for apps to subscribe to certain change events in HubSpot. Once configured, apps will receive event payloads containing details about the changes at a specified target URL. There can only be one target URL for receiving event notifications per app.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseSubscriptionResponseWithErrors = void 0;
class BatchResponseSubscriptionResponseWithErrors {
    static getAttributeTypeMap() {
        return BatchResponseSubscriptionResponseWithErrors.attributeTypeMap;
    }
}
exports.BatchResponseSubscriptionResponseWithErrors = BatchResponseSubscriptionResponseWithErrors;
BatchResponseSubscriptionResponseWithErrors.discriminator = undefined;
BatchResponseSubscriptionResponseWithErrors.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseSubscriptionResponseWithErrors.StatusEnum"
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<SubscriptionResponse>"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number"
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date"
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }"
    }
];
(function (BatchResponseSubscriptionResponseWithErrors) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["PENDING"] = 'PENDING'] = "PENDING";
        StatusEnum[StatusEnum["PROCESSING"] = 'PROCESSING'] = "PROCESSING";
        StatusEnum[StatusEnum["CANCELED"] = 'CANCELED'] = "CANCELED";
        StatusEnum[StatusEnum["COMPLETE"] = 'COMPLETE'] = "COMPLETE";
    })(StatusEnum = BatchResponseSubscriptionResponseWithErrors.StatusEnum || (BatchResponseSubscriptionResponseWithErrors.StatusEnum = {}));
})(BatchResponseSubscriptionResponseWithErrors = exports.BatchResponseSubscriptionResponseWithErrors || (exports.BatchResponseSubscriptionResponseWithErrors = {}));
