// @hubspot/api-client/lib/codegen/webhooks/model/subscriptionResponse.js
"use strict";
/**
 * Webhooks API
 * Provides a way for apps to subscribe to certain change events in HubSpot. Once configured, apps will receive event payloads containing details about the changes at a specified target URL. There can only be one target URL for receiving event notifications per app.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionResponse = void 0;
/**
* Complete details for an event subscription.
*/
class SubscriptionResponse {
    static getAttributeTypeMap() {
        return SubscriptionResponse.attributeTypeMap;
    }
}
exports.SubscriptionResponse = SubscriptionResponse;
SubscriptionResponse.discriminator = undefined;
SubscriptionResponse.attributeTypeMap = [
    {
        "name": "eventType",
        "baseName": "eventType",
        "type": "SubscriptionResponse.EventTypeEnum"
    },
    {
        "name": "propertyName",
        "baseName": "propertyName",
        "type": "string"
    },
    {
        "name": "active",
        "baseName": "active",
        "type": "boolean"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date"
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "Date"
    }
];
(function (SubscriptionResponse) {
    let EventTypeEnum;
    (function (EventTypeEnum) {
        EventTypeEnum[EventTypeEnum["ContactPropertyChange"] = 'contact.propertyChange'] = "ContactPropertyChange";
        EventTypeEnum[EventTypeEnum["CompanyPropertyChange"] = 'company.propertyChange'] = "CompanyPropertyChange";
        EventTypeEnum[EventTypeEnum["DealPropertyChange"] = 'deal.propertyChange'] = "DealPropertyChange";
        EventTypeEnum[EventTypeEnum["ContactCreation"] = 'contact.creation'] = "ContactCreation";
        EventTypeEnum[EventTypeEnum["ContactDeletion"] = 'contact.deletion'] = "ContactDeletion";
        EventTypeEnum[EventTypeEnum["ContactPrivacyDeletion"] = 'contact.privacyDeletion'] = "ContactPrivacyDeletion";
        EventTypeEnum[EventTypeEnum["CompanyCreation"] = 'company.creation'] = "CompanyCreation";
        EventTypeEnum[EventTypeEnum["CompanyDeletion"] = 'company.deletion'] = "CompanyDeletion";
        EventTypeEnum[EventTypeEnum["DealCreation"] = 'deal.creation'] = "DealCreation";
        EventTypeEnum[EventTypeEnum["DealDeletion"] = 'deal.deletion'] = "DealDeletion";
    })(EventTypeEnum = SubscriptionResponse.EventTypeEnum || (SubscriptionResponse.EventTypeEnum = {}));
})(SubscriptionResponse = exports.SubscriptionResponse || (exports.SubscriptionResponse = {}));
