// @hubspot/api-client/lib/codegen/webhooks/model/throttlingSettings.js
"use strict";
/**
 * Webhooks API
 * Provides a way for apps to subscribe to certain change events in HubSpot. Once configured, apps will receive event payloads containing details about the changes at a specified target URL. There can only be one target URL for receiving event notifications per app.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThrottlingSettings = void 0;
/**
* Configuration details for webhook throttling.
*/
class ThrottlingSettings {
    static getAttributeTypeMap() {
        return ThrottlingSettings.attributeTypeMap;
    }
}
exports.ThrottlingSettings = ThrottlingSettings;
ThrottlingSettings.discriminator = undefined;
ThrottlingSettings.attributeTypeMap = [
    {
        "name": "maxConcurrentRequests",
        "baseName": "maxConcurrentRequests",
        "type": "number"
    },
    {
        "name": "period",
        "baseName": "period",
        "type": "ThrottlingSettings.PeriodEnum"
    }
];
(function (ThrottlingSettings) {
    let PeriodEnum;
    (function (PeriodEnum) {
        PeriodEnum[PeriodEnum["SECONDLY"] = 'SECONDLY'] = "SECONDLY";
        PeriodEnum[PeriodEnum["ROLLINGMINUTE"] = 'ROLLING_MINUTE'] = "ROLLINGMINUTE";
    })(PeriodEnum = ThrottlingSettings.PeriodEnum || (ThrottlingSettings.PeriodEnum = {}));
})(ThrottlingSettings = exports.ThrottlingSettings || (exports.ThrottlingSettings = {}));
