// @hubspot/api-client/lib/codegen/crm/imports/model/publicImportError.js
"use strict";
/**
 * CRM Imports
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicImportError = void 0;
class PublicImportError {
    static getAttributeTypeMap() {
        return PublicImportError.attributeTypeMap;
    }
}
exports.PublicImportError = PublicImportError;
PublicImportError.discriminator = undefined;
PublicImportError.attributeTypeMap = [
    {
        "name": "errorType",
        "baseName": "errorType",
        "type": "PublicImportError.ErrorTypeEnum"
    },
    {
        "name": "objectType",
        "baseName": "objectType",
        "type": "PublicImportError.ObjectTypeEnum"
    },
    {
        "name": "invalidValue",
        "baseName": "invalidValue",
        "type": "string"
    },
    {
        "name": "extraContext",
        "baseName": "extraContext",
        "type": "string"
    },
    {
        "name": "objectTypeId",
        "baseName": "objectTypeId",
        "type": "string"
    },
    {
        "name": "knownColumnNumber",
        "baseName": "knownColumnNumber",
        "type": "number"
    },
    {
        "name": "sourceData",
        "baseName": "sourceData",
        "type": "ImportRowCore"
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "number"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    }
];
(function (PublicImportError) {
    let ErrorTypeEnum;
    (function (ErrorTypeEnum) {
        ErrorTypeEnum[ErrorTypeEnum["INCORRECTNUMBEROFCOLUMNS"] = 'INCORRECT_NUMBER_OF_COLUMNS'] = "INCORRECTNUMBEROFCOLUMNS";
        ErrorTypeEnum[ErrorTypeEnum["INVALIDOBJECTID"] = 'INVALID_OBJECT_ID'] = "INVALIDOBJECTID";
        ErrorTypeEnum[ErrorTypeEnum["INVALIDASSOCIATIONIDENTIFIER"] = 'INVALID_ASSOCIATION_IDENTIFIER'] = "INVALIDASSOCIATIONIDENTIFIER";
        ErrorTypeEnum[ErrorTypeEnum["NOOBJECTIDFROMASSOCIATIONIDENTIFIER"] = 'NO_OBJECT_ID_FROM_ASSOCIATION_IDENTIFIER'] = "NOOBJECTIDFROMASSOCIATIONIDENTIFIER";
        ErrorTypeEnum[ErrorTypeEnum["MULTIPLECOMPANIESWITHTHISDOMAIN"] = 'MULTIPLE_COMPANIES_WITH_THIS_DOMAIN'] = "MULTIPLECOMPANIESWITHTHISDOMAIN";
        ErrorTypeEnum[ErrorTypeEnum["PROPERTYDEFINITIONNOTFOUND"] = 'PROPERTY_DEFINITION_NOT_FOUND'] = "PROPERTYDEFINITIONNOTFOUND";
        ErrorTypeEnum[ErrorTypeEnum["PROPERTYVALUENOTFOUND"] = 'PROPERTY_VALUE_NOT_FOUND'] = "PROPERTYVALUENOTFOUND";
        ErrorTypeEnum[ErrorTypeEnum["COULDNOTFINDOWNER"] = 'COULD_NOT_FIND_OWNER'] = "COULDNOTFINDOWNER";
        ErrorTypeEnum[ErrorTypeEnum["MULTIPLEOWNERSFOUND"] = 'MULTIPLE_OWNERS_FOUND'] = "MULTIPLEOWNERSFOUND";
        ErrorTypeEnum[ErrorTypeEnum["COULDNOTPARSENUMBER"] = 'COULD_NOT_PARSE_NUMBER'] = "COULDNOTPARSENUMBER";
        ErrorTypeEnum[ErrorTypeEnum["COULDNOTPARSEDATE"] = 'COULD_NOT_PARSE_DATE'] = "COULDNOTPARSEDATE";
        ErrorTypeEnum[ErrorTypeEnum["COULDNOTPARSETERM"] = 'COULD_NOT_PARSE_TERM'] = "COULDNOTPARSETERM";
        ErrorTypeEnum[ErrorTypeEnum["OUTSIDEVALIDTIMERANGE"] = 'OUTSIDE_VALID_TIME_RANGE'] = "OUTSIDEVALIDTIMERANGE";
        ErrorTypeEnum[ErrorTypeEnum["OUTSIDEVALIDTERMRANGE"] = 'OUTSIDE_VALID_TERM_RANGE'] = "OUTSIDEVALIDTERMRANGE";
        ErrorTypeEnum[ErrorTypeEnum["COULDNOTPARSEROW"] = 'COULD_NOT_PARSE_ROW'] = "COULDNOTPARSEROW";
        ErrorTypeEnum[ErrorTypeEnum["INVALIDENUMERATIONOPTION"] = 'INVALID_ENUMERATION_OPTION'] = "INVALIDENUMERATIONOPTION";
        ErrorTypeEnum[ErrorTypeEnum["AMBIGUOUSENUMERATIONOPTION"] = 'AMBIGUOUS_ENUMERATION_OPTION'] = "AMBIGUOUSENUMERATIONOPTION";
        ErrorTypeEnum[ErrorTypeEnum["FAILEDVALIDATION"] = 'FAILED_VALIDATION'] = "FAILEDVALIDATION";
        ErrorTypeEnum[ErrorTypeEnum["FAILEDTOCREATEASSOCIATION"] = 'FAILED_TO_CREATE_ASSOCIATION'] = "FAILEDTOCREATEASSOCIATION";
        ErrorTypeEnum[ErrorTypeEnum["FILENOTFOUND"] = 'FILE_NOT_FOUND'] = "FILENOTFOUND";
        ErrorTypeEnum[ErrorTypeEnum["INVALIDCOLUMNCONFIGURATION"] = 'INVALID_COLUMN_CONFIGURATION'] = "INVALIDCOLUMNCONFIGURATION";
        ErrorTypeEnum[ErrorTypeEnum["INVALIDFILETYPE"] = 'INVALID_FILE_TYPE'] = "INVALIDFILETYPE";
        ErrorTypeEnum[ErrorTypeEnum["INVALIDSPREADSHEET"] = 'INVALID_SPREADSHEET'] = "INVALIDSPREADSHEET";
        ErrorTypeEnum[ErrorTypeEnum["INVALIDSHEETCOUNT"] = 'INVALID_SHEET_COUNT'] = "INVALIDSHEETCOUNT";
        ErrorTypeEnum[ErrorTypeEnum["FAILEDTOPROCESSOBJECTWITHEMPTYPROPERTYVALUES"] = 'FAILED_TO_PROCESS_OBJECT_WITH_EMPTY_PROPERTY_VALUES'] = "FAILEDTOPROCESSOBJECTWITHEMPTYPROPERTYVALUES";
        ErrorTypeEnum[ErrorTypeEnum["UNKNOWNBADREQUEST"] = 'UNKNOWN_BAD_REQUEST'] = "UNKNOWNBADREQUEST";
        ErrorTypeEnum[ErrorTypeEnum["GDPRBLACKLISTEDEMAIL"] = 'GDPR_BLACKLISTED_EMAIL'] = "GDPRBLACKLISTEDEMAIL";
        ErrorTypeEnum[ErrorTypeEnum["DUPLICATEASSOCIATIONID"] = 'DUPLICATE_ASSOCIATION_ID'] = "DUPLICATEASSOCIATIONID";
        ErrorTypeEnum[ErrorTypeEnum["LIMITEXCEEDED"] = 'LIMIT_EXCEEDED'] = "LIMITEXCEEDED";
        ErrorTypeEnum[ErrorTypeEnum["INVALIDALTERNATEID"] = 'INVALID_ALTERNATE_ID'] = "INVALIDALTERNATEID";
        ErrorTypeEnum[ErrorTypeEnum["INVALIDEMAIL"] = 'INVALID_EMAIL'] = "INVALIDEMAIL";
        ErrorTypeEnum[ErrorTypeEnum["INVALIDDOMAIN"] = 'INVALID_DOMAIN'] = "INVALIDDOMAIN";
        ErrorTypeEnum[ErrorTypeEnum["DUPLICATEROWCONTENT"] = 'DUPLICATE_ROW_CONTENT'] = "DUPLICATEROWCONTENT";
        ErrorTypeEnum[ErrorTypeEnum["INVALIDNUMBERSIZE"] = 'INVALID_NUMBER_SIZE'] = "INVALIDNUMBERSIZE";
        ErrorTypeEnum[ErrorTypeEnum["UNKNOWNERROR"] = 'UNKNOWN_ERROR'] = "UNKNOWNERROR";
        ErrorTypeEnum[ErrorTypeEnum["FAILEDTOOPTOUTCONTACT"] = 'FAILED_TO_OPT_OUT_CONTACT'] = "FAILEDTOOPTOUTCONTACT";
        ErrorTypeEnum[ErrorTypeEnum["INVALIDREQUIREDPROPERTY"] = 'INVALID_REQUIRED_PROPERTY'] = "INVALIDREQUIREDPROPERTY";
        ErrorTypeEnum[ErrorTypeEnum["DUPLICATEALTERNATEID"] = 'DUPLICATE_ALTERNATE_ID'] = "DUPLICATEALTERNATEID";
        ErrorTypeEnum[ErrorTypeEnum["DUPLICATEOBJECTID"] = 'DUPLICATE_OBJECT_ID'] = "DUPLICATEOBJECTID";
        ErrorTypeEnum[ErrorTypeEnum["DUPLICATEUNIQUEPROPERTYVALUE"] = 'DUPLICATE_UNIQUE_PROPERTY_VALUE'] = "DUPLICATEUNIQUEPROPERTYVALUE";
    })(ErrorTypeEnum = PublicImportError.ErrorTypeEnum || (PublicImportError.ErrorTypeEnum = {}));
    let ObjectTypeEnum;
    (function (ObjectTypeEnum) {
        ObjectTypeEnum[ObjectTypeEnum["CONTACT"] = 'CONTACT'] = "CONTACT";
        ObjectTypeEnum[ObjectTypeEnum["COMPANY"] = 'COMPANY'] = "COMPANY";
        ObjectTypeEnum[ObjectTypeEnum["DEAL"] = 'DEAL'] = "DEAL";
        ObjectTypeEnum[ObjectTypeEnum["ENGAGEMENT"] = 'ENGAGEMENT'] = "ENGAGEMENT";
        ObjectTypeEnum[ObjectTypeEnum["TICKET"] = 'TICKET'] = "TICKET";
        ObjectTypeEnum[ObjectTypeEnum["OWNER"] = 'OWNER'] = "OWNER";
        ObjectTypeEnum[ObjectTypeEnum["PRODUCT"] = 'PRODUCT'] = "PRODUCT";
        ObjectTypeEnum[ObjectTypeEnum["LINEITEM"] = 'LINE_ITEM'] = "LINEITEM";
        ObjectTypeEnum[ObjectTypeEnum["BETDELIVERABLESERVICE"] = 'BET_DELIVERABLE_SERVICE'] = "BETDELIVERABLESERVICE";
        ObjectTypeEnum[ObjectTypeEnum["CONTENT"] = 'CONTENT'] = "CONTENT";
        ObjectTypeEnum[ObjectTypeEnum["CONVERSATION"] = 'CONVERSATION'] = "CONVERSATION";
        ObjectTypeEnum[ObjectTypeEnum["BETALERT"] = 'BET_ALERT'] = "BETALERT";
        ObjectTypeEnum[ObjectTypeEnum["PORTAL"] = 'PORTAL'] = "PORTAL";
        ObjectTypeEnum[ObjectTypeEnum["QUOTE"] = 'QUOTE'] = "QUOTE";
        ObjectTypeEnum[ObjectTypeEnum["FORMSUBMISSIONINBOUNDDB"] = 'FORM_SUBMISSION_INBOUNDDB'] = "FORMSUBMISSIONINBOUNDDB";
        ObjectTypeEnum[ObjectTypeEnum["QUOTA"] = 'QUOTA'] = "QUOTA";
        ObjectTypeEnum[ObjectTypeEnum["UNSUBSCRIBE"] = 'UNSUBSCRIBE'] = "UNSUBSCRIBE";
        ObjectTypeEnum[ObjectTypeEnum["COMMUNICATION"] = 'COMMUNICATION'] = "COMMUNICATION";
        ObjectTypeEnum[ObjectTypeEnum["FEEDBACKSUBMISSION"] = 'FEEDBACK_SUBMISSION'] = "FEEDBACKSUBMISSION";
        ObjectTypeEnum[ObjectTypeEnum["ATTRIBUTION"] = 'ATTRIBUTION'] = "ATTRIBUTION";
        ObjectTypeEnum[ObjectTypeEnum["SALESFORCESYNCERROR"] = 'SALESFORCE_SYNC_ERROR'] = "SALESFORCESYNCERROR";
        ObjectTypeEnum[ObjectTypeEnum["RESTORABLECRMOBJECT"] = 'RESTORABLE_CRM_OBJECT'] = "RESTORABLECRMOBJECT";
        ObjectTypeEnum[ObjectTypeEnum["HUB"] = 'HUB'] = "HUB";
        ObjectTypeEnum[ObjectTypeEnum["LANDINGPAGE"] = 'LANDING_PAGE'] = "LANDINGPAGE";
        ObjectTypeEnum[ObjectTypeEnum["PRODUCTORFOLDER"] = 'PRODUCT_OR_FOLDER'] = "PRODUCTORFOLDER";
        ObjectTypeEnum[ObjectTypeEnum["TASK"] = 'TASK'] = "TASK";
        ObjectTypeEnum[ObjectTypeEnum["FORM"] = 'FORM'] = "FORM";
        ObjectTypeEnum[ObjectTypeEnum["MARKETINGEMAIL"] = 'MARKETING_EMAIL'] = "MARKETINGEMAIL";
        ObjectTypeEnum[ObjectTypeEnum["ADACCOUNT"] = 'AD_ACCOUNT'] = "ADACCOUNT";
        ObjectTypeEnum[ObjectTypeEnum["ADCAMPAIGN"] = 'AD_CAMPAIGN'] = "ADCAMPAIGN";
        ObjectTypeEnum[ObjectTypeEnum["ADGROUP"] = 'AD_GROUP'] = "ADGROUP";
        ObjectTypeEnum[ObjectTypeEnum["AD"] = 'AD'] = "AD";
        ObjectTypeEnum[ObjectTypeEnum["KEYWORD"] = 'KEYWORD'] = "KEYWORD";
        ObjectTypeEnum[ObjectTypeEnum["CAMPAIGN"] = 'CAMPAIGN'] = "CAMPAIGN";
        ObjectTypeEnum[ObjectTypeEnum["SOCIALCHANNEL"] = 'SOCIAL_CHANNEL'] = "SOCIALCHANNEL";
        ObjectTypeEnum[ObjectTypeEnum["SOCIALPOST"] = 'SOCIAL_POST'] = "SOCIALPOST";
        ObjectTypeEnum[ObjectTypeEnum["SITEPAGE"] = 'SITE_PAGE'] = "SITEPAGE";
        ObjectTypeEnum[ObjectTypeEnum["BLOGPOST"] = 'BLOG_POST'] = "BLOGPOST";
        ObjectTypeEnum[ObjectTypeEnum["IMPORT"] = 'IMPORT'] = "IMPORT";
        ObjectTypeEnum[ObjectTypeEnum["EXPORT"] = 'EXPORT'] = "EXPORT";
        ObjectTypeEnum[ObjectTypeEnum["CTA"] = 'CTA'] = "CTA";
        ObjectTypeEnum[ObjectTypeEnum["TASKTEMPLATE"] = 'TASK_TEMPLATE'] = "TASKTEMPLATE";
        ObjectTypeEnum[ObjectTypeEnum["AUTOMATIONPLATFORMFLOW"] = 'AUTOMATION_PLATFORM_FLOW'] = "AUTOMATIONPLATFORMFLOW";
        ObjectTypeEnum[ObjectTypeEnum["OBJECTLIST"] = 'OBJECT_LIST'] = "OBJECTLIST";
        ObjectTypeEnum[ObjectTypeEnum["NOTE"] = 'NOTE'] = "NOTE";
        ObjectTypeEnum[ObjectTypeEnum["MEETINGEVENT"] = 'MEETING_EVENT'] = "MEETINGEVENT";
        ObjectTypeEnum[ObjectTypeEnum["CALL"] = 'CALL'] = "CALL";
        ObjectTypeEnum[ObjectTypeEnum["EMAIL"] = 'EMAIL'] = "EMAIL";
        ObjectTypeEnum[ObjectTypeEnum["PUBLISHINGTASK"] = 'PUBLISHING_TASK'] = "PUBLISHINGTASK";
        ObjectTypeEnum[ObjectTypeEnum["CONVERSATIONSESSION"] = 'CONVERSATION_SESSION'] = "CONVERSATIONSESSION";
        ObjectTypeEnum[ObjectTypeEnum["CONTACTCREATEATTRIBUTION"] = 'CONTACT_CREATE_ATTRIBUTION'] = "CONTACTCREATEATTRIBUTION";
        ObjectTypeEnum[ObjectTypeEnum["INVOICE"] = 'INVOICE'] = "INVOICE";
        ObjectTypeEnum[ObjectTypeEnum["MARKETINGEVENT"] = 'MARKETING_EVENT'] = "MARKETINGEVENT";
        ObjectTypeEnum[ObjectTypeEnum["CONVERSATIONINBOX"] = 'CONVERSATION_INBOX'] = "CONVERSATIONINBOX";
        ObjectTypeEnum[ObjectTypeEnum["CHATFLOW"] = 'CHATFLOW'] = "CHATFLOW";
        ObjectTypeEnum[ObjectTypeEnum["MEDIABRIDGE"] = 'MEDIA_BRIDGE'] = "MEDIABRIDGE";
        ObjectTypeEnum[ObjectTypeEnum["SEQUENCE"] = 'SEQUENCE'] = "SEQUENCE";
        ObjectTypeEnum[ObjectTypeEnum["SEQUENCESTEP"] = 'SEQUENCE_STEP'] = "SEQUENCESTEP";
        ObjectTypeEnum[ObjectTypeEnum["FORECAST"] = 'FORECAST'] = "FORECAST";
        ObjectTypeEnum[ObjectTypeEnum["SNIPPET"] = 'SNIPPET'] = "SNIPPET";
        ObjectTypeEnum[ObjectTypeEnum["TEMPLATE"] = 'TEMPLATE'] = "TEMPLATE";
        ObjectTypeEnum[ObjectTypeEnum["DEALCREATEATTRIBUTION"] = 'DEAL_CREATE_ATTRIBUTION'] = "DEALCREATEATTRIBUTION";
        ObjectTypeEnum[ObjectTypeEnum["QUOTETEMPLATE"] = 'QUOTE_TEMPLATE'] = "QUOTETEMPLATE";
        ObjectTypeEnum[ObjectTypeEnum["QUOTEMODULE"] = 'QUOTE_MODULE'] = "QUOTEMODULE";
        ObjectTypeEnum[ObjectTypeEnum["QUOTEMODULEFIELD"] = 'QUOTE_MODULE_FIELD'] = "QUOTEMODULEFIELD";
        ObjectTypeEnum[ObjectTypeEnum["QUOTEFIELD"] = 'QUOTE_FIELD'] = "QUOTEFIELD";
        ObjectTypeEnum[ObjectTypeEnum["SEQUENCEENROLLMENT"] = 'SEQUENCE_ENROLLMENT'] = "SEQUENCEENROLLMENT";
        ObjectTypeEnum[ObjectTypeEnum["SUBSCRIPTION"] = 'SUBSCRIPTION'] = "SUBSCRIPTION";
        ObjectTypeEnum[ObjectTypeEnum["ACCEPTANCETEST"] = 'ACCEPTANCE_TEST'] = "ACCEPTANCETEST";
        ObjectTypeEnum[ObjectTypeEnum["SOCIALBROADCAST"] = 'SOCIAL_BROADCAST'] = "SOCIALBROADCAST";
        ObjectTypeEnum[ObjectTypeEnum["DEALSPLIT"] = 'DEAL_SPLIT'] = "DEALSPLIT";
        ObjectTypeEnum[ObjectTypeEnum["DEALREGISTRATION"] = 'DEAL_REGISTRATION'] = "DEALREGISTRATION";
        ObjectTypeEnum[ObjectTypeEnum["GOALTARGET"] = 'GOAL_TARGET'] = "GOALTARGET";
        ObjectTypeEnum[ObjectTypeEnum["GOALTARGETGROUP"] = 'GOAL_TARGET_GROUP'] = "GOALTARGETGROUP";
        ObjectTypeEnum[ObjectTypeEnum["PORTALOBJECTSYNCMESSAGE"] = 'PORTAL_OBJECT_SYNC_MESSAGE'] = "PORTALOBJECTSYNCMESSAGE";
        ObjectTypeEnum[ObjectTypeEnum["FILEMANAGERFILE"] = 'FILE_MANAGER_FILE'] = "FILEMANAGERFILE";
        ObjectTypeEnum[ObjectTypeEnum["FILEMANAGERFOLDER"] = 'FILE_MANAGER_FOLDER'] = "FILEMANAGERFOLDER";
        ObjectTypeEnum[ObjectTypeEnum["SEQUENCESTEPENROLLMENT"] = 'SEQUENCE_STEP_ENROLLMENT'] = "SEQUENCESTEPENROLLMENT";
        ObjectTypeEnum[ObjectTypeEnum["APPROVAL"] = 'APPROVAL'] = "APPROVAL";
        ObjectTypeEnum[ObjectTypeEnum["APPROVALSTEP"] = 'APPROVAL_STEP'] = "APPROVALSTEP";
        ObjectTypeEnum[ObjectTypeEnum["CTAVARIANT"] = 'CTA_VARIANT'] = "CTAVARIANT";
        ObjectTypeEnum[ObjectTypeEnum["UNKNOWN"] = 'UNKNOWN'] = "UNKNOWN";
    })(ObjectTypeEnum = PublicImportError.ObjectTypeEnum || (PublicImportError.ObjectTypeEnum = {}));
})(PublicImportError = exports.PublicImportError || (exports.PublicImportError = {}));
