// @hubspot/api-client/lib/codegen/crm/imports/model/publicImportResponse.js
"use strict";
/**
 * CRM Imports
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicImportResponse = void 0;
/**
* A current summary of the import, whether complete or not.
*/
class PublicImportResponse {
    static getAttributeTypeMap() {
        return PublicImportResponse.attributeTypeMap;
    }
}
exports.PublicImportResponse = PublicImportResponse;
PublicImportResponse.discriminator = undefined;
PublicImportResponse.attributeTypeMap = [
    {
        "name": "state",
        "baseName": "state",
        "type": "PublicImportResponse.StateEnum"
    },
    {
        "name": "importRequestJson",
        "baseName": "importRequestJson",
        "type": "object"
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date"
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "PublicImportMetadata"
    },
    {
        "name": "importName",
        "baseName": "importName",
        "type": "string"
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "Date"
    },
    {
        "name": "optOutImport",
        "baseName": "optOutImport",
        "type": "boolean"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    }
];
(function (PublicImportResponse) {
    let StateEnum;
    (function (StateEnum) {
        StateEnum[StateEnum["STARTED"] = 'STARTED'] = "STARTED";
        StateEnum[StateEnum["PROCESSING"] = 'PROCESSING'] = "PROCESSING";
        StateEnum[StateEnum["DONE"] = 'DONE'] = "DONE";
        StateEnum[StateEnum["FAILED"] = 'FAILED'] = "FAILED";
        StateEnum[StateEnum["CANCELED"] = 'CANCELED'] = "CANCELED";
        StateEnum[StateEnum["DEFERRED"] = 'DEFERRED'] = "DEFERRED";
    })(StateEnum = PublicImportResponse.StateEnum || (PublicImportResponse.StateEnum = {}));
})(PublicImportResponse = exports.PublicImportResponse || (exports.PublicImportResponse = {}));
