// @hubspot/api-client/lib/codegen/automation/actions/model/fieldTypeDefinition.js
"use strict";
/**
 * Custom Workflow Actions
 * Create custom workflow actions
 *
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FieldTypeDefinition = void 0;
/**
* The data type expected by an input field.
*/
class FieldTypeDefinition {
    static getAttributeTypeMap() {
        return FieldTypeDefinition.attributeTypeMap;
    }
}
exports.FieldTypeDefinition = FieldTypeDefinition;
FieldTypeDefinition.discriminator = undefined;
FieldTypeDefinition.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "FieldTypeDefinition.TypeEnum"
    },
    {
        "name": "fieldType",
        "baseName": "fieldType",
        "type": "FieldTypeDefinition.FieldTypeEnum"
    },
    {
        "name": "options",
        "baseName": "options",
        "type": "Array<Option>"
    },
    {
        "name": "optionsUrl",
        "baseName": "optionsUrl",
        "type": "string"
    },
    {
        "name": "referencedObjectType",
        "baseName": "referencedObjectType",
        "type": "FieldTypeDefinition.ReferencedObjectTypeEnum"
    }
];
(function (FieldTypeDefinition) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["String"] = 'string'] = "String";
        TypeEnum[TypeEnum["Number"] = 'number'] = "Number";
        TypeEnum[TypeEnum["Bool"] = 'bool'] = "Bool";
        TypeEnum[TypeEnum["Datetime"] = 'datetime'] = "Datetime";
        TypeEnum[TypeEnum["Enumeration"] = 'enumeration'] = "Enumeration";
        TypeEnum[TypeEnum["Date"] = 'date'] = "Date";
        TypeEnum[TypeEnum["PhoneNumber"] = 'phone_number'] = "PhoneNumber";
        TypeEnum[TypeEnum["CurrencyNumber"] = 'currency_number'] = "CurrencyNumber";
        TypeEnum[TypeEnum["Json"] = 'json'] = "Json";
        TypeEnum[TypeEnum["ObjectCoordinates"] = 'object_coordinates'] = "ObjectCoordinates";
    })(TypeEnum = FieldTypeDefinition.TypeEnum || (FieldTypeDefinition.TypeEnum = {}));
    let FieldTypeEnum;
    (function (FieldTypeEnum) {
        FieldTypeEnum[FieldTypeEnum["Booleancheckbox"] = 'booleancheckbox'] = "Booleancheckbox";
        FieldTypeEnum[FieldTypeEnum["Checkbox"] = 'checkbox'] = "Checkbox";
        FieldTypeEnum[FieldTypeEnum["Date"] = 'date'] = "Date";
        FieldTypeEnum[FieldTypeEnum["File"] = 'file'] = "File";
        FieldTypeEnum[FieldTypeEnum["Number"] = 'number'] = "Number";
        FieldTypeEnum[FieldTypeEnum["Phonenumber"] = 'phonenumber'] = "Phonenumber";
        FieldTypeEnum[FieldTypeEnum["Radio"] = 'radio'] = "Radio";
        FieldTypeEnum[FieldTypeEnum["Select"] = 'select'] = "Select";
        FieldTypeEnum[FieldTypeEnum["Text"] = 'text'] = "Text";
        FieldTypeEnum[FieldTypeEnum["Textarea"] = 'textarea'] = "Textarea";
        FieldTypeEnum[FieldTypeEnum["CalculationEquation"] = 'calculation_equation'] = "CalculationEquation";
        FieldTypeEnum[FieldTypeEnum["CalculationRollup"] = 'calculation_rollup'] = "CalculationRollup";
        FieldTypeEnum[FieldTypeEnum["CalculationScore"] = 'calculation_score'] = "CalculationScore";
        FieldTypeEnum[FieldTypeEnum["CalculationReadTime"] = 'calculation_read_time'] = "CalculationReadTime";
        FieldTypeEnum[FieldTypeEnum["Unknown"] = 'unknown'] = "Unknown";
    })(FieldTypeEnum = FieldTypeDefinition.FieldTypeEnum || (FieldTypeDefinition.FieldTypeEnum = {}));
    let ReferencedObjectTypeEnum;
    (function (ReferencedObjectTypeEnum) {
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["CONTACT"] = 'CONTACT'] = "CONTACT";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["COMPANY"] = 'COMPANY'] = "COMPANY";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["DEAL"] = 'DEAL'] = "DEAL";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["ENGAGEMENT"] = 'ENGAGEMENT'] = "ENGAGEMENT";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["TICKET"] = 'TICKET'] = "TICKET";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["OWNER"] = 'OWNER'] = "OWNER";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["PRODUCT"] = 'PRODUCT'] = "PRODUCT";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["LINEITEM"] = 'LINE_ITEM'] = "LINEITEM";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["BETDELIVERABLESERVICE"] = 'BET_DELIVERABLE_SERVICE'] = "BETDELIVERABLESERVICE";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["CONTENT"] = 'CONTENT'] = "CONTENT";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["CONVERSATION"] = 'CONVERSATION'] = "CONVERSATION";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["BETALERT"] = 'BET_ALERT'] = "BETALERT";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["PORTAL"] = 'PORTAL'] = "PORTAL";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["QUOTE"] = 'QUOTE'] = "QUOTE";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["FORMSUBMISSIONINBOUNDDB"] = 'FORM_SUBMISSION_INBOUNDDB'] = "FORMSUBMISSIONINBOUNDDB";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["QUOTA"] = 'QUOTA'] = "QUOTA";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["UNSUBSCRIBE"] = 'UNSUBSCRIBE'] = "UNSUBSCRIBE";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["COMMUNICATION"] = 'COMMUNICATION'] = "COMMUNICATION";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["FEEDBACKSUBMISSION"] = 'FEEDBACK_SUBMISSION'] = "FEEDBACKSUBMISSION";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["ATTRIBUTION"] = 'ATTRIBUTION'] = "ATTRIBUTION";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["SALESFORCESYNCERROR"] = 'SALESFORCE_SYNC_ERROR'] = "SALESFORCESYNCERROR";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["RESTORABLECRMOBJECT"] = 'RESTORABLE_CRM_OBJECT'] = "RESTORABLECRMOBJECT";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["HUB"] = 'HUB'] = "HUB";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["LANDINGPAGE"] = 'LANDING_PAGE'] = "LANDINGPAGE";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["PRODUCTORFOLDER"] = 'PRODUCT_OR_FOLDER'] = "PRODUCTORFOLDER";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["TASK"] = 'TASK'] = "TASK";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["FORM"] = 'FORM'] = "FORM";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["MARKETINGEMAIL"] = 'MARKETING_EMAIL'] = "MARKETINGEMAIL";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["ADACCOUNT"] = 'AD_ACCOUNT'] = "ADACCOUNT";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["ADCAMPAIGN"] = 'AD_CAMPAIGN'] = "ADCAMPAIGN";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["ADGROUP"] = 'AD_GROUP'] = "ADGROUP";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["AD"] = 'AD'] = "AD";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["KEYWORD"] = 'KEYWORD'] = "KEYWORD";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["CAMPAIGN"] = 'CAMPAIGN'] = "CAMPAIGN";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["SOCIALCHANNEL"] = 'SOCIAL_CHANNEL'] = "SOCIALCHANNEL";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["SOCIALPOST"] = 'SOCIAL_POST'] = "SOCIALPOST";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["SITEPAGE"] = 'SITE_PAGE'] = "SITEPAGE";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["BLOGPOST"] = 'BLOG_POST'] = "BLOGPOST";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["IMPORT"] = 'IMPORT'] = "IMPORT";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["EXPORT"] = 'EXPORT'] = "EXPORT";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["CTA"] = 'CTA'] = "CTA";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["TASKTEMPLATE"] = 'TASK_TEMPLATE'] = "TASKTEMPLATE";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["AUTOMATIONPLATFORMFLOW"] = 'AUTOMATION_PLATFORM_FLOW'] = "AUTOMATIONPLATFORMFLOW";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["OBJECTLIST"] = 'OBJECT_LIST'] = "OBJECTLIST";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["NOTE"] = 'NOTE'] = "NOTE";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["MEETINGEVENT"] = 'MEETING_EVENT'] = "MEETINGEVENT";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["CALL"] = 'CALL'] = "CALL";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["EMAIL"] = 'EMAIL'] = "EMAIL";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["PUBLISHINGTASK"] = 'PUBLISHING_TASK'] = "PUBLISHINGTASK";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["CONVERSATIONSESSION"] = 'CONVERSATION_SESSION'] = "CONVERSATIONSESSION";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["CONTACTCREATEATTRIBUTION"] = 'CONTACT_CREATE_ATTRIBUTION'] = "CONTACTCREATEATTRIBUTION";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["INVOICE"] = 'INVOICE'] = "INVOICE";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["MARKETINGEVENT"] = 'MARKETING_EVENT'] = "MARKETINGEVENT";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["CONVERSATIONINBOX"] = 'CONVERSATION_INBOX'] = "CONVERSATIONINBOX";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["CHATFLOW"] = 'CHATFLOW'] = "CHATFLOW";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["MEDIABRIDGE"] = 'MEDIA_BRIDGE'] = "MEDIABRIDGE";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["SEQUENCE"] = 'SEQUENCE'] = "SEQUENCE";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["SEQUENCESTEP"] = 'SEQUENCE_STEP'] = "SEQUENCESTEP";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["FORECAST"] = 'FORECAST'] = "FORECAST";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["SNIPPET"] = 'SNIPPET'] = "SNIPPET";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["TEMPLATE"] = 'TEMPLATE'] = "TEMPLATE";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["UNKNOWN"] = 'UNKNOWN'] = "UNKNOWN";
    })(ReferencedObjectTypeEnum = FieldTypeDefinition.ReferencedObjectTypeEnum || (FieldTypeDefinition.ReferencedObjectTypeEnum = {}));
})(FieldTypeDefinition = exports.FieldTypeDefinition || (exports.FieldTypeDefinition = {}));
