// @hubspot/api-client/lib/codegen/automation/actions/model/inputFieldDefinition.js
"use strict";
/**
 * Custom Workflow Actions
 * Create custom workflow actions
 *
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InputFieldDefinition = void 0;
/**
* Configuration for an input field on the custom action
*/
class InputFieldDefinition {
    static getAttributeTypeMap() {
        return InputFieldDefinition.attributeTypeMap;
    }
}
exports.InputFieldDefinition = InputFieldDefinition;
InputFieldDefinition.discriminator = undefined;
InputFieldDefinition.attributeTypeMap = [
    {
        "name": "typeDefinition",
        "baseName": "typeDefinition",
        "type": "FieldTypeDefinition"
    },
    {
        "name": "supportedValueTypes",
        "baseName": "supportedValueTypes",
        "type": "Array<InputFieldDefinition.SupportedValueTypesEnum>"
    },
    {
        "name": "isRequired",
        "baseName": "isRequired",
        "type": "boolean"
    }
];
(function (InputFieldDefinition) {
    let SupportedValueTypesEnum;
    (function (SupportedValueTypesEnum) {
        SupportedValueTypesEnum[SupportedValueTypesEnum["STATICVALUE"] = 'STATIC_VALUE'] = "STATICVALUE";
        SupportedValueTypesEnum[SupportedValueTypesEnum["OBJECTPROPERTY"] = 'OBJECT_PROPERTY'] = "OBJECTPROPERTY";
        SupportedValueTypesEnum[SupportedValueTypesEnum["FIELDDATA"] = 'FIELD_DATA'] = "FIELDDATA";
    })(SupportedValueTypesEnum = InputFieldDefinition.SupportedValueTypesEnum || (InputFieldDefinition.SupportedValueTypesEnum = {}));
})(InputFieldDefinition = exports.InputFieldDefinition || (exports.InputFieldDefinition = {}));
