// @hubspot/api-client/lib/codegen/crm/companies/api/apis.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.APIS = exports.HttpError = void 0;
__exportStar(require("./associationsApi"), exports);
const associationsApi_1 = require("./associationsApi");
__exportStar(require("./basicApi"), exports);
const basicApi_1 = require("./basicApi");
__exportStar(require("./batchApi"), exports);
const batchApi_1 = require("./batchApi");
__exportStar(require("./searchApi"), exports);
const searchApi_1 = require("./searchApi");
class HttpError extends Error {
    constructor(response, body, statusCode) {
        super('HTTP request failed');
        this.response = response;
        this.body = body;
        this.statusCode = statusCode;
        this.name = 'HttpError';
    }
}
exports.HttpError = HttpError;
exports.APIS = [associationsApi_1.AssociationsApi, basicApi_1.BasicApi, batchApi_1.BatchApi, searchApi_1.SearchApi];
