// @hubspot/api-client/lib/codegen/crm/schemas/model/models.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoidAuth = exports.OAuth = exports.ApiKeyAuth = exports.HttpBearerAuth = exports.HttpBasicAuth = exports.ObjectSerializer = void 0;
__exportStar(require("./associationDefinition"), exports);
__exportStar(require("./associationDefinitionEgg"), exports);
__exportStar(require("./collectionResponseObjectSchemaNoPaging"), exports);
__exportStar(require("./errorDetail"), exports);
__exportStar(require("./modelError"), exports);
__exportStar(require("./objectSchema"), exports);
__exportStar(require("./objectSchemaEgg"), exports);
__exportStar(require("./objectTypeDefinition"), exports);
__exportStar(require("./objectTypeDefinitionLabels"), exports);
__exportStar(require("./objectTypeDefinitionPatch"), exports);
__exportStar(require("./objectTypePropertyCreate"), exports);
__exportStar(require("./option"), exports);
__exportStar(require("./optionInput"), exports);
__exportStar(require("./property"), exports);
__exportStar(require("./propertyModificationMetadata"), exports);
const associationDefinition_1 = require("./associationDefinition");
const associationDefinitionEgg_1 = require("./associationDefinitionEgg");
const collectionResponseObjectSchemaNoPaging_1 = require("./collectionResponseObjectSchemaNoPaging");
const errorDetail_1 = require("./errorDetail");
const modelError_1 = require("./modelError");
const objectSchema_1 = require("./objectSchema");
const objectSchemaEgg_1 = require("./objectSchemaEgg");
const objectTypeDefinition_1 = require("./objectTypeDefinition");
const objectTypeDefinitionLabels_1 = require("./objectTypeDefinitionLabels");
const objectTypeDefinitionPatch_1 = require("./objectTypeDefinitionPatch");
const objectTypePropertyCreate_1 = require("./objectTypePropertyCreate");
const option_1 = require("./option");
const optionInput_1 = require("./optionInput");
const property_1 = require("./property");
const propertyModificationMetadata_1 = require("./propertyModificationMetadata");
/* tslint:disable:no-unused-variable */
let primitives = [
    "string",
    "boolean",
    "double",
    "integer",
    "long",
    "float",
    "number",
    "any"
];
let enumsMap = {
    "ObjectTypePropertyCreate.TypeEnum": objectTypePropertyCreate_1.ObjectTypePropertyCreate.TypeEnum,
};
let typeMap = {
    "AssociationDefinition": associationDefinition_1.AssociationDefinition,
    "AssociationDefinitionEgg": associationDefinitionEgg_1.AssociationDefinitionEgg,
    "CollectionResponseObjectSchemaNoPaging": collectionResponseObjectSchemaNoPaging_1.CollectionResponseObjectSchemaNoPaging,
    "ErrorDetail": errorDetail_1.ErrorDetail,
    "ModelError": modelError_1.ModelError,
    "ObjectSchema": objectSchema_1.ObjectSchema,
    "ObjectSchemaEgg": objectSchemaEgg_1.ObjectSchemaEgg,
    "ObjectTypeDefinition": objectTypeDefinition_1.ObjectTypeDefinition,
    "ObjectTypeDefinitionLabels": objectTypeDefinitionLabels_1.ObjectTypeDefinitionLabels,
    "ObjectTypeDefinitionPatch": objectTypeDefinitionPatch_1.ObjectTypeDefinitionPatch,
    "ObjectTypePropertyCreate": objectTypePropertyCreate_1.ObjectTypePropertyCreate,
    "Option": option_1.Option,
    "OptionInput": optionInput_1.OptionInput,
    "Property": property_1.Property,
    "PropertyModificationMetadata": propertyModificationMetadata_1.PropertyModificationMetadata,
};
class ObjectSerializer {
    static findCorrectType(data, expectedType) {
        if (data == undefined) {
            return expectedType;
        }
        else if (primitives.indexOf(expectedType.toLowerCase()) !== -1) {
            return expectedType;
        }
        else if (expectedType === "Date") {
            return expectedType;
        }
        else {
            if (enumsMap[expectedType]) {
                return expectedType;
            }
            if (!typeMap[expectedType]) {
                return expectedType; // w/e we don't know the type
            }
            // Check the discriminator
            let discriminatorProperty = typeMap[expectedType].discriminator;
            if (discriminatorProperty == null) {
                return expectedType; // the type does not have a discriminator. use it.
            }
            else {
                if (data[discriminatorProperty]) {
                    var discriminatorType = data[discriminatorProperty];
                    if (typeMap[discriminatorType]) {
                        return discriminatorType; // use the type given in the discriminator
                    }
                    else {
                        return expectedType; // discriminator did not map to a type
                    }
                }
                else {
                    return expectedType; // discriminator was not present (or an empty string)
                }
            }
        }
    }
    static serialize(data, type) {
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index in data) {
                let date = data[index];
                transformedData.push(ObjectSerializer.serialize(date, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return data.toISOString();
        }
        else {
            if (enumsMap[type]) {
                return data;
            }
            if (!typeMap[type]) { // in case we dont know the type
                return data;
            }
            // Get the actual type of this object
            type = this.findCorrectType(data, type);
            // get the map for the correct type.
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            let instance = {};
            for (let index in attributeTypes) {
                let attributeType = attributeTypes[index];
                instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type);
            }
            return instance;
        }
    }
    static deserialize(data, type) {
        // polymorphism may change the actual type.
        type = ObjectSerializer.findCorrectType(data, type);
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index in data) {
                let date = data[index];
                transformedData.push(ObjectSerializer.deserialize(date, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return new Date(data);
        }
        else {
            if (enumsMap[type]) { // is Enum
                return data;
            }
            if (!typeMap[type]) { // dont know the type
                return data;
            }
            let instance = new typeMap[type]();
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            for (let index in attributeTypes) {
                let attributeType = attributeTypes[index];
                instance[attributeType.name] = ObjectSerializer.deserialize(data[attributeType.baseName], attributeType.type);
            }
            return instance;
        }
    }
}
exports.ObjectSerializer = ObjectSerializer;
class HttpBasicAuth {
    constructor() {
        this.username = '';
        this.password = '';
    }
    applyToRequest(requestOptions) {
        requestOptions.auth = {
            username: this.username, password: this.password
        };
    }
}
exports.HttpBasicAuth = HttpBasicAuth;
class HttpBearerAuth {
    constructor() {
        this.accessToken = '';
    }
    applyToRequest(requestOptions) {
        if (requestOptions && requestOptions.headers) {
            const accessToken = typeof this.accessToken === 'function'
                ? this.accessToken()
                : this.accessToken;
            requestOptions.headers["Authorization"] = "Bearer " + accessToken;
        }
    }
}
exports.HttpBearerAuth = HttpBearerAuth;
class ApiKeyAuth {
    constructor(location, paramName) {
        this.location = location;
        this.paramName = paramName;
        this.apiKey = '';
    }
    applyToRequest(requestOptions) {
        if (this.location == "query") {
            requestOptions.qs[this.paramName] = this.apiKey;
        }
        else if (this.location == "header" && requestOptions && requestOptions.headers) {
            requestOptions.headers[this.paramName] = this.apiKey;
        }
        else if (this.location == 'cookie' && requestOptions && requestOptions.headers) {
            if (requestOptions.headers['Cookie']) {
                requestOptions.headers['Cookie'] += '; ' + this.paramName + '=' + encodeURIComponent(this.apiKey);
            }
            else {
                requestOptions.headers['Cookie'] = this.paramName + '=' + encodeURIComponent(this.apiKey);
            }
        }
    }
}
exports.ApiKeyAuth = ApiKeyAuth;
class OAuth {
    constructor() {
        this.accessToken = '';
    }
    applyToRequest(requestOptions) {
        if (requestOptions && requestOptions.headers) {
            requestOptions.headers["Authorization"] = "Bearer " + this.accessToken;
        }
    }
}
exports.OAuth = OAuth;
class VoidAuth {
    constructor() {
        this.username = '';
        this.password = '';
    }
    applyToRequest(_) {
        // Do nothing
    }
}
exports.VoidAuth = VoidAuth;
