// @hubspot/api-client/lib/codegen/crm/schemas/model/objectTypePropertyCreate.js
"use strict";
/**
 * Schemas
 * The CRM uses schemas to define how custom objects should store and represent information in the HubSpot CRM. Schemas define details about an object\'s type, properties, and associations. The schema can be uniquely identified by its **object type ID**.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectTypePropertyCreate = void 0;
/**
* Defines a property to create.
*/
class ObjectTypePropertyCreate {
    static getAttributeTypeMap() {
        return ObjectTypePropertyCreate.attributeTypeMap;
    }
}
exports.ObjectTypePropertyCreate = ObjectTypePropertyCreate;
ObjectTypePropertyCreate.discriminator = undefined;
ObjectTypePropertyCreate.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string"
    },
    {
        "name": "groupName",
        "baseName": "groupName",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "options",
        "baseName": "options",
        "type": "Array<OptionInput>"
    },
    {
        "name": "displayOrder",
        "baseName": "displayOrder",
        "type": "number"
    },
    {
        "name": "hasUniqueValue",
        "baseName": "hasUniqueValue",
        "type": "boolean"
    },
    {
        "name": "hidden",
        "baseName": "hidden",
        "type": "boolean"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "ObjectTypePropertyCreate.TypeEnum"
    },
    {
        "name": "fieldType",
        "baseName": "fieldType",
        "type": "string"
    }
];
(function (ObjectTypePropertyCreate) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["String"] = 'string'] = "String";
        TypeEnum[TypeEnum["Number"] = 'number'] = "Number";
        TypeEnum[TypeEnum["Date"] = 'date'] = "Date";
        TypeEnum[TypeEnum["Datetime"] = 'datetime'] = "Datetime";
        TypeEnum[TypeEnum["Enumeration"] = 'enumeration'] = "Enumeration";
        TypeEnum[TypeEnum["Bool"] = 'bool'] = "Bool";
    })(TypeEnum = ObjectTypePropertyCreate.TypeEnum || (ObjectTypePropertyCreate.TypeEnum = {}));
})(ObjectTypePropertyCreate = exports.ObjectTypePropertyCreate || (exports.ObjectTypePropertyCreate = {}));
