// @hubspot/api-client/lib/codegen/crm/extensions/cards/model/actionHookActionBody.js
"use strict";
/**
 * CRM cards
 * Allows an app to extend the CRM UI by surfacing custom cards in the sidebar of record pages. These cards are defined up-front as part of app configuration, then populated by external data fetch requests when the record page is accessed by a user.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionHookActionBody = void 0;
class ActionHookActionBody {
    static getAttributeTypeMap() {
        return ActionHookActionBody.attributeTypeMap;
    }
}
exports.ActionHookActionBody = ActionHookActionBody;
ActionHookActionBody.discriminator = undefined;
ActionHookActionBody.attributeTypeMap = [
    {
        "name": "type",
        "baseName": "type",
        "type": "ActionHookActionBody.TypeEnum"
    },
    {
        "name": "confirmation",
        "baseName": "confirmation",
        "type": "ActionConfirmationBody"
    },
    {
        "name": "httpMethod",
        "baseName": "httpMethod",
        "type": "ActionHookActionBody.HttpMethodEnum"
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string"
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string"
    },
    {
        "name": "propertyNamesIncluded",
        "baseName": "propertyNamesIncluded",
        "type": "Array<string>"
    }
];
(function (ActionHookActionBody) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["ACTIONHOOK"] = 'ACTION_HOOK'] = "ACTIONHOOK";
    })(TypeEnum = ActionHookActionBody.TypeEnum || (ActionHookActionBody.TypeEnum = {}));
    let HttpMethodEnum;
    (function (HttpMethodEnum) {
        HttpMethodEnum[HttpMethodEnum["CONNECT"] = 'CONNECT'] = "CONNECT";
        HttpMethodEnum[HttpMethodEnum["DELETE"] = 'DELETE'] = "DELETE";
        HttpMethodEnum[HttpMethodEnum["GET"] = 'GET'] = "GET";
        HttpMethodEnum[HttpMethodEnum["HEAD"] = 'HEAD'] = "HEAD";
        HttpMethodEnum[HttpMethodEnum["OPTIONS"] = 'OPTIONS'] = "OPTIONS";
        HttpMethodEnum[HttpMethodEnum["PATCH"] = 'PATCH'] = "PATCH";
        HttpMethodEnum[HttpMethodEnum["POST"] = 'POST'] = "POST";
        HttpMethodEnum[HttpMethodEnum["PUT"] = 'PUT'] = "PUT";
        HttpMethodEnum[HttpMethodEnum["TRACE"] = 'TRACE'] = "TRACE";
    })(HttpMethodEnum = ActionHookActionBody.HttpMethodEnum || (ActionHookActionBody.HttpMethodEnum = {}));
})(ActionHookActionBody = exports.ActionHookActionBody || (exports.ActionHookActionBody = {}));
