// @hubspot/api-client/lib/codegen/crm/extensions/cards/model/cardObjectTypeBody.js
"use strict";
/**
 * CRM cards
 * Allows an app to extend the CRM UI by surfacing custom cards in the sidebar of record pages. These cards are defined up-front as part of app configuration, then populated by external data fetch requests when the record page is accessed by a user.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardObjectTypeBody = void 0;
class CardObjectTypeBody {
    static getAttributeTypeMap() {
        return CardObjectTypeBody.attributeTypeMap;
    }
}
exports.CardObjectTypeBody = CardObjectTypeBody;
CardObjectTypeBody.discriminator = undefined;
CardObjectTypeBody.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "CardObjectTypeBody.NameEnum"
    },
    {
        "name": "propertiesToSend",
        "baseName": "propertiesToSend",
        "type": "Array<string>"
    }
];
(function (CardObjectTypeBody) {
    let NameEnum;
    (function (NameEnum) {
        NameEnum[NameEnum["Contacts"] = 'contacts'] = "Contacts";
        NameEnum[NameEnum["Deals"] = 'deals'] = "Deals";
        NameEnum[NameEnum["Companies"] = 'companies'] = "Companies";
        NameEnum[NameEnum["Tickets"] = 'tickets'] = "Tickets";
    })(NameEnum = CardObjectTypeBody.NameEnum || (CardObjectTypeBody.NameEnum = {}));
})(CardObjectTypeBody = exports.CardObjectTypeBody || (exports.CardObjectTypeBody = {}));
