// @hubspot/api-client/lib/codegen/crm/extensions/cards/model/displayOption.js
"use strict";
/**
 * CRM cards
 * Allows an app to extend the CRM UI by surfacing custom cards in the sidebar of record pages. These cards are defined up-front as part of app configuration, then populated by external data fetch requests when the record page is accessed by a user.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DisplayOption = void 0;
/**
* Option definition for STATUS dataTypes.
*/
class DisplayOption {
    static getAttributeTypeMap() {
        return DisplayOption.attributeTypeMap;
    }
}
exports.DisplayOption = DisplayOption;
DisplayOption.discriminator = undefined;
DisplayOption.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "DisplayOption.TypeEnum"
    }
];
(function (DisplayOption) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["DEFAULT"] = 'DEFAULT'] = "DEFAULT";
        TypeEnum[TypeEnum["SUCCESS"] = 'SUCCESS'] = "SUCCESS";
        TypeEnum[TypeEnum["WARNING"] = 'WARNING'] = "WARNING";
        TypeEnum[TypeEnum["DANGER"] = 'DANGER'] = "DANGER";
        TypeEnum[TypeEnum["INFO"] = 'INFO'] = "INFO";
    })(TypeEnum = DisplayOption.TypeEnum || (DisplayOption.TypeEnum = {}));
})(DisplayOption = exports.DisplayOption || (exports.DisplayOption = {}));
