// @hubspot/api-client/lib/codegen/crm/extensions/cards/model/objectToken.js
"use strict";
/**
 * CRM cards
 * Allows an app to extend the CRM UI by surfacing custom cards in the sidebar of record pages. These cards are defined up-front as part of app configuration, then populated by external data fetch requests when the record page is accessed by a user.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectToken = void 0;
class ObjectToken {
    static getAttributeTypeMap() {
        return ObjectToken.attributeTypeMap;
    }
}
exports.ObjectToken = ObjectToken;
ObjectToken.discriminator = undefined;
ObjectToken.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string"
    },
    {
        "name": "dataType",
        "baseName": "dataType",
        "type": "ObjectToken.DataTypeEnum"
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "string"
    }
];
(function (ObjectToken) {
    let DataTypeEnum;
    (function (DataTypeEnum) {
        DataTypeEnum[DataTypeEnum["BOOLEAN"] = 'BOOLEAN'] = "BOOLEAN";
        DataTypeEnum[DataTypeEnum["CURRENCY"] = 'CURRENCY'] = "CURRENCY";
        DataTypeEnum[DataTypeEnum["DATE"] = 'DATE'] = "DATE";
        DataTypeEnum[DataTypeEnum["DATETIME"] = 'DATETIME'] = "DATETIME";
        DataTypeEnum[DataTypeEnum["EMAIL"] = 'EMAIL'] = "EMAIL";
        DataTypeEnum[DataTypeEnum["LINK"] = 'LINK'] = "LINK";
        DataTypeEnum[DataTypeEnum["NUMERIC"] = 'NUMERIC'] = "NUMERIC";
        DataTypeEnum[DataTypeEnum["STRING"] = 'STRING'] = "STRING";
        DataTypeEnum[DataTypeEnum["STATUS"] = 'STATUS'] = "STATUS";
    })(DataTypeEnum = ObjectToken.DataTypeEnum || (ObjectToken.DataTypeEnum = {}));
})(ObjectToken = exports.ObjectToken || (exports.ObjectToken = {}));
