// @hubspot/api-client/lib/codegen/crm/properties/model/batchResponseProperty.js
"use strict";
/**
 * Properties
 * All HubSpot objects store data in default and custom properties. These endpoints provide access to read and modify object properties in HubSpot.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseProperty = void 0;
class BatchResponseProperty {
    static getAttributeTypeMap() {
        return BatchResponseProperty.attributeTypeMap;
    }
}
exports.BatchResponseProperty = BatchResponseProperty;
BatchResponseProperty.discriminator = undefined;
BatchResponseProperty.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseProperty.StatusEnum"
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<Property>"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number"
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date"
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }"
    }
];
(function (BatchResponseProperty) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["PENDING"] = 'PENDING'] = "PENDING";
        StatusEnum[StatusEnum["PROCESSING"] = 'PROCESSING'] = "PROCESSING";
        StatusEnum[StatusEnum["CANCELED"] = 'CANCELED'] = "CANCELED";
        StatusEnum[StatusEnum["COMPLETE"] = 'COMPLETE'] = "COMPLETE";
    })(StatusEnum = BatchResponseProperty.StatusEnum || (BatchResponseProperty.StatusEnum = {}));
})(BatchResponseProperty = exports.BatchResponseProperty || (exports.BatchResponseProperty = {}));
