// @hubspot/api-client/lib/codegen/crm/properties/model/propertyUpdate.js
"use strict";
/**
 * Properties
 * All HubSpot objects store data in default and custom properties. These endpoints provide access to read and modify object properties in HubSpot.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PropertyUpdate = void 0;
class PropertyUpdate {
    static getAttributeTypeMap() {
        return PropertyUpdate.attributeTypeMap;
    }
}
exports.PropertyUpdate = PropertyUpdate;
PropertyUpdate.discriminator = undefined;
PropertyUpdate.attributeTypeMap = [
    {
        "name": "label",
        "baseName": "label",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PropertyUpdate.TypeEnum"
    },
    {
        "name": "fieldType",
        "baseName": "fieldType",
        "type": "PropertyUpdate.FieldTypeEnum"
    },
    {
        "name": "groupName",
        "baseName": "groupName",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "options",
        "baseName": "options",
        "type": "Array<OptionInput>"
    },
    {
        "name": "displayOrder",
        "baseName": "displayOrder",
        "type": "number"
    },
    {
        "name": "hidden",
        "baseName": "hidden",
        "type": "boolean"
    },
    {
        "name": "formField",
        "baseName": "formField",
        "type": "boolean"
    }
];
(function (PropertyUpdate) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["String"] = 'string'] = "String";
        TypeEnum[TypeEnum["Number"] = 'number'] = "Number";
        TypeEnum[TypeEnum["Date"] = 'date'] = "Date";
        TypeEnum[TypeEnum["Datetime"] = 'datetime'] = "Datetime";
        TypeEnum[TypeEnum["Enumeration"] = 'enumeration'] = "Enumeration";
    })(TypeEnum = PropertyUpdate.TypeEnum || (PropertyUpdate.TypeEnum = {}));
    let FieldTypeEnum;
    (function (FieldTypeEnum) {
        FieldTypeEnum[FieldTypeEnum["Textarea"] = 'textarea'] = "Textarea";
        FieldTypeEnum[FieldTypeEnum["Text"] = 'text'] = "Text";
        FieldTypeEnum[FieldTypeEnum["Date"] = 'date'] = "Date";
        FieldTypeEnum[FieldTypeEnum["File"] = 'file'] = "File";
        FieldTypeEnum[FieldTypeEnum["Number"] = 'number'] = "Number";
        FieldTypeEnum[FieldTypeEnum["Select"] = 'select'] = "Select";
        FieldTypeEnum[FieldTypeEnum["Radio"] = 'radio'] = "Radio";
        FieldTypeEnum[FieldTypeEnum["Checkbox"] = 'checkbox'] = "Checkbox";
        FieldTypeEnum[FieldTypeEnum["Booleancheckbox"] = 'booleancheckbox'] = "Booleancheckbox";
    })(FieldTypeEnum = PropertyUpdate.FieldTypeEnum || (PropertyUpdate.FieldTypeEnum = {}));
})(PropertyUpdate = exports.PropertyUpdate || (exports.PropertyUpdate = {}));
