// @hubspot/api-client/lib/codegen/cms/blogs/authors/model/batchResponseBlogAuthorWithErrors.js
"use strict";
/**
 * Blog Post endpoints
 * \"Use these endpoints for interacting with Blog Posts, Blog Authors, and Blog Tags\"
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseBlogAuthorWithErrors = void 0;
class BatchResponseBlogAuthorWithErrors {
    static getAttributeTypeMap() {
        return BatchResponseBlogAuthorWithErrors.attributeTypeMap;
    }
}
exports.BatchResponseBlogAuthorWithErrors = BatchResponseBlogAuthorWithErrors;
BatchResponseBlogAuthorWithErrors.discriminator = undefined;
BatchResponseBlogAuthorWithErrors.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseBlogAuthorWithErrors.StatusEnum"
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<BlogAuthor>"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number"
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date"
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }"
    }
];
(function (BatchResponseBlogAuthorWithErrors) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["PENDING"] = 'PENDING'] = "PENDING";
        StatusEnum[StatusEnum["PROCESSING"] = 'PROCESSING'] = "PROCESSING";
        StatusEnum[StatusEnum["CANCELED"] = 'CANCELED'] = "CANCELED";
        StatusEnum[StatusEnum["COMPLETE"] = 'COMPLETE'] = "COMPLETE";
    })(StatusEnum = BatchResponseBlogAuthorWithErrors.StatusEnum || (BatchResponseBlogAuthorWithErrors.StatusEnum = {}));
})(BatchResponseBlogAuthorWithErrors = exports.BatchResponseBlogAuthorWithErrors || (exports.BatchResponseBlogAuthorWithErrors = {}));
