// @hubspot/api-client/lib/codegen/cms/hubdb/model/models.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoidAuth = exports.OAuth = exports.ApiKeyAuth = exports.HttpBearerAuth = exports.HttpBasicAuth = exports.ObjectSerializer = void 0;
__exportStar(require("./batchInputHubDbTableRowV3"), exports);
__exportStar(require("./batchInputJsonNode"), exports);
__exportStar(require("./batchInputString"), exports);
__exportStar(require("./batchResponseHubDbTableRowV3WithErrors"), exports);
__exportStar(require("./collectionResponseWithTotalHubDbTableRowV3ForwardPaging"), exports);
__exportStar(require("./collectionResponseWithTotalHubDbTableV3ForwardPaging"), exports);
__exportStar(require("./column"), exports);
__exportStar(require("./columnInput"), exports);
__exportStar(require("./errorDetail"), exports);
__exportStar(require("./foreignId"), exports);
__exportStar(require("./forwardPaging"), exports);
__exportStar(require("./hubDbTableCloneRequest"), exports);
__exportStar(require("./hubDbTableRowV3"), exports);
__exportStar(require("./hubDbTableRowV3Input"), exports);
__exportStar(require("./hubDbTableV3"), exports);
__exportStar(require("./hubDbTableV3Input"), exports);
__exportStar(require("./hubDbTableV3LiveInput"), exports);
__exportStar(require("./importResult"), exports);
__exportStar(require("./modelError"), exports);
__exportStar(require("./nextPage"), exports);
__exportStar(require("./option"), exports);
__exportStar(require("./simpleUser"), exports);
__exportStar(require("./standardError"), exports);
const batchInputHubDbTableRowV3_1 = require("./batchInputHubDbTableRowV3");
const batchInputJsonNode_1 = require("./batchInputJsonNode");
const batchInputString_1 = require("./batchInputString");
const batchResponseHubDbTableRowV3WithErrors_1 = require("./batchResponseHubDbTableRowV3WithErrors");
const collectionResponseWithTotalHubDbTableRowV3ForwardPaging_1 = require("./collectionResponseWithTotalHubDbTableRowV3ForwardPaging");
const collectionResponseWithTotalHubDbTableV3ForwardPaging_1 = require("./collectionResponseWithTotalHubDbTableV3ForwardPaging");
const column_1 = require("./column");
const columnInput_1 = require("./columnInput");
const errorDetail_1 = require("./errorDetail");
const foreignId_1 = require("./foreignId");
const forwardPaging_1 = require("./forwardPaging");
const hubDbTableCloneRequest_1 = require("./hubDbTableCloneRequest");
const hubDbTableRowV3_1 = require("./hubDbTableRowV3");
const hubDbTableRowV3Input_1 = require("./hubDbTableRowV3Input");
const hubDbTableV3_1 = require("./hubDbTableV3");
const hubDbTableV3Input_1 = require("./hubDbTableV3Input");
const hubDbTableV3LiveInput_1 = require("./hubDbTableV3LiveInput");
const importResult_1 = require("./importResult");
const modelError_1 = require("./modelError");
const nextPage_1 = require("./nextPage");
const option_1 = require("./option");
const simpleUser_1 = require("./simpleUser");
const standardError_1 = require("./standardError");
/* tslint:disable:no-unused-variable */
let primitives = [
    "string",
    "boolean",
    "double",
    "integer",
    "long",
    "float",
    "number",
    "any"
];
let enumsMap = {
    "BatchResponseHubDbTableRowV3WithErrors.StatusEnum": batchResponseHubDbTableRowV3WithErrors_1.BatchResponseHubDbTableRowV3WithErrors.StatusEnum,
    "Column.TypeEnum": column_1.Column.TypeEnum,
};
let typeMap = {
    "BatchInputHubDbTableRowV3": batchInputHubDbTableRowV3_1.BatchInputHubDbTableRowV3,
    "BatchInputJsonNode": batchInputJsonNode_1.BatchInputJsonNode,
    "BatchInputString": batchInputString_1.BatchInputString,
    "BatchResponseHubDbTableRowV3WithErrors": batchResponseHubDbTableRowV3WithErrors_1.BatchResponseHubDbTableRowV3WithErrors,
    "CollectionResponseWithTotalHubDbTableRowV3ForwardPaging": collectionResponseWithTotalHubDbTableRowV3ForwardPaging_1.CollectionResponseWithTotalHubDbTableRowV3ForwardPaging,
    "CollectionResponseWithTotalHubDbTableV3ForwardPaging": collectionResponseWithTotalHubDbTableV3ForwardPaging_1.CollectionResponseWithTotalHubDbTableV3ForwardPaging,
    "Column": column_1.Column,
    "ColumnInput": columnInput_1.ColumnInput,
    "ErrorDetail": errorDetail_1.ErrorDetail,
    "ForeignId": foreignId_1.ForeignId,
    "ForwardPaging": forwardPaging_1.ForwardPaging,
    "HubDbTableCloneRequest": hubDbTableCloneRequest_1.HubDbTableCloneRequest,
    "HubDbTableRowV3": hubDbTableRowV3_1.HubDbTableRowV3,
    "HubDbTableRowV3Input": hubDbTableRowV3Input_1.HubDbTableRowV3Input,
    "HubDbTableV3": hubDbTableV3_1.HubDbTableV3,
    "HubDbTableV3Input": hubDbTableV3Input_1.HubDbTableV3Input,
    "HubDbTableV3LiveInput": hubDbTableV3LiveInput_1.HubDbTableV3LiveInput,
    "ImportResult": importResult_1.ImportResult,
    "ModelError": modelError_1.ModelError,
    "NextPage": nextPage_1.NextPage,
    "Option": option_1.Option,
    "SimpleUser": simpleUser_1.SimpleUser,
    "StandardError": standardError_1.StandardError,
};
class ObjectSerializer {
    static findCorrectType(data, expectedType) {
        if (data == undefined) {
            return expectedType;
        }
        else if (primitives.indexOf(expectedType.toLowerCase()) !== -1) {
            return expectedType;
        }
        else if (expectedType === "Date") {
            return expectedType;
        }
        else {
            if (enumsMap[expectedType]) {
                return expectedType;
            }
            if (!typeMap[expectedType]) {
                return expectedType; // w/e we don't know the type
            }
            // Check the discriminator
            let discriminatorProperty = typeMap[expectedType].discriminator;
            if (discriminatorProperty == null) {
                return expectedType; // the type does not have a discriminator. use it.
            }
            else {
                if (data[discriminatorProperty]) {
                    var discriminatorType = data[discriminatorProperty];
                    if (typeMap[discriminatorType]) {
                        return discriminatorType; // use the type given in the discriminator
                    }
                    else {
                        return expectedType; // discriminator did not map to a type
                    }
                }
                else {
                    return expectedType; // discriminator was not present (or an empty string)
                }
            }
        }
    }
    static serialize(data, type) {
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index in data) {
                let date = data[index];
                transformedData.push(ObjectSerializer.serialize(date, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return data.toISOString();
        }
        else {
            if (enumsMap[type]) {
                return data;
            }
            if (!typeMap[type]) { // in case we dont know the type
                return data;
            }
            // Get the actual type of this object
            type = this.findCorrectType(data, type);
            // get the map for the correct type.
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            let instance = {};
            for (let index in attributeTypes) {
                let attributeType = attributeTypes[index];
                instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type);
            }
            return instance;
        }
    }
    static deserialize(data, type) {
        // polymorphism may change the actual type.
        type = ObjectSerializer.findCorrectType(data, type);
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) { // string.startsWith pre es6
            let subType = type.replace("Array<", ""); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index in data) {
                let date = data[index];
                transformedData.push(ObjectSerializer.deserialize(date, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return new Date(data);
        }
        else {
            if (enumsMap[type]) { // is Enum
                return data;
            }
            if (!typeMap[type]) { // dont know the type
                return data;
            }
            let instance = new typeMap[type]();
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            for (let index in attributeTypes) {
                let attributeType = attributeTypes[index];
                instance[attributeType.name] = ObjectSerializer.deserialize(data[attributeType.baseName], attributeType.type);
            }
            return instance;
        }
    }
}
exports.ObjectSerializer = ObjectSerializer;
class HttpBasicAuth {
    constructor() {
        this.username = '';
        this.password = '';
    }
    applyToRequest(requestOptions) {
        requestOptions.auth = {
            username: this.username, password: this.password
        };
    }
}
exports.HttpBasicAuth = HttpBasicAuth;
class HttpBearerAuth {
    constructor() {
        this.accessToken = '';
    }
    applyToRequest(requestOptions) {
        if (requestOptions && requestOptions.headers) {
            const accessToken = typeof this.accessToken === 'function'
                ? this.accessToken()
                : this.accessToken;
            requestOptions.headers["Authorization"] = "Bearer " + accessToken;
        }
    }
}
exports.HttpBearerAuth = HttpBearerAuth;
class ApiKeyAuth {
    constructor(location, paramName) {
        this.location = location;
        this.paramName = paramName;
        this.apiKey = '';
    }
    applyToRequest(requestOptions) {
        if (this.location == "query") {
            requestOptions.qs[this.paramName] = this.apiKey;
        }
        else if (this.location == "header" && requestOptions && requestOptions.headers) {
            requestOptions.headers[this.paramName] = this.apiKey;
        }
        else if (this.location == 'cookie' && requestOptions && requestOptions.headers) {
            if (requestOptions.headers['Cookie']) {
                requestOptions.headers['Cookie'] += '; ' + this.paramName + '=' + encodeURIComponent(this.apiKey);
            }
            else {
                requestOptions.headers['Cookie'] = this.paramName + '=' + encodeURIComponent(this.apiKey);
            }
        }
    }
}
exports.ApiKeyAuth = ApiKeyAuth;
class OAuth {
    constructor() {
        this.accessToken = '';
    }
    applyToRequest(requestOptions) {
        if (requestOptions && requestOptions.headers) {
            requestOptions.headers["Authorization"] = "Bearer " + this.accessToken;
        }
    }
}
exports.OAuth = OAuth;
class VoidAuth {
    constructor() {
        this.username = '';
        this.password = '';
    }
    applyToRequest(_) {
        // Do nothing
    }
}
exports.VoidAuth = VoidAuth;
