// @hubspot/api-client/lib/codegen/crm/extensions/accounting/model/accountingExtensionInvoice.js
"use strict";
/**
 * Accounting Extension
 * These APIs allow you to interact with HubSpot\'s Accounting Extension. It allows you to: * Specify the URLs that HubSpot will use when making webhook requests to your external accounting system. * Respond to webhook calls made to your external accounting system by HubSpot
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountingExtensionInvoice = void 0;
/**
* Representation of an invoice in the external accounting system.
*/
class AccountingExtensionInvoice {
    static getAttributeTypeMap() {
        return AccountingExtensionInvoice.attributeTypeMap;
    }
}
exports.AccountingExtensionInvoice = AccountingExtensionInvoice;
AccountingExtensionInvoice.discriminator = undefined;
AccountingExtensionInvoice.attributeTypeMap = [
    {
        "name": "amountDue",
        "baseName": "amountDue",
        "type": "number"
    },
    {
        "name": "balance",
        "baseName": "balance",
        "type": "number"
    },
    {
        "name": "dueDate",
        "baseName": "dueDate",
        "type": "string"
    },
    {
        "name": "invoiceNumber",
        "baseName": "invoiceNumber",
        "type": "string"
    },
    {
        "name": "customerId",
        "baseName": "customerId",
        "type": "string"
    },
    {
        "name": "currency",
        "baseName": "currency",
        "type": "string"
    },
    {
        "name": "invoiceLink",
        "baseName": "invoiceLink",
        "type": "string"
    },
    {
        "name": "customerName",
        "baseName": "customerName",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "AccountingExtensionInvoice.StatusEnum"
    }
];
(function (AccountingExtensionInvoice) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["CREATED"] = 'CREATED'] = "CREATED";
        StatusEnum[StatusEnum["SENT"] = 'SENT'] = "SENT";
        StatusEnum[StatusEnum["PAID"] = 'PAID'] = "PAID";
        StatusEnum[StatusEnum["CLOSED"] = 'CLOSED'] = "CLOSED";
        StatusEnum[StatusEnum["OVERDUE"] = 'OVERDUE'] = "OVERDUE";
        StatusEnum[StatusEnum["VOIDED"] = 'VOIDED'] = "VOIDED";
    })(StatusEnum = AccountingExtensionInvoice.StatusEnum || (AccountingExtensionInvoice.StatusEnum = {}));
})(AccountingExtensionInvoice = exports.AccountingExtensionInvoice || (exports.AccountingExtensionInvoice = {}));
