// @hubspot/api-client/lib/codegen/crm/extensions/accounting/model/exchangeRateResponse.js
"use strict";
/**
 * Accounting Extension
 * These APIs allow you to interact with HubSpot\'s Accounting Extension. It allows you to: * Specify the URLs that HubSpot will use when making webhook requests to your external accounting system. * Respond to webhook calls made to your external accounting system by HubSpot
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExchangeRateResponse = void 0;
/**
* A response to the request for an exchange rate value. It represents the exchange rate from the source currency to the target currency.
*/
class ExchangeRateResponse {
    static getAttributeTypeMap() {
        return ExchangeRateResponse.attributeTypeMap;
    }
}
exports.ExchangeRateResponse = ExchangeRateResponse;
ExchangeRateResponse.discriminator = undefined;
ExchangeRateResponse.attributeTypeMap = [
    {
        "name": "result",
        "baseName": "@result",
        "type": "ExchangeRateResponse.ResultEnum"
    },
    {
        "name": "exchangeRate",
        "baseName": "exchangeRate",
        "type": "number"
    },
    {
        "name": "sourceCurrencyCode",
        "baseName": "sourceCurrencyCode",
        "type": "string"
    },
    {
        "name": "targetCurrencyCode",
        "baseName": "targetCurrencyCode",
        "type": "string"
    }
];
(function (ExchangeRateResponse) {
    let ResultEnum;
    (function (ResultEnum) {
        ResultEnum[ResultEnum["OK"] = 'OK'] = "OK";
        ResultEnum[ResultEnum["ERR"] = 'ERR'] = "ERR";
    })(ResultEnum = ExchangeRateResponse.ResultEnum || (ExchangeRateResponse.ResultEnum = {}));
})(ExchangeRateResponse = exports.ExchangeRateResponse || (exports.ExchangeRateResponse = {}));
