// @hubspot/api-client/lib/codegen/crm/extensions/accounting/model/invoicePdfResponse.js
"use strict";
/**
 * Accounting Extension
 * These APIs allow you to interact with HubSpot\'s Accounting Extension. It allows you to: * Specify the URLs that HubSpot will use when making webhook requests to your external accounting system. * Respond to webhook calls made to your external accounting system by HubSpot
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvoicePdfResponse = void 0;
/**
* A response that contains the PDF of an invoice
*/
class InvoicePdfResponse {
    static getAttributeTypeMap() {
        return InvoicePdfResponse.attributeTypeMap;
    }
}
exports.InvoicePdfResponse = InvoicePdfResponse;
InvoicePdfResponse.discriminator = undefined;
InvoicePdfResponse.attributeTypeMap = [
    {
        "name": "result",
        "baseName": "@result",
        "type": "InvoicePdfResponse.ResultEnum"
    },
    {
        "name": "invoice",
        "baseName": "invoice",
        "type": "string"
    }
];
(function (InvoicePdfResponse) {
    let ResultEnum;
    (function (ResultEnum) {
        ResultEnum[ResultEnum["OK"] = 'OK'] = "OK";
        ResultEnum[ResultEnum["ERR"] = 'ERR'] = "ERR";
    })(ResultEnum = InvoicePdfResponse.ResultEnum || (InvoicePdfResponse.ResultEnum = {}));
})(InvoicePdfResponse = exports.InvoicePdfResponse || (exports.InvoicePdfResponse = {}));
