// @hubspot/api-client/lib/codegen/crm/extensions/accounting/model/invoiceSearchResponse.js
"use strict";
/**
 * Accounting Extension
 * These APIs allow you to interact with HubSpot\'s Accounting Extension. It allows you to: * Specify the URLs that HubSpot will use when making webhook requests to your external accounting system. * Respond to webhook calls made to your external accounting system by HubSpot
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvoiceSearchResponse = void 0;
/**
* A response to a search for invoices.
*/
class InvoiceSearchResponse {
    static getAttributeTypeMap() {
        return InvoiceSearchResponse.attributeTypeMap;
    }
}
exports.InvoiceSearchResponse = InvoiceSearchResponse;
InvoiceSearchResponse.discriminator = undefined;
InvoiceSearchResponse.attributeTypeMap = [
    {
        "name": "result",
        "baseName": "@result",
        "type": "InvoiceSearchResponse.ResultEnum"
    },
    {
        "name": "invoices",
        "baseName": "invoices",
        "type": "Array<AccountingExtensionInvoice>"
    }
];
(function (InvoiceSearchResponse) {
    let ResultEnum;
    (function (ResultEnum) {
        ResultEnum[ResultEnum["OK"] = 'OK'] = "OK";
        ResultEnum[ResultEnum["ERR"] = 'ERR'] = "ERR";
    })(ResultEnum = InvoiceSearchResponse.ResultEnum || (InvoiceSearchResponse.ResultEnum = {}));
})(InvoiceSearchResponse = exports.InvoiceSearchResponse || (exports.InvoiceSearchResponse = {}));
