// @hubspot/api-client/lib/codegen/crm/extensions/accounting/model/invoiceUpdateResponse.js
"use strict";
/**
 * Accounting Extension
 * These APIs allow you to interact with HubSpot\'s Accounting Extension. It allows you to: * Specify the URLs that HubSpot will use when making webhook requests to your external accounting system. * Respond to webhook calls made to your external accounting system by HubSpot
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvoiceUpdateResponse = void 0;
class InvoiceUpdateResponse {
    static getAttributeTypeMap() {
        return InvoiceUpdateResponse.attributeTypeMap;
    }
}
exports.InvoiceUpdateResponse = InvoiceUpdateResponse;
InvoiceUpdateResponse.discriminator = undefined;
InvoiceUpdateResponse.attributeTypeMap = [
    {
        "name": "externalInvoiceNumber",
        "baseName": "externalInvoiceNumber",
        "type": "string"
    },
    {
        "name": "totalAmountBilled",
        "baseName": "totalAmountBilled",
        "type": "number"
    },
    {
        "name": "balanceDue",
        "baseName": "balanceDue",
        "type": "number"
    },
    {
        "name": "currencyCode",
        "baseName": "currencyCode",
        "type": "string"
    },
    {
        "name": "dueDate",
        "baseName": "dueDate",
        "type": "string"
    },
    {
        "name": "externalRecipientId",
        "baseName": "externalRecipientId",
        "type": "string"
    },
    {
        "name": "receivedByRecipientDate",
        "baseName": "receivedByRecipientDate",
        "type": "number"
    },
    {
        "name": "externalCreateDateTime",
        "baseName": "externalCreateDateTime",
        "type": "number"
    },
    {
        "name": "isVoided",
        "baseName": "isVoided",
        "type": "boolean"
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date"
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "Date"
    },
    {
        "name": "archivedAt",
        "baseName": "archivedAt",
        "type": "Date"
    },
    {
        "name": "archived",
        "baseName": "archived",
        "type": "boolean"
    },
    {
        "name": "externalAccountId",
        "baseName": "externalAccountId",
        "type": "string"
    },
    {
        "name": "invoiceStatus",
        "baseName": "invoiceStatus",
        "type": "InvoiceUpdateResponse.InvoiceStatusEnum"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    }
];
(function (InvoiceUpdateResponse) {
    let InvoiceStatusEnum;
    (function (InvoiceStatusEnum) {
        InvoiceStatusEnum[InvoiceStatusEnum["CREATED"] = 'CREATED'] = "CREATED";
        InvoiceStatusEnum[InvoiceStatusEnum["SENT"] = 'SENT'] = "SENT";
        InvoiceStatusEnum[InvoiceStatusEnum["PAID"] = 'PAID'] = "PAID";
        InvoiceStatusEnum[InvoiceStatusEnum["CLOSED"] = 'CLOSED'] = "CLOSED";
        InvoiceStatusEnum[InvoiceStatusEnum["OVERDUE"] = 'OVERDUE'] = "OVERDUE";
        InvoiceStatusEnum[InvoiceStatusEnum["VOIDED"] = 'VOIDED'] = "VOIDED";
        InvoiceStatusEnum[InvoiceStatusEnum["NONE"] = 'NONE'] = "NONE";
        InvoiceStatusEnum[InvoiceStatusEnum["UNPAID"] = 'UNPAID'] = "UNPAID";
    })(InvoiceStatusEnum = InvoiceUpdateResponse.InvoiceStatusEnum || (InvoiceUpdateResponse.InvoiceStatusEnum = {}));
})(InvoiceUpdateResponse = exports.InvoiceUpdateResponse || (exports.InvoiceUpdateResponse = {}));
