// @hubspot/api-client/lib/codegen/crm/extensions/accounting/model/invoicesResponseExternal.js
"use strict";
/**
 * Accounting Extension
 * These APIs allow you to interact with HubSpot\'s Accounting Extension. It allows you to: * Specify the URLs that HubSpot will use when making webhook requests to your external accounting system. * Respond to webhook calls made to your external accounting system by HubSpot
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvoicesResponseExternal = void 0;
/**
* A response to a request for invoices.
*/
class InvoicesResponseExternal {
    static getAttributeTypeMap() {
        return InvoicesResponseExternal.attributeTypeMap;
    }
}
exports.InvoicesResponseExternal = InvoicesResponseExternal;
InvoicesResponseExternal.discriminator = undefined;
InvoicesResponseExternal.attributeTypeMap = [
    {
        "name": "result",
        "baseName": "@result",
        "type": "InvoicesResponseExternal.ResultEnum"
    },
    {
        "name": "invoices",
        "baseName": "invoices",
        "type": "Array<AccountingExtensionInvoice>"
    }
];
(function (InvoicesResponseExternal) {
    let ResultEnum;
    (function (ResultEnum) {
        ResultEnum[ResultEnum["OK"] = 'OK'] = "OK";
        ResultEnum[ResultEnum["ERR"] = 'ERR'] = "ERR";
    })(ResultEnum = InvoicesResponseExternal.ResultEnum || (InvoicesResponseExternal.ResultEnum = {}));
})(InvoicesResponseExternal = exports.InvoicesResponseExternal || (exports.InvoicesResponseExternal = {}));
