// @hubspot/api-client/lib/codegen/crm/extensions/accounting/model/productSearchResponse.js
"use strict";
/**
 * Accounting Extension
 * These APIs allow you to interact with HubSpot\'s Accounting Extension. It allows you to: * Specify the URLs that HubSpot will use when making webhook requests to your external accounting system. * Respond to webhook calls made to your external accounting system by HubSpot
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductSearchResponse = void 0;
/**
* A response to a search for products.
*/
class ProductSearchResponse {
    static getAttributeTypeMap() {
        return ProductSearchResponse.attributeTypeMap;
    }
}
exports.ProductSearchResponse = ProductSearchResponse;
ProductSearchResponse.discriminator = undefined;
ProductSearchResponse.attributeTypeMap = [
    {
        "name": "result",
        "baseName": "@result",
        "type": "ProductSearchResponse.ResultEnum"
    },
    {
        "name": "products",
        "baseName": "products",
        "type": "Array<Product>"
    }
];
(function (ProductSearchResponse) {
    let ResultEnum;
    (function (ResultEnum) {
        ResultEnum[ResultEnum["OK"] = 'OK'] = "OK";
        ResultEnum[ResultEnum["ERR"] = 'ERR'] = "ERR";
    })(ResultEnum = ProductSearchResponse.ResultEnum || (ProductSearchResponse.ResultEnum = {}));
})(ProductSearchResponse = exports.ProductSearchResponse || (exports.ProductSearchResponse = {}));
