// @hubspot/api-client/lib/codegen/crm/extensions/accounting/model/resultIdAccountingResponse.js
"use strict";
/**
 * Accounting Extension
 * These APIs allow you to interact with HubSpot\'s Accounting Extension. It allows you to: * Specify the URLs that HubSpot will use when making webhook requests to your external accounting system. * Respond to webhook calls made to your external accounting system by HubSpot
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResultIdAccountingResponse = void 0;
/**
* A response to the creation of an entity (eg. invoice, customer).
*/
class ResultIdAccountingResponse {
    static getAttributeTypeMap() {
        return ResultIdAccountingResponse.attributeTypeMap;
    }
}
exports.ResultIdAccountingResponse = ResultIdAccountingResponse;
ResultIdAccountingResponse.discriminator = undefined;
ResultIdAccountingResponse.attributeTypeMap = [
    {
        "name": "result",
        "baseName": "@result",
        "type": "ResultIdAccountingResponse.ResultEnum"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    }
];
(function (ResultIdAccountingResponse) {
    let ResultEnum;
    (function (ResultEnum) {
        ResultEnum[ResultEnum["OK"] = 'OK'] = "OK";
        ResultEnum[ResultEnum["ERR"] = 'ERR'] = "ERR";
    })(ResultEnum = ResultIdAccountingResponse.ResultEnum || (ResultIdAccountingResponse.ResultEnum = {}));
})(ResultIdAccountingResponse = exports.ResultIdAccountingResponse || (exports.ResultIdAccountingResponse = {}));
