// @hubspot/api-client/lib/codegen/crm/extensions/accounting/model/taxSearchResponse.js
"use strict";
/**
 * Accounting Extension
 * These APIs allow you to interact with HubSpot\'s Accounting Extension. It allows you to: * Specify the URLs that HubSpot will use when making webhook requests to your external accounting system. * Respond to webhook calls made to your external accounting system by HubSpot
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxSearchResponse = void 0;
/**
* A response to a search for taxes.
*/
class TaxSearchResponse {
    static getAttributeTypeMap() {
        return TaxSearchResponse.attributeTypeMap;
    }
}
exports.TaxSearchResponse = TaxSearchResponse;
TaxSearchResponse.discriminator = undefined;
TaxSearchResponse.attributeTypeMap = [
    {
        "name": "result",
        "baseName": "@result",
        "type": "TaxSearchResponse.ResultEnum"
    },
    {
        "name": "taxes",
        "baseName": "taxes",
        "type": "Array<Tax>"
    }
];
(function (TaxSearchResponse) {
    let ResultEnum;
    (function (ResultEnum) {
        ResultEnum[ResultEnum["OK"] = 'OK'] = "OK";
        ResultEnum[ResultEnum["ERR"] = 'ERR'] = "ERR";
    })(ResultEnum = TaxSearchResponse.ResultEnum || (TaxSearchResponse.ResultEnum = {}));
})(TaxSearchResponse = exports.TaxSearchResponse || (exports.TaxSearchResponse = {}));
