// @hubspot/api-client/lib/codegen/crm/extensions/accounting/model/termsResponse.js
"use strict";
/**
 * Accounting Extension
 * These APIs allow you to interact with HubSpot\'s Accounting Extension. It allows you to: * Specify the URLs that HubSpot will use when making webhook requests to your external accounting system. * Respond to webhook calls made to your external accounting system by HubSpot
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TermsResponse = void 0;
/**
* A response to a search for payment terms.
*/
class TermsResponse {
    static getAttributeTypeMap() {
        return TermsResponse.attributeTypeMap;
    }
}
exports.TermsResponse = TermsResponse;
TermsResponse.discriminator = undefined;
TermsResponse.attributeTypeMap = [
    {
        "name": "result",
        "baseName": "@result",
        "type": "TermsResponse.ResultEnum"
    },
    {
        "name": "terms",
        "baseName": "terms",
        "type": "Array<AccountingExtensionTerm>"
    }
];
(function (TermsResponse) {
    let ResultEnum;
    (function (ResultEnum) {
        ResultEnum[ResultEnum["OK"] = 'OK'] = "OK";
        ResultEnum[ResultEnum["ERR"] = 'ERR'] = "ERR";
    })(ResultEnum = TermsResponse.ResultEnum || (TermsResponse.ResultEnum = {}));
})(TermsResponse = exports.TermsResponse || (exports.TermsResponse = {}));
