// @hubspot/api-client/lib/codegen/crm/extensions/accounting/model/updatedContact.js
"use strict";
/**
 * Accounting Extension
 * These APIs allow you to interact with HubSpot\'s Accounting Extension. It allows you to: * Specify the URLs that HubSpot will use when making webhook requests to your external accounting system. * Respond to webhook calls made to your external accounting system by HubSpot
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdatedContact = void 0;
/**
* A request to import external accounting contact properties in HubSpot
*/
class UpdatedContact {
    static getAttributeTypeMap() {
        return UpdatedContact.attributeTypeMap;
    }
}
exports.UpdatedContact = UpdatedContact;
UpdatedContact.discriminator = undefined;
UpdatedContact.attributeTypeMap = [
    {
        "name": "syncAction",
        "baseName": "syncAction",
        "type": "UpdatedContact.SyncActionEnum"
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "Date"
    },
    {
        "name": "emailAddress",
        "baseName": "emailAddress",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "customerType",
        "baseName": "customerType",
        "type": "UpdatedContact.CustomerTypeEnum"
    }
];
(function (UpdatedContact) {
    let SyncActionEnum;
    (function (SyncActionEnum) {
        SyncActionEnum[SyncActionEnum["CREATE"] = 'CREATE'] = "CREATE";
        SyncActionEnum[SyncActionEnum["UPDATE"] = 'UPDATE'] = "UPDATE";
        SyncActionEnum[SyncActionEnum["DELETE"] = 'DELETE'] = "DELETE";
    })(SyncActionEnum = UpdatedContact.SyncActionEnum || (UpdatedContact.SyncActionEnum = {}));
    let CustomerTypeEnum;
    (function (CustomerTypeEnum) {
        CustomerTypeEnum[CustomerTypeEnum["CONTACT"] = 'CONTACT'] = "CONTACT";
        CustomerTypeEnum[CustomerTypeEnum["COMPANY"] = 'COMPANY'] = "COMPANY";
    })(CustomerTypeEnum = UpdatedContact.CustomerTypeEnum || (UpdatedContact.CustomerTypeEnum = {}));
})(UpdatedContact = exports.UpdatedContact || (exports.UpdatedContact = {}));
