// @hubspot/api-client/lib/codegen/crm/extensions/accounting/model/updatedProduct.js
"use strict";
/**
 * Accounting Extension
 * These APIs allow you to interact with HubSpot\'s Accounting Extension. It allows you to: * Specify the URLs that HubSpot will use when making webhook requests to your external accounting system. * Respond to webhook calls made to your external accounting system by HubSpot
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdatedProduct = void 0;
/**
* A request to import external accounting product properties in HubSpot
*/
class UpdatedProduct {
    static getAttributeTypeMap() {
        return UpdatedProduct.attributeTypeMap;
    }
}
exports.UpdatedProduct = UpdatedProduct;
UpdatedProduct.discriminator = undefined;
UpdatedProduct.attributeTypeMap = [
    {
        "name": "syncAction",
        "baseName": "syncAction",
        "type": "UpdatedProduct.SyncActionEnum"
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "Date"
    },
    {
        "name": "price",
        "baseName": "price",
        "type": "number"
    },
    {
        "name": "currencyCode",
        "baseName": "currencyCode",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "properties",
        "baseName": "properties",
        "type": "{ [key: string]: string; }"
    }
];
(function (UpdatedProduct) {
    let SyncActionEnum;
    (function (SyncActionEnum) {
        SyncActionEnum[SyncActionEnum["CREATE"] = 'CREATE'] = "CREATE";
        SyncActionEnum[SyncActionEnum["UPDATE"] = 'UPDATE'] = "UPDATE";
        SyncActionEnum[SyncActionEnum["DELETE"] = 'DELETE'] = "DELETE";
    })(SyncActionEnum = UpdatedProduct.SyncActionEnum || (UpdatedProduct.SyncActionEnum = {}));
})(UpdatedProduct = exports.UpdatedProduct || (exports.UpdatedProduct = {}));
